From d89c37e285dcae733be3569c5c8ae93d9cd62d32 Mon Sep 17 00:00:00 2001
From: Duy Nguyen <duynt26@fsoft.com.vn>
Date: Tue, 8 Oct 2019 13:53:22 +0700
Subject: [PATCH 279/557] ARM: dts: r8a77470: add vin0 description

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Pham Thanh Tam [FGA.AIS] <tampt8@fsoft.com.vn>
Signed-off-by: Duy Nguyen <duynt26@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 1f59b91..94f45d0 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -26,6 +26,7 @@
 		spi2 = &msiof0;
 		spi3 = &msiof1;
 		spi4 = &msiof2;
+		vin0 = &vin0;
 	};
 
 	cpus {
@@ -393,6 +394,15 @@
 			};
 		};
 
+		vin0: video@e6ef0000 {
+			compatible = "renesas,vin-r8a77470";
+			reg = <0 0xe6ef0000 0 0x1000>;
+			interrupts = <0 188 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp8_clks R8A77470_CLK_VIN0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
 		i2c0: i2c@e6508000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.7.4

