From 64946e3ee861d456856a2e8040b17f08b2562af1 Mon Sep 17 00:00:00 2001
From: TamPT8 <tampt8@fsoft.com.vn>
Date: Tue, 17 Dec 2019 12:42:35 +0700
Subject: [PATCH 314/557] usb: phy: Add support for r8a7744

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNv25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: TamPT8 <tampt8@fsoft.com.vn>
---
 drivers/phy/phy-rcar-gen2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index 1bfbf3a..2f9ade1 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -595,6 +595,10 @@ static const struct of_device_id rcar_gen2_phy_match_table[] = {
 		.data = &rcar_gen2_usb_phy_data,
 	},
 	{
+		.compatible = "renesas,usb-phy-r8a7744",
+		.data = &rcar_gen2_usb_phy_data,
+	},
+	{
 		.compatible = "renesas,rcar-gen2-usb-phy",
 		.data = &rcar_gen2_usb_phy_data,
 	},
-- 
2.7.4

