From d4fbf3d44bafa2ffac065258d2e2e3011eec69f6 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Fri, 4 Dec 2015 16:11:41 +0000
Subject: [PATCH] Build: look for python >= 3.5 when building python bindings

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
Signed-off-by: Duy Bao Dang <duy.dang.yb@renesas.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7c592ad..ad8cf7c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -359,7 +359,7 @@ AM_CONDITIONAL([PYTHON_BINDING], [test "x${enable_python_binding:-yes}" = xyes])

 if test "x${enable_python_binding:-yes}" = xyes; then
   AX_PKG_SWIG($swig_version, [], [ AC_MSG_ERROR([SWIG $swig_version or newer is needed]) ])
-  AM_PATH_PYTHON([3.0])
+  AM_PATH_PYTHON([3.5])

   AC_ARG_VAR([PYTHON_INCLUDE], [Include flags for python, bypassing python-config])
   AC_ARG_VAR([PYTHON_CONFIG], [Path to python-config])
--
2.7.4

