From 8aabb49b8549d85668bc4109791268acd6d57dce Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Sun, 21 Aug 2016 11:45:20 +0700
Subject: [PATCH 13/14] gstvspmfilter re add memory alignment

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 gst/vspmfilter/gstvspmfilter.c | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/gst/vspmfilter/gstvspmfilter.c b/gst/vspmfilter/gstvspmfilter.c
index 3217307..b4ba910 100644
--- a/gst/vspmfilter/gstvspmfilter.c
+++ b/gst/vspmfilter/gstvspmfilter.c
@@ -1541,7 +1541,15 @@ gst_vspm_filter_transform_frame (GstVideoFilter * filter,
     src_par.addr        = (void *) p_adr[0].hard_addr;
     src_par.stride      = stride[0];
     if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) src_par.stride_c = stride[1];
-    if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) src_par.addr_c0  = (void*) ((guint)src_par.addr + offset[1]);
+    if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) {
+        gint uv_offset = 0;
+        if (((in_width * in_height) & (4096 - 1))
+                && (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) == 2))
+            uv_offset = get_uv_offset_aligned_to_page (4096, in_width, in_height);
+        else
+            uv_offset = offset[1];
+        src_par.addr_c0  = (void*) ((guint)src_par.addr + uv_offset);
+    }
     if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 2) src_par.addr_c1  = (void*) ((guint)src_par.addr + offset[2]);
 
     src_par.csc         = VSP_CSC_OFF;  /* do not convert colorspace */
@@ -1790,7 +1798,15 @@ gst_vspm_filter_transform_buffer (GstVideoFilter * filter,
     src_par.addr        = (void *) p_adr[0].hard_addr;
     src_par.stride      = stride[0];
     if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) src_par.stride_c = stride[1];
-    if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) src_par.addr_c0  = (void*) ((guint)src_par.addr + offset[1]);
+    if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 1) {
+        gint uv_offset = 0;
+        if (((in_width * in_height) & (4096 - 1))
+                && (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) == 2))
+            uv_offset = get_uv_offset_aligned_to_page (4096, in_width, in_height);
+        else
+            uv_offset = offset[1];
+        src_par.addr_c0  = (void*) ((guint)src_par.addr + uv_offset);
+    }
     if (GST_VIDEO_FORMAT_INFO_N_PLANES(vspm_in_vinfo) > 2) src_par.addr_c1  = (void*) ((guint)src_par.addr + offset[2]);
 
     /* not convert colorspace */
-- 
1.9.1

