From bf772eaa4eb17552c27e5bf785dddcc95c5fd783 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Mon, 11 Jul 2016 16:42:47 +0300
Subject: [PATCH] switch of usbhs ip block

---
 arch/arm/boot/dts/r8a7743-skrzg1m.dts | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7743-skrzg1m.dts b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
index 79c7cc3..d21fc6d 100644
--- a/arch/arm/boot/dts/r8a7743-skrzg1m.dts
+++ b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
@@ -421,8 +421,6 @@
 &hsusb {
 	pinctrl-0 = <&usb0_pins>;
 	pinctrl-names = "default";
-
-	status = "okay";
 };
 
 &usbphy {
-- 
1.9.1

