From 69e3ff7a8bf9299f79cd9e0e26fb0f55392c6b23 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Mon, 18 Jul 2016 20:09:12 +0300
Subject: [PATCH] add gpio keys and da9063 pmic

---
 arch/arm/boot/dts/r8a7794-silk.dts | 42 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7794-silk.dts b/arch/arm/boot/dts/r8a7794-silk.dts
index f8da429..d4cbbba 100644
--- a/arch/arm/boot/dts/r8a7794-silk.dts
+++ b/arch/arm/boot/dts/r8a7794-silk.dts
@@ -42,6 +42,31 @@
 		device_type = "memory";
 		reg = <0 0x40000000 0 0x40000000>;
 	};
+	gpio-keys {
+		compatible = "gpio-keys";
+		autorepeat;
+		key@0 {
+                       gpios = <&gpio5 12 GPIO_ACTIVE_LOW>;
+                       linux,code = <2>;
+                       label = "SW6";
+                       gpio-key,wakeup;
+                       debounce-interval = <20>;
+               };
+               key@2{
+                       gpios = <&gpio5 11 GPIO_ACTIVE_LOW>;
+                       linux,code = <3>;
+                       label = "SW4";
+                       gpio-key,wakeup;
+                       debounce-interval = <20>;
+               };
+               key@3 {
+                       gpios = <&gpio5 10 GPIO_ACTIVE_LOW>;
+                       linux,code = <4>;
+                       label = "SW3";
+                       gpio-key,wakeup;
+                       debounce-interval = <20>;
+               };
+	};
 
 	d3_3v: regulator@0 {
 		compatible = "regulator-fixed";
@@ -303,6 +328,23 @@
 		};
 	};
 };
+&i2c7 {
+        status = "okay";
+        clock-frequency = <100000>;
+
+        pmic@58 {
+                compatible = "dlg,da9063";
+                reg = <0x58>;
+                interrupt-parent = <&gpio3>;
+                interrupts = <31 IRQ_TYPE_LEVEL_LOW>;
+                rtc {
+                        compatible = "dlg,da9063-rtc";
+                };
+                wdt {
+                        compatible = "dlg,da9063-watchdog";
+                };
+        };
+};
 
 &mmcif0 {
 	pinctrl-0 = <&mmcif0_pins>;
-- 
2.7.4

