From 46f8037039f51db36a23664ff459f6023195ec36 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 28 Jul 2016 09:33:30 +0300
Subject: [PATCH] app operation points for cpufreq working

---
 arch/arm/boot/dts/r8a7794.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/r8a7794.dtsi b/arch/arm/boot/dts/r8a7794.dtsi
index 4ef2e72..9118ce1 100644
--- a/arch/arm/boot/dts/r8a7794.dtsi
+++ b/arch/arm/boot/dts/r8a7794.dtsi
@@ -47,6 +47,7 @@
 			clock-frequency = <1000000000>;
 			power-domains = <&z2>;
 			next-level-cache = <&L2_CA7>;
+			operating-points = <1000000 1000000>;
 		};
 
 		cpu1: cpu@1 {
-- 
1.9.1

