From 303b6fd756c3318b67f1435eaff8c2de0ef7af8f Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Wed, 2 Dec 2015 13:59:58 +0300
Subject: [PATCH 1/4] add ravb compatible for r8a7743

---
 drivers/net/ethernet/renesas/ravb_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index def788f..7d60de1 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -2152,6 +2152,7 @@ static struct ravb_plat_data *ravb_parse_dt(struct device *dev,
 
 static const struct of_device_id ravb_match_table[] = {
 	{ .compatible = "renesas,gether-r8a7790", .data = &r8a779x_data_giga },
+	{ .compatible = "renesas,gether-r8a7743", .data = &r8a774_3_5_data_giga },
 	{ .compatible = "renesas,gether-r8a7794", .data = &r8a779x_data_giga },
 	{ .compatible = "renesas,gether-r8a7745", .data = &r8a774_3_5_data_giga },
 	{ }
-- 
1.9.1

