From 997f873dc74938e5a6c8692d1908d6e8862880f6 Mon Sep 17 00:00:00 2001
From: Chuan Anh Nguyen <chuan.nguyen.jy@rvc.renesas.com>
Date: Thu, 25 Aug 2016 10:26:43 +0700
Subject: [PATCH] r8a7745.dtsi: register for 2ddmac, vsp1, vspd0

---
 arch/arm/boot/dts/r8a7745.dtsi | 38 +++++++++++++++++++++++++++++++++++++-
 1 file changed, 37 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 764c378..eff6ac9 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -853,6 +853,42 @@
 		};
 	};
 
+	dmac2d: dma-controller@fea00000 {
+		compatible = "renesas,2ddmac-r8a7745";
+		reg = <0 0xfea00000 0 0x0200>;
+		interrupts = <0 285 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7745_CLK_2DDMAC>;
+		power-domains = <&cpg_clocks>;
+	};
+
+	vsps: vspm@fe928000 {
+		compatible = "renesas,vsps";
+		reg = <0 0xfe928000 0 0x8000>;
+		interrupts = <0 267 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7745_CLK_VSP1_S>;
+		power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
+
+		renesas,has-lut;
+		renesas,has-sru;
+		renesas,#rpf = <5>;
+		renesas,#uds = <3>;
+		renesas,#wpf = <4>;
+	};
+
+	vspd0: vspm@fe930000 {
+		compatible = "renesas,vspd0";
+		reg = <0 0xfe930000 0 0x8000>;
+		interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7745_CLK_VSP1_DU0>;
+		power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
+
+		renesas,has-lif;
+		renesas,has-lut;
+		renesas,#rpf = <4>;
+		renesas,#uds = <1>;
+		renesas,#wpf = <4>;
+	};
+
 	du: display@feb00000 {
 		compatible = "renesas,du-r8a7745";
 		reg = <0 0xfeb00000 0 0x40000>;
@@ -1142,7 +1178,7 @@
 			>;
 			clock-output-names =
 				"vcp0", "vpc0", "tmu1", "3dg", "2ddmac", "fdp1-0",
-				"tmu3", "tmu2", "cmt0", "tmu0", "vsp1-du0", "vsps";
+				"tmu3", "tmu2", "cmt0", "tmu0", "vsp1-du0", "vsp1-sy";
 		};
 		mstp2_clks: mstp2_clks@e6150138 {
 			compatible = "renesas,r8a7745-mstp-clocks", "renesas,cpg-mstp-clocks";
-- 
1.9.1

