From ca77255e864c1543ce05e8455f5b4541ef30ea73 Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@renesas.com>
Date: Tue, 8 Oct 2019 13:53:22 +0700
Subject: [PATCH 241/475] ARM: dts: r8a77470: Add VIN0 node

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index f7af911..5d49b7c 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -26,6 +26,7 @@
 		spi2 = &msiof0;
 		spi3 = &msiof1;
 		spi4 = &msiof2;
+		vin0 = &vin0;
 	};
 
 	cpus {
@@ -1259,6 +1260,15 @@
 					"scu-src1";
 			};
 		};
+
+		vin0: video@e6ef0000 {
+			compatible = "renesas,vin-r8a77470";
+			reg = <0 0xe6ef0000 0 0x1000>;
+			interrupts = <0 188 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp8_clks R8A77470_CLK_VIN0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
 	};
 
 	timer {
-- 
2.7.4

