From ebd81aeb686845f197b8d1391ab8a6d6a58b5795 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Fri, 31 Jul 2020 17:56:37 +0700
Subject: [PATCH 398/475] ARM: dts: r8a7742: Add VSP nodes

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index a27de87..b2021c5 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1036,6 +1036,8 @@
 			clock-names = "du.0", "du.1", "lvds.0", "lvds.1";
 			status = "disabled";
 
+			vsps = <&vspd0 &vspd1>;
+
 			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -1601,6 +1603,38 @@
 			};
 		};
 
+		vspm@fe920000 {
+			compatible = "renesas,vspm-vspr", "renesas-vspm";
+			reg = <0 0xfe920000 0 0x8000>;
+			interrupts = <0 266 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_S>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspm@fe928000 {
+			compatible = "renesas,vspm-vsps", "renesas-vspm";
+			reg = <0 0xfe928000 0 0x7404>;
+			interrupts = <0 267 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_S>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspd0: vspd0@fe930000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe930000 0 0x7404>;
+			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_DU0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspd1: vspd1@fe938000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe938000 0 0x7404>;
+			interrupts = <0 247 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_DU1>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
 		scifb0: serial@e6c20000 {
 			compatible = "renesas,scifb-r8a7742",
 				     "renesas,scifb";
-- 
2.7.4

