From a366d06147be63939b56f903ab9b4df7d7554748 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 29 Oct 2020 17:31:47 +0700
Subject: [PATCH 462/475] ARM: dts: r8a7742: Add xhci node which supports
 USB3.0

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 03eb81b..7e8e33f 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -809,6 +809,17 @@
 			};
 		};
 
+		xhci: xhci@ee000000 {
+			compatible = "renesas,xhci-r8a7742";
+			reg = <0 0xee000000 0 0xc00>;
+			interrupts = <0 101 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp3_clks R8A7742_CLK_SSUSB>;
+			phys = <&usb2 1>;
+			phy-names = "usb";
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
 		pci1: pci@ee0b0000 {
 			compatible = "renesas,pci-r8a7742",
 				     "renesas,pci-rcar-gen2";
-- 
2.7.4

