From 9874ca3f5555830c3e6e554eecea9b487656d921 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 15 Oct 2020 17:11:35 +0700
Subject: [PATCH 465/475] ARM: dts: r8a7742: Add hsusb node

Signed-off-by: tin le <tin.le.wh@rvc.renesas.com>
Signed-off-by: quynh nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 7e8e33f..c992956 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -809,6 +809,18 @@
 			};
 		};
 
+		hsusb: usb@e6590000 {
+			compatible = "renesas,usbhs-r8a7742";
+			reg = <0 0xe6590000 0 0x100>;
+			interrupts = <0 107 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp7_clks R8A7742_CLK_HSUSB>;
+			renesas,buswait = <4>;
+			phys = <&usb0 1>;
+			phy-names = "usb";
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
 		xhci: xhci@ee000000 {
 			compatible = "renesas,xhci-r8a7742";
 			reg = <0 0xee000000 0 0xc00>;
-- 
2.7.4

