From 2104fe51826a06730923ec93059bb1199e18a886 Mon Sep 17 00:00:00 2001
From: ChinhPC <chinhpc@fsoft.com.vn>
Date: Tue, 8 Oct 2019 10:34:02 +0700
Subject: [PATCH 228/559] ARM: dts: r8a77470: Add DU to dtsi

Signed-off-by: Hung Dong <hung.dong.xd@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Pham Thanh Tam [FGA.AIS] <tampt8@fsoft.com.vn>
Signed-off-by: ChinhPC <chinhpc@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 35f0e46..f2913a7 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -249,6 +249,32 @@
 			status = "disabled";
 		};
 
+		vspdu0: vspdu0@fe930000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe930000 0 0x7404>;
+			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A77470_CLK_VSP1DU0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+		};
+
+		du: display@feb00000 {
+			compatible = "renesas,du-r8a77470";
+			reg = <0 0xfeb00000 0 0x40000>;
+			reg-names = "du";
+			interrupts = <0 256 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp7_clks R8A77470_CLK_DU0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			clock-names = "du.0";
+			status = "disabled";
+
+			vsps = <&vspdu0>;
+
+			port@0 {
+				du_out_rgb0: endpoint {
+				};
+			};
+		};
+
 		i2c0: i2c@e6508000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.7.4

