From 7ab4fd079e0b1a174f3df495d1f7d093b9683849 Mon Sep 17 00:00:00 2001
From: Duy Nguyen <duynt26@fsoft.com.vn>
Date: Wed, 9 Oct 2019 09:39:07 +0700
Subject: [PATCH 258/559] ARM: dts: r8a77470: Add spi to dtsi

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Pham Thanh Tam [FGA.AIS] <tampt8@fsoft.com.vn>
Signed-off-by: Duy Nguyen <duynt26@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index b9e56ba..710bd2a 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -21,6 +21,8 @@
 		i2c2 = &i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
+		spi0 = &qspi0;
+		spi1 = &qspi1;
 	};
 
 	cpus {
@@ -630,7 +632,7 @@
 			dmas = <&dmac0 0x17>, <&dmac0 0x18>,
 			       <&dmac1 0x17>, <&dmac1 0x18>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			num-cs = <1>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -645,7 +647,7 @@
 			dmas = <&dmac0 0xd1>, <&dmac0 0xd2>,
 			       <&dmac1 0xd1>, <&dmac1 0xd2>;
 			dma-names = "tx", "rx", "tx", "rx";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			num-cs = <1>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-- 
2.7.4

