From c008a4a8116285ae5e6d6c798670d88438a2f2e5 Mon Sep 17 00:00:00 2001
From: Kha Tran <kha.tran.px@rvc.renesas.com>
Date: Thu, 25 Jan 2018 09:46:21 +0700
Subject: [PATCH 320/559] ARM: DTS: r8a7744: Add description for PowerVR (SGX
 module)

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a7744.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 9cd6754..c3b3619 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -321,6 +321,17 @@
 			power-domains = <&cpg_clocks>;
 		};
 
+		sgx@fd800000 {
+			compatible = "renesas,sgx";
+			reg = <0 0xfd800000 0 0x10000>;
+			interrupts = <0 119 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "irq_sgx";
+			clocks = <&mstp1_clks R8A7744_CLK_3DG>,
+				 <&mstp8_clks R8A7744_CLK_IPMMU_SGX>;
+			clock-names = "sgx_clk", "ipmmu_clk";
+			power-domains = <&sysc R8A7744_PD_SGX>;
+		};
+
 		thermal: thermal@e61f0000 {
 			compatible = "renesas,thermal-r8a7744",
 				     "renesas,rcar-gen2-thermal";
-- 
2.7.4

