From 6a7635ead2e4c28086ce074e93c7b891f22cc501 Mon Sep 17 00:00:00 2001
From: TamPT8 <tampt8@fsoft.com.vn>
Date: Mon, 16 Dec 2019 16:46:33 +0700
Subject: [PATCH 390/559] phy: phy_rcar_gen2: add support for r8a7742

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: TamPT8 <tampt8@fsoft.com.vn>
---
 drivers/phy/phy-rcar-gen2.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index 40306da..c923608 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -235,6 +235,7 @@ static int rcar_gen2_phy_power_on(struct phy *p)
 
 	/* USBHS_UGCTRL_CONNECT bit only exists on RZG1H,M,N,E SoC USBHS_UGSTS reg */
 	if (of_machine_is_compatible("renesas,r8a7743")
+		|| of_machine_is_compatible("renesas,r8a7742")
 		|| of_machine_is_compatible("renesas,r8a7744")
 		|| of_machine_is_compatible("renesas,r8a7745")) {
 		for (i = 0; i < 20; i++) {
@@ -275,6 +276,7 @@ static int rcar_gen2_phy_power_off(struct phy *p)
 
 	/* Power off USBHS PHY */
 	if (of_machine_is_compatible("renesas,r8a7743")
+		|| of_machine_is_compatible("renesas,r8a7742")
 		|| of_machine_is_compatible("renesas,r8a7744")
 		|| of_machine_is_compatible("renesas,r8a7745")) {
 		value = readl(usbhs_base + USBHS_UGCTRL);
@@ -597,6 +599,10 @@ static const struct of_device_id rcar_gen2_phy_match_table[] = {
 		.data = &rcar_gen2_usb_phy_data,
 	},
 	{
+		.compatible = "renesas,usb-phy-r8a7742",
+		.data = &rcar_gen2_usb_phy_data,
+	},
+	{
 		.compatible = "renesas,usb-phy-r8a7794",
 		.data = &rcar_gen2_usb_phy_data,
 	},
@@ -861,6 +867,7 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 			return error;
 		}
 		if (of_machine_is_compatible("renesas,r8a7743")
+			|| of_machine_is_compatible("renesas,r8a7742")
 			|| of_machine_is_compatible("renesas,r8a7744")
 			|| of_machine_is_compatible("renesas,r8a7745")) {
 			channel->select_mask = select_mask[channel_num];
@@ -876,6 +883,7 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 			phy->channel = channel;
 			phy->number = n;
 			if (of_machine_is_compatible("renesas,r8a7743")
+				|| of_machine_is_compatible("renesas,r8a7742")
 				|| of_machine_is_compatible("renesas,r8a7744")
 				|| of_machine_is_compatible("renesas,r8a7745"))
 				phy->select_value = select_value[channel_num][n];
-- 
2.7.4

