From 2209174694627e8534066834ea8b9c2b3cbb96a1 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 16 May 2022 16:17:34 +0700
Subject: [PATCH] s3ctl: Port kernel module s3ctl to Linux kernel v5.10

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drv/s3ctl_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drv/s3ctl_drv.c b/drv/s3ctl_drv.c
index a7b4bee..7866411 100755
--- a/drv/s3ctl_drv.c
+++ b/drv/s3ctl_drv.c
@@ -90,7 +90,7 @@ static int map_register(void)
 	if (!request_mem_region(S3_ADDR, S3_SIZE, DRVNAME))
 		return -1;
 
-	top = (unsigned long)ioremap_nocache(S3_ADDR, S3_SIZE);
+	top = (unsigned long)ioremap(S3_ADDR, S3_SIZE);
 	if (!top) {
 		release_mem_region(S3_ADDR, S3_SIZE);
 		return -1;
@@ -98,7 +98,7 @@ static int map_register(void)
 
 	xytl = (struct S3_XYTL *)(top + S3_OFFSET);
 
-	top_prr = (unsigned long)ioremap_nocache(S3_PRR_ADDR, S3_PRR_SIZE);
+	top_prr = (unsigned long)ioremap(S3_PRR_ADDR, S3_PRR_SIZE);
 	if (!top_prr)
 		return -1;
 
-- 
2.7.4

