From 42156bb04781f8ee1d9e6a7f74fd14428a213e84 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Fri, 2 Nov 2018 17:59:47 +0700
Subject: [PATCH 028/628] r8a7745.dtsi: register for 2ddmac

Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7745.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index dfdef01..7f1c3e2 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1349,6 +1349,14 @@
 			status = "disabled";
 		};
 
+		dmac2d: dma-controller@fea00000 {
+			compatible = "renesas,2ddmac-r8a7745";
+			reg = <0 0xfea00000 0 0x0200>;
+			interrupts = <0 285 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7745_CLK_2DDMAC>;
+			power-domains = <&cpg_clocks>;
+		};
+
 		vsp@fe928000 {
 			compatible = "renesas,vsp1";
 			reg = <0 0xfe928000 0 0x8000>;
-- 
2.7.4

