From e2d7f6da4a19aa2968d4703eb3b75f80ae3e91b1 Mon Sep 17 00:00:00 2001
From: Tin Le <tin.le.wh@rvc.renesas.com>
Date: Fri, 20 Apr 2018 10:07:14 +0700
Subject: [PATCH 532/629] ARM: DTS: r8a7742: add gpio for usb otg

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index dffae04792a5..0e61ab4f56ae 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -403,6 +403,9 @@
 
 &usbphy {
 	status = "okay";
+	renesas,id-gpio = <&gpio5 20 GPIO_ACTIVE_LOW>;
+	renesas,vbus-gpio = <&gpio5 18 GPIO_ACTIVE_LOW>;
+	renesas,wakeup;
 };
 
 &xhci {
-- 
2.17.1

