From bbf6d9816d25a99afc45f6dd2e21b21c3b94864e Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Tue, 13 Oct 2020 10:36:01 +0700
Subject: [PATCH 078/476] ARM: dts: iwg21m: Add RTC node into i2c0

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dtsi b/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
index 518d140..8ddc3f5 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
@@ -41,6 +41,11 @@
 
 	clock-frequency = <400000>;
 	status = "okay";
+
+	rtc@68 {
+		compatible = "bq32000";
+		reg = <0x68>;
+	};
 };
 
 &i2c1 {
-- 
2.7.4

