From f4a61073d7849f169685dfc14cc86ee1adabb54a Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Mon, 9 Apr 2018 10:54:06 +0700
Subject: [PATCH 110/476] ARM: dts: r8a7742: Add Audio CTU support

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 45524e3..2203e70 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1175,6 +1175,8 @@
 					<&mstp10_clks R8A7742_CLK_SCU_SRC0>,
 					<&mstp10_clks R8A7742_CLK_SCU_DVC1>,
 					<&mstp10_clks R8A7742_CLK_SCU_DVC0>,
+					<&mstp10_clks R8A7742_CLK_SCU_CTU1_MIX1>,
+					<&mstp10_clks R8A7742_CLK_SCU_CTU0_MIX0>,
 					<&audio_clk_a>, <&audio_clk_b>,
 					<&audio_clk_c>, <&m2_clk>;
 			clock-names =	"ssi-all",
@@ -1185,11 +1187,31 @@
 					"src.5", "src.4", "src.3", "src.2",
 					"src.1", "src.0",
 					"dvc.1", "dvc.0",
+					"ctu.1", "ctu.0",
 					"clk_a", "clk_b", "clk_c", "clk_i";
 			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 
 			status = "disabled";
 
+			rcar_sound,ctu {
+				ctu00: ctu@0 {
+				};
+				ctu01: ctu@1 {
+				};
+				ctu02: ctu@2 {
+				};
+				ctu03: ctu@3 {
+				};
+				ctu10: ctu@4 {
+				};
+				ctu11: ctu@5 {
+				};
+				ctu12: ctu@6 {
+				};
+				ctu13: ctu@7 {
+				};
+			};
+
 			rcar_sound,dvc {
 				dvc0: dvc@0 {
 				};
-- 
2.7.4

