From 00db152395f940274893b223987e9bb8c0ae5a33 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Wed, 22 Jul 2020 09:04:24 +0700
Subject: [PATCH 137/476] ARM: dts: iwg21m: Add qspi and msiof0 support

Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dtsi | 63 +++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dtsi b/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
index 8ddc3f5..1c8b69d 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dtsi
@@ -35,6 +35,20 @@
 	clock-frequency = <20000000>;
 };
 
+/*
+ * When “GP0_18” pin is set to low, QSPI is selected
+ * When “GP0_18” pin is set to high, VIN2 is selected
+*/
+
+&gpio0 {
+	enable-qspi{
+		gpio-hog;
+		gpios = <18 GPIO_ACTIVE_HIGH>;
+		output-low;
+		line-name = "enable-qspi";
+	};
+};
+
 &i2c0 {
 	pinctrl-0 = <&i2c0_pins>;
 	pinctrl-names = "default";
@@ -89,6 +103,16 @@
 		groups = "mmc1_data4", "mmc1_ctrl";
 		function = "mmc1";
 	};
+
+	msiof0_pins: spi1 {
+		groups = "msiof0_clk", "msiof0_tx", "msiof0_rx";
+		function = "msiof0";
+	};
+
+	qspi_pins: spi0 {
+		groups = "qspi_ctrl", "qspi_data2";
+		function = "qspi";
+	};
 };
 
 &mmcif1 {
@@ -100,3 +124,42 @@
 	non-removable;
 	status = "okay";
 };
+
+&msiof0 {
+	pinctrl-0 = <&msiof0_pins>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
+
+	status = "okay";
+
+	flash1: flash1@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "sst,sst25vf016b";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+		m25p,fast-read;
+	};
+};
+
+&qspi {
+	pinctrl-0 = <&qspi_pins>;
+	pinctrl-names = "default";
+
+	status = "okay"; /* multiplexed with VI2 */
+
+	flash: flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "sst,sst25vf016b";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+		m25p,fast-read;
+		spi-cpol;
+		spi-cpha;
+	};
+};
-- 
2.7.4

