From 5f37c14a75e19cbd3e7e8328acb13232ce07ebd0 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Wed, 14 Oct 2020 10:58:27 +0700
Subject: [PATCH 203/476] net: ethernet: ravb: Add R8A7744 support

Signed-off-by: thanhbui <thann.aui.fz@renesas.com>
---
 drivers/net/ethernet/renesas/ravb_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index d76262e..5f10a2a 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -1716,6 +1716,7 @@ static const struct of_device_id ravb_match_table[] = {
 	{ .compatible = "renesas,etheravb-r8a7795", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-rcar-gen3", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-r8a7743", .data = (void *)RCAR_GEN2 },
+	{ .compatible = "renesas,etheravb-r8a7744", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7745", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a77470", .data = (void *)RCAR_GEN2 },
 	{ }
-- 
2.7.4

