From 90b8642448705cf46a220bdb4006b13b55a19365 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Thu, 22 Oct 2020 15:43:04 +0700
Subject: [PATCH 269/476] ARM: dts: r8a7742-iwg21d-q7: enable cmt1

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index 3a9f150..464d1ba 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -116,6 +116,10 @@
 	status = "okay";
 };
 
+&cmt1{
+	status = "okay";
+};
+
 &gpio1 {
 	can1-trx-en-gpio{
 		gpio-hog;
-- 
2.7.4

