From 1955f61d3d253df785aee4d31dd51635ed09ec4a Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 29 Oct 2020 10:29:07 +0700
Subject: [PATCH 319/476] ARM: dts: r8a7742-iwg21d-q7: Enable PCI[01]

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index c272ef8..f04fd6d 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -241,6 +241,14 @@
 	};
 };
 
+&pci0 {
+	status = "okay";
+};
+
+&pci1 {
+	status = "okay";
+};
+
 &rcar_sound {
 	pinctrl-0 = <&sound_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

