From 91ecf689795a80491b67f73fca0c6d44dc602dc5 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 4 Aug 2020 16:28:46 +0700
Subject: [PATCH 402/476] ARM: dts: r8a7745: Use sysc power domain for VSP

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index abb3f0d..675aba1 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1429,7 +1429,7 @@
 			reg = <0 0xfe928000 0 0x8000>;
 			interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1_SY>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		vspm@fe930000 {
@@ -1437,7 +1437,7 @@
 			reg = <0 0xfe930000 0 0x8000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		fdp0: fdpm@fe940000 {
@@ -1453,7 +1453,7 @@
 			reg = <0 0xfe930000 0 0x7404>;
 			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		du: display@feb00000 {
-- 
2.7.4

