From 5ec3eeb0534233436ecd9c0d2d6ad5296ce6739e Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 29 Oct 2020 17:33:28 +0700
Subject: [PATCH 469/476] ARM: dts: r8a7742-iwg21d-q7: Enable USB PHY

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index 0b36e01..f02525d 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -328,6 +328,10 @@
 	};
 };
 
+&usbphy {
+	status = "okay";
+};
+
 &i2c2	{
 	status = "okay";
 	clock-frequency = <400000>;
-- 
2.7.4

