diff --git a/include/configs/rzv2m.h b/include/configs/rzv2m.h
index 46c3932..1782fd2 100755
--- a/include/configs/rzv2m.h
+++ b/include/configs/rzv2m.h
@@ -60,11 +60,11 @@
 	"loadaddr=0x180080000\0" \
         "fdt_addr=0x180000000\0" \
 	"fdt_file=r9a09g011gbg-evaluation-board.dtb\0" \
-	"kernel=Image\0" \
+	"kernel=Image-rzv2m.bin\0" \
 	"rtos=core1_uart.bin\0" \
 	"core1addr=0x0\0" \
 	"core1_vector=0x0\0" \
-	"bootargs_sd=setenv bootargs root=/dev/mmcblk0p2 rootwait rootfstype=ext3 rw\0" \
+	"bootargs_sd=setenv bootargs root=/dev/mmcblk0p2 rootwait rootfstype=ext4 rw\0" \
 	"bootsd=run bootargs_sd;" \
 	"fatload mmc 0:1 ${loadaddr} ${kernel};fatload mmc 0:1 ${fdt_addr} ${fdt_file};booti ${loadaddr} - ${fdt_addr}\0" \
     "bootargs_nfs=setenv bootargs root=/dev/nfs rw nfsroot=${serverip}:/nfs/rzv2m,nfsvers=3 ip=${ipaddr}:${serverip}::${netmask}:rzv2m:eth0\0" \
