diff --git a/cmd/Makefile b/cmd/Makefile
index 020a41e..a056118 100755
--- a/cmd/Makefile
+++ b/cmd/Makefile
@@ -213,6 +213,7 @@ obj-$(CONFIG_$(SPL_)CMD_TLV_EEPROM) += tlv_eeprom.o
 # core command
 obj-y += nvedit.o
 obj-y += ecc.o
+obj-y += evk.o
 
 obj-$(CONFIG_CMD_BCM_EXT_UTILS) += broadcom/
 
diff --git b/cmd/evk.c b/cmd/evk.c
new file mode 100644
index 0000000..8c612d1
--- /dev/null
+++ b/cmd/evk.c
@@ -0,0 +1,53 @@
+/*******************************************************************************
+* DISCLAIMER
+* This software is supplied by Renesas Electronics Corporation and is only
+* intended for use with Renesas products. No other uses are authorized. This
+* software is owned by Renesas Electronics Corporation and is protected under
+* all applicable laws, including copyright laws.
+* THIS SOFTWARE IS PROVIDED "AS IS" AND RENESAS MAKES NO WARRANTIES REGARDING
+* THIS SOFTWARE, WHETHER EXPRESS, IMPLIED OR STATUTORY, INCLUDING BUT NOT
+* LIMITED TO WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
+* AND NON-INFRINGEMENT. ALL SUCH WARRANTIES ARE EXPRESSLY DISCLAIMED.
+* TO THE MAXIMUM EXTENT PERMITTED NOT PROHIBITED BY LAW, NEITHER RENESAS
+* ELECTRONICS CORPORATION NOR ANY OF ITS AFFILIATED COMPANIES SHALL BE LIABLE
+* FOR ANY DIRECT, INDIRECT, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES FOR
+* ANY REASON RELATED TO THIS SOFTWARE, EVEN IF RENESAS OR ITS AFFILIATES HAVE
+* BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
+* Renesas reserves the right, without notice, to make changes to this software
+* and to discontinue the availability of this software. By using this software,
+* you agree to the additional terms and conditions found by accessing the
+* following link:
+* http://www.renesas.com/disclaimer
+* Copyright (C) 2022 Renesas Electronics Corporation. All rights reserved.
+*******************************************************************************/ 
+#include <common.h>
+#include <command.h>
+
+#define BASE_ADDR_PSC       0xA3700000
+
+extern void CMN_DelayInUSec(uint64_t us);
+
+static inline void psc_write32( uintptr_t addr, uint32_t val )
+{
+	*(volatile unsigned int*)(BASE_ADDR_PSC+addr) = val;
+}
+
+
+static int do_evk_shutdown(caddr_t *cmdtp, int flag, int argc, char * const argv[])
+
+{
+    
+    printf("Shutdown the evk board...\n");
+    psc_write32(0x50,0x1);
+    CMN_DelayInUSec(1000*1000); //wait 1sec
+    return 0;
+}
+
+U_BOOT_CMD(
+	evk_shutdown, 1, 0,	do_evk_shutdown,
+	"Shutdown the EVK board.",
+	"\n"
+	"\t- Shutdown the EVK board on u-boot"
+);
+
+
