diff --git a/common/image-fdt.c b/common/image-fdt.c
index 286c989..45637ad 100644
--- a/common/image-fdt.c
+++ b/common/image-fdt.c
@@ -85,8 +85,8 @@ static void boot_fdt_reserve_region(struct lmb *lmb, uint64_t addr,
 		      (unsigned long long)addr,
 		      (unsigned long long)size, flags);
 	} else {
-		puts("ERROR: reserving fdt memory region failed ");
-		printf("(addr=%llx size=%llx flags=%x)\n",
+		debug("ERROR: reserving fdt memory region failed ");
+		debug("(addr=%llx size=%llx flags=%x)\n",
 		       (unsigned long long)addr,
 		       (unsigned long long)size, flags);
 	}
