From f6cff749c8b701db3fc54abc7525edb9afc8b8d7 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renesas.com>
Date: Wed, 27 Dec 2017 17:00:25 +0700
Subject: [PATCH 235/641] ARM: DTS: r8a77470: Add device node to support CAN

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index a4f3647..df0dfb1 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -438,6 +438,16 @@
 		status = "disabled";
 	};
 
+	can0: can@e6e80000 {
+		compatible = "renesas,can-r8a77470";
+		reg = <0 0xe6e80000 0 0x1000>;
+		interrupts = <0 186 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp9_clks R8A77470_CLK_RCAN0>;
+		clock-names = "clkp1";
+		power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+		status = "disabled";
+	};
+
 	/* The memory map in the User's Manual maps the cores to bus numbers */
 	i2c0: i2c@e6508000 {
 		#address-cells = <1>;
-- 
2.7.4

