From e6a7276442f572941e511a216db13888ab8f4f3b Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Tue, 22 May 2018 10:45:56 +0700
Subject: [PATCH 572/641] ARM: DTS: iwg23s: enable tmu1, tmu2 and tm3

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470-iwg23s.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470-iwg23s.dts b/arch/arm/boot/dts/r8a77470-iwg23s.dts
index 6abe110..ef2bfb1 100644
--- a/arch/arm/boot/dts/r8a77470-iwg23s.dts
+++ b/arch/arm/boot/dts/r8a77470-iwg23s.dts
@@ -426,6 +426,18 @@
 	};
 };
 
+&tmu1 {
+	status = "okay";
+};
+
+&tmu2 {
+	status = "okay";
+};
+
+&tmu3 {
+	status = "okay";
+};
+
 &du {
 	pinctrl-0 = <&du0_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

