From 2a257697f894abe47984c96a2d8723ae3dc146fd Mon Sep 17 00:00:00 2001
From: "Pham Thanh Tam [FGA.AIS]" <tampt8@fsoft.com.vn>
Date: Mon, 22 Jul 2019 09:20:23 +0700
Subject: [PATCH 226/628] ARM: DTS: r8a77470: Enable DMA for serial ports

Signed-off-by: tungvo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Pham Thanh Tam [FGA.AIS] <tampt8@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 3d7bfd2..72890cf 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -430,6 +430,8 @@
 			interrupts = <0 154 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF0>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x39>, <&dmac0 0x3a>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
@@ -440,6 +442,8 @@
 			interrupts = <0 155 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF1>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x4d>, <&dmac0 0x4e>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
@@ -450,6 +454,8 @@
 			interrupts = <0 21 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF2>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x3b>, <&dmac0 0x3c>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
-- 
2.7.4

