From 9f5b542a42d7493b5aebc1aebaa24ccb0957b2ca Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Thu, 27 Sep 2018 14:53:53 +0700
Subject: [PATCH 295/628] watchdog: renesas wdt: add support for iwg23s

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 drivers/watchdog/renesas_wdt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/watchdog/renesas_wdt.c b/drivers/watchdog/renesas_wdt.c
index d57e69f..e27ad5a 100644
--- a/drivers/watchdog/renesas_wdt.c
+++ b/drivers/watchdog/renesas_wdt.c
@@ -290,7 +290,8 @@ static int rwdt_probe(struct platform_device *pdev)
 	 * register restart handler base on machine here
 	 * same ARM core architecture (e.g ARM cortex A7) can use same handler
 	 */
-	if (of_machine_is_compatible("renesas,r8a7745")) {
+	if (of_machine_is_compatible("renesas,r8a7745") ||
+		of_machine_is_compatible("renesas,r8a77470")) {
 		priv->restart_handler.notifier_call = rwdt_restart_handler_ca7;
 		/* 255: Highest priority restart handler */
 		priv->restart_handler.priority = 255;
-- 
2.7.4

