From 50edc7bebd8d92b96d957a0e37306ae8730bad55 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Thu, 27 Sep 2018 16:00:04 +0700
Subject: [PATCH 351/628] ARM: DTS: r8a7744-iwg20m: Enable PWM3

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7744-iwg20m.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744-iwg20m.dts b/arch/arm/boot/dts/r8a7744-iwg20m.dts
index 3f2d3c4..d583063 100644
--- a/arch/arm/boot/dts/r8a7744-iwg20m.dts
+++ b/arch/arm/boot/dts/r8a7744-iwg20m.dts
@@ -148,6 +148,11 @@
 		renesas,groups = "i2c2";
 		renesas,function = "i2c2";
 	};
+
+	pwm3_pins: pwm3 {
+		renesas,groups = "pwm3";
+		renesas,function = "pwm3";
+	};
 };
 
 &scif0 {
@@ -279,4 +284,11 @@
 	status = "okay";
 	clock-frequency = <400000>;
 
+};
+
+&pwm3 {
+	pinctrl-0 = <&pwm3_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
 };
\ No newline at end of file
-- 
2.7.4

