From 19fb797773e1c9b3b945de28ff9c7d89256506d1 Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Mon, 21 May 2018 16:26:28 +0700
Subject: [PATCH 417/628] net: ethernet: ravb: add r8a7742 to of_device_id

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/net/ethernet/renesas/ravb_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index eaea82f..b2d2e12 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -1687,6 +1687,7 @@ static const struct of_device_id ravb_match_table[] = {
 	{ .compatible = "renesas,etheravb-rcar-gen2", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7795", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-rcar-gen3", .data = (void *)RCAR_GEN3 },
+	{ .compatible = "renesas,etheravb-r8a7742", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7743", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7744", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7745", .data = (void *)RCAR_GEN2 },
-- 
2.7.4

