From 0bb8803e403eb1a17d1ce249371ac39e6e5568b5 Mon Sep 17 00:00:00 2001
From: Tung Vo <tung.vo.eb@rvc.renesas.com>
Date: Tue, 20 Feb 2018 03:11:43 +0700
Subject: [PATCH 429/628] ARM: DTS: r8a7742: Add device node to support PWM

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 73 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 73 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index f7e8d91a..a121068 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -172,6 +172,79 @@
 		status = "disabled";
 	};
 
+	tpu: pwm@e60f0000 {
+		compatible = "renesas,tpu-r8a7742", "renesas,tpu";
+		reg = <0 0xe60f0000 0 0x40>;
+		interrupts = <0 135 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp3_clks R8A7742_CLK_TPU0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <3>;
+	};
+
+	pwm0: pwm@e6e30000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e30000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm1: pwm@e6e31000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e31000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm2: pwm@e6e32000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e32000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm3: pwm@e6e33000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e33000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm4: pwm@e6e34000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e34000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm5: pwm@e6e35000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e35000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm6: pwm@e6e36000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e36000 0 0x8>;
+		clocks = <&mstp5_clks R8A7742_CLK_PWM>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
 	clocks {
 		#address-cells = <2>;
 		#size-cells = <2>;
-- 
2.7.4

