From bc17efdd6bea918f857b11227ec626768a6d3091 Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Fri, 25 May 2018 11:19:15 +0700
Subject: [PATCH 453/628] spi: sh-msiof: add compatible for r8a7742

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/spi/spi-sh-msiof.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-sh-msiof.c b/drivers/spi/spi-sh-msiof.c
index 8fe2c36..0c408f9 100644
--- a/drivers/spi/spi-sh-msiof.c
+++ b/drivers/spi/spi-sh-msiof.c
@@ -1010,6 +1010,7 @@ static const struct of_device_id sh_msiof_match[] = {
 	{ .compatible = "renesas,msiof-r8a7744",   .data = &r8a779x_data },
 	{ .compatible = "renesas,msiof-r8a7745",   .data = &r8a779x_data },
 	{ .compatible = "renesas,msiof-r8a77470",   .data = &r8a779x_data },
+	{ .compatible = "renesas,msiof-r8a7742",   .data = &r8a779x_data },
 	{ .compatible = "renesas,msiof-r8a7790",   .data = &r8a779x_data },
 	{ .compatible = "renesas,msiof-r8a7791",   .data = &r8a779x_data },
 	{ .compatible = "renesas,msiof-r8a7792",   .data = &r8a779x_data },
-- 
2.7.4

