From 4e999e83dee85054028f37c5d46eed6cca56f668 Mon Sep 17 00:00:00 2001
From: khanhluu <khanh.luu.xw@rvc.renesas.com>
Date: Sat, 10 Mar 2018 14:00:01 +0700
Subject: [PATCH 458/628] ARM: DTS: add power domain for pcie in dtsi

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 47971d6..27cb44f 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1201,6 +1201,7 @@
 		reg = <0 0xfe000000 0 0x80000>;
 		#address-cells = <3>;
 		#size-cells = <2>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		bus-range = <0x00 0xff>;
 		device_type = "pci";
 		ranges = <0x01000000 0 0x00000000 0 0xfe100000 0 0x00100000
-- 
2.7.4

