From a6e4c6d43e856af716b1c44d4550518ef04b55af Mon Sep 17 00:00:00 2001
From: Tin Le <tin.le.wh@rvc.renesas.com>
Date: Fri, 6 Apr 2018 10:48:57 +0700
Subject: [PATCH 466/628] ARM: DTS: iwg21m: enable USB PHY

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 8fefb55..0a5fa0c 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -254,6 +254,10 @@
 	status = "okay";
 };
 
+&usbphy {
+	status = "okay";
+};
+
 &scifa2 {
 	pinctrl-0 = <&scifa2_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

