From b0e8cc91830e802bfbdf52da2830f2993a28d9b8 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Mon, 19 Mar 2018 20:47:42 +0700
Subject: [PATCH 474/628] ARM: DTS: r8a7742: Add power-domains for all CPUs

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index b946df3..6a37b95 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -14,6 +14,7 @@
 #include <dt-bindings/clock/r8a7742-clock.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/power/r8a7742-sysc.h>
 
 / {
 	compatible = "renesas,r8a7742";
@@ -44,6 +45,7 @@
 			clock-frequency = <1400000000>;
 			voltage-tolerance = <1>; /* 1% */
 			clocks = <&cpg_clocks R8A7742_CLK_Z>;
+			power-domains = <&sysc R8A7742_PD_CA15_CPU0>;
 			clock-latency = <300000>; /* 300 us */
 
 			/* kHz - uV - OPPs unknown yet */
@@ -60,6 +62,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <1>;
 			clock-frequency = <1400000000>;
+			power-domains = <&sysc R8A7742_PD_CA15_CPU1>;
 		};
 
 		cpu2: cpu@2 {
@@ -67,6 +70,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <2>;
 			clock-frequency = <1400000000>;
+			power-domains = <&sysc R8A7742_PD_CA15_CPU2>;
 		};
 
 		cpu3: cpu@3 {
@@ -74,6 +78,7 @@
 			compatible = "arm,cortex-a15";
 			reg = <3>;
 			clock-frequency = <1400000000>;
+			power-domains = <&sysc R8A7742_PD_CA15_CPU3>;
 		};
 
 		cpu4: cpu@4 {
@@ -81,6 +86,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x100>;
 			clock-frequency = <780000000>;
+			power-domains = <&sysc R8A7742_PD_CA7_CPU0>;
 		};
 
 		cpu5: cpu@5 {
@@ -88,6 +94,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x101>;
 			clock-frequency = <780000000>;
+			power-domains = <&sysc R8A7742_PD_CA7_CPU1>;
 		};
 
 		cpu6: cpu@6 {
@@ -95,6 +102,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x102>;
 			clock-frequency = <780000000>;
+			power-domains = <&sysc R8A7742_PD_CA7_CPU2>;
 		};
 
 		cpu7: cpu@7 {
@@ -102,6 +110,7 @@
 			compatible = "arm,cortex-a7";
 			reg = <0x103>;
 			clock-frequency = <780000000>;
+			power-domains = <&sysc R8A7742_PD_CA7_CPU3>;
 		};
 	};
 
-- 
2.7.4

