From 868bf397fe684de03ac635d1ab4fd7d0734c7366 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Mon, 9 Apr 2018 10:40:51 +0700
Subject: [PATCH 479/628] ARM: DTS: r8a7742: add DVC support for sound

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 0f3a216..e8309b7 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1350,6 +1350,8 @@
 				<&mstp10_clks R8A7742_CLK_SCU_SRC2>,
 				<&mstp10_clks R8A7742_CLK_SCU_SRC1>,
 				<&mstp10_clks R8A7742_CLK_SCU_SRC0>,
+				<&mstp10_clks R8A7742_CLK_SCU_DVC1>,
+				<&mstp10_clks R8A7742_CLK_SCU_DVC0>,
 				<&audio_clk_a>, <&audio_clk_b>,
 				<&audio_clk_c>, <&m2_clk>;
 		clock-names =	"ssi-all",
@@ -1357,11 +1359,19 @@
 				"ssi.4", "ssi.3", "ssi.2", "ssi.1", "ssi.0",
 				"src.9", "src.8", "src.7", "src.6", "src.5",
 				"src.4", "src.3", "src.2", "src.1", "src.0",
+				"dvc.1", "dvc.0",
 				"clk_a", "clk_b", "clk_c", "clk_i";
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 
 		status = "disabled";
 
+		rcar_sound,dvc {
+			dvc0: dvc@0 {
+			};
+			dvc1: dvc@1 {
+			};
+		};
+
 		rcar_sound,src {
 			src0: src@0 {
 			};
-- 
2.7.4

