From e6b209eb42283ca83421c362a18fa855e5cb1b95 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Mon, 9 Apr 2018 10:54:06 +0700
Subject: [PATCH 483/628] ARM: DTS: r8a7742: Add Audio CTU support

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 07da334..901cc05 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1216,6 +1216,8 @@
 				<&mstp10_clks R8A7742_CLK_SCU_ALL>,
 				<&mstp10_clks R8A7742_CLK_SCU_ALL>,
 				<&mstp10_clks R8A7742_CLK_SCU_ALL>,
+				<&mstp10_clks R8A7742_CLK_SCU_ALL>,
+				<&mstp10_clks R8A7742_CLK_SCU_ALL>,
 				<&mstp10_clks R8A7742_CLK_SCU_ALL>;
 			#clock-cells = <1>;
 			clock-indices = <
@@ -1226,6 +1228,8 @@
 				R8A7742_CLK_SSI3 R8A7742_CLK_SSI2
 				R8A7742_CLK_SSI1 R8A7742_CLK_SSI0
 				R8A7742_CLK_SCU_DVC1 R8A7742_CLK_SCU_DVC0
+				R8A7742_CLK_SCU_CTU1_MIX1
+				R8A7742_CLK_SCU_CTU0_MIX0
 				R8A7742_CLK_SCU_SRC9 R8A7742_CLK_SCU_SRC8
 				R8A7742_CLK_SCU_SRC7 R8A7742_CLK_SCU_SRC6
 				R8A7742_CLK_SCU_SRC5 R8A7742_CLK_SCU_SRC4
@@ -1237,6 +1241,7 @@
 				"ssi9", "ssi8", "ssi7", "ssi6", "ssi5",
 				"ssi4", "ssi3", "ssi2", "ssi1", "ssi0",
 				"dvc1", "dvc0",
+				"ctu1", "ctu0",
 				"src9", "src8", "src7", "src6", "src5",
 				"src4", "src3", "src2", "src1", "src0";
 		};
@@ -1414,6 +1419,8 @@
 				<&mstp10_clks R8A7742_CLK_SCU_SRC0>,
 				<&mstp10_clks R8A7742_CLK_SCU_DVC1>,
 				<&mstp10_clks R8A7742_CLK_SCU_DVC0>,
+				<&mstp10_clks R8A7742_CLK_SCU_CTU1_MIX1>,
+				<&mstp10_clks R8A7742_CLK_SCU_CTU0_MIX0>,
 				<&audio_clk_a>, <&audio_clk_b>,
 				<&audio_clk_c>, <&m2_clk>;
 		clock-names =	"ssi-all",
@@ -1422,11 +1429,23 @@
 				"src.9", "src.8", "src.7", "src.6", "src.5",
 				"src.4", "src.3", "src.2", "src.1", "src.0",
 				"dvc.1", "dvc.0",
+				"ctu.1", "ctu.0",
 				"clk_a", "clk_b", "clk_c", "clk_i";
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 
 		status = "disabled";
 
+		rcar_sound,ctu {
+			ctu00: ctu@0 { };
+			ctu01: ctu@1 { };
+			ctu02: ctu@2 { };
+			ctu03: ctu@3 { };
+			ctu10: ctu@4 { };
+			ctu11: ctu@5 { };
+			ctu12: ctu@6 { };
+			ctu13: ctu@7 { };
+		};
+
 		rcar_sound,dvc {
 			dvc0: dvc@0 {
 			};
-- 
2.7.4

