From 6a76fb147456b1ea3c066f382ca1e35443512ba5 Mon Sep 17 00:00:00 2001
From: Tung Vo <tung.vo.eb@rvc.renesas.com>
Date: Wed, 21 Feb 2018 02:23:43 +0700
Subject: [PATCH 498/628] pinctrl: sh-pfc: r8a7742: add pinctrl for CAN module

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/pinctrl/sh-pfc/pfc-r8a7742.c | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a7742.c b/drivers/pinctrl/sh-pfc/pfc-r8a7742.c
index 9382848..0042c05 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a7742.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a7742.c
@@ -3251,6 +3251,28 @@ static const unsigned int tpu0_to3_pins[] = {
 static const unsigned int tpu0_to3_mux[] = {
 	TPU0TO3_MARK,
 };
+/* - CAN0 -------------------------------------------------------------------- */
+static const unsigned int can0_data_d_pins[] = {
+	/* CAN0 RXD */
+	RCAR_GP_PIN(4, 26),
+	/* CAN0 TXD */
+	RCAR_GP_PIN(4, 18),
+};
+static const unsigned int can0_data_d_mux[] = {
+	CAN0_RX_D_MARK,
+	CAN0_TX_D_MARK,
+};
+/* - CAN1 -------------------------------------------------------------------- */
+static const unsigned int can1_data_b_pins[] = {
+	/* CAN1 TXD */
+	RCAR_GP_PIN(4, 6),
+	/* CAN1 RXD */
+	RCAR_GP_PIN(4, 7),
+};
+static const unsigned int can1_data_b_mux[] = {
+	CAN1_TX_B_MARK,
+	CAN1_RX_B_MARK,
+};
 /* - USB0 ------------------------------------------------------------------- */
 static const unsigned int usb0_pins[] = {
 	/* PWEN, OVC/VBUS */
@@ -3847,6 +3869,8 @@ static const struct sh_pfc_pin_group pinmux_groups[] = {
 	SH_PFC_PIN_GROUP(tpu0_to1),
 	SH_PFC_PIN_GROUP(tpu0_to2),
 	SH_PFC_PIN_GROUP(tpu0_to3),
+	SH_PFC_PIN_GROUP(can0_data_d),
+	SH_PFC_PIN_GROUP(can1_data_b),
 	SH_PFC_PIN_GROUP(usb0),
 	SH_PFC_PIN_GROUP(usb0_ovc_vbus),
 	SH_PFC_PIN_GROUP(usb1),
@@ -4277,6 +4301,14 @@ static const char * const tpu0_groups[] = {
 	"tpu0_to3",
 };
 
+static const char * const can0_groups[] = {
+	"can0_data_d",
+};
+
+static const char * const can1_groups[] = {
+	"can1_data_b",
+};
+
 static const char * const usb0_groups[] = {
 	"usb0",
 	"usb0_ovc_vbus",
@@ -4388,6 +4420,8 @@ static const struct sh_pfc_function pinmux_functions[] = {
 	SH_PFC_FUNCTION(sdhi3),
 	SH_PFC_FUNCTION(ssi),
 	SH_PFC_FUNCTION(tpu0),
+	SH_PFC_FUNCTION(can0),
+	SH_PFC_FUNCTION(can1),
 	SH_PFC_FUNCTION(usb0),
 	SH_PFC_FUNCTION(usb1),
 	SH_PFC_FUNCTION(usb2),
-- 
2.7.4

