From 9b4d5dbda0c86bb8585ee147657de0cd2b385e48 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Tue, 17 Apr 2018 09:53:48 +0700
Subject: [PATCH 514/628] ARM: DTS: r8a7742: Enable UHS SDR50 for sdhi2

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 798412b..9bf124d 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -436,6 +436,9 @@
 		dmas = <&dmac1 0xc1>, <&dmac1 0xc2>;
 		dma-names = "tx", "rx";
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,clk-rate = <97500000>;
+		renesas,mmc-scc-tapnum = <8>;
+		sd-uhs-sdr50;
 		status = "disabled";
 	};
 
-- 
2.7.4

