From 5b008b61ef0e5579393c39cda3218a5e8e912a9e Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Fri, 25 May 2018 14:11:37 +0700
Subject: [PATCH 538/628] phy: phy-rcar-gen2: add support for r8a7742 usb otg

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: LiNTC <lintc@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/phy/phy-rcar-gen2.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index 150eba1..b8d7cc6 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -357,7 +357,8 @@ static void gpio_id_work(struct work_struct *work)
 	if (!id) {
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743") ||
+		else if (of_machine_is_compatible("renesas,r8a7742") ||
+			of_machine_is_compatible("renesas,r8a7743") ||
 			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -365,7 +366,8 @@ static void gpio_id_work(struct work_struct *work)
 	} else {
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743") ||
+		else if (of_machine_is_compatible("renesas,r8a7742") ||
+			of_machine_is_compatible("renesas,r8a7743") ||
 			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -397,7 +399,8 @@ static void gpio_vbus_work(struct work_struct *work)
 	if (id) {
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743") ||
+		else if (of_machine_is_compatible("renesas,r8a7742") ||
+			of_machine_is_compatible("renesas,r8a7743") ||
 			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -405,7 +408,8 @@ static void gpio_vbus_work(struct work_struct *work)
 	} else {
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743") ||
+		else if (of_machine_is_compatible("renesas,r8a7742") ||
+			of_machine_is_compatible("renesas,r8a7743") ||
 			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -518,7 +522,8 @@ static int rcar_gen2_usb_set_peripheral(struct usb_otg *otg,
 		otg->gadget = gadget;
 
 		/* initialize connection state */
-		if (of_machine_is_compatible("renesas,r8a7743")
+		if (of_machine_is_compatible("renesas,r8a7742")
+			|| of_machine_is_compatible("renesas,r8a7743")
 			|| of_machine_is_compatible("renesas,r8a7744")
 			|| of_machine_is_compatible("renesas,r8a7745")) {
 			gpio_vbus_irq(channel->irq_vbus, channel);
-- 
2.7.4

