From 9fd1e7bebd4ef617efd3cb074dcf1f91cf78154e Mon Sep 17 00:00:00 2001
From: khatran <kha.tran.px@rvc.renesas.com>
Date: Mon, 23 Apr 2018 16:33:49 +0700
Subject: [PATCH 540/628] ARM: DTS: r8a7742: add description for cmt0 and cmt1

Signed-off-by: khatran <kha.tran.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 8346ba1..7abd3ec 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -401,6 +401,40 @@
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 	};
 
+	cmt0: timer@ffca0000 {
+		compatible = "renesas,cmt-48-gen2";
+		reg = <0 0xffca0000 0 0x1004>;
+		interrupts = <0 142 IRQ_TYPE_LEVEL_HIGH>,
+				<0 143 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_CMT0>;
+		clock-names = "fck";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+
+		renesas,channels-mask = <0x60>;
+
+		status = "disabled";
+	};
+
+	cmt1: timer@e6130000 {
+		compatible = "renesas,cmt-48-gen2";
+		reg = <0 0xe6130000 0 0x1004>;
+		interrupts = <0 120 IRQ_TYPE_LEVEL_HIGH>,
+				<0 121 IRQ_TYPE_LEVEL_HIGH>,
+				<0 122 IRQ_TYPE_LEVEL_HIGH>,
+				<0 123 IRQ_TYPE_LEVEL_HIGH>,
+				<0 124 IRQ_TYPE_LEVEL_HIGH>,
+				<0 125 IRQ_TYPE_LEVEL_HIGH>,
+				<0 126 IRQ_TYPE_LEVEL_HIGH>,
+				<0 127 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp3_clks R8A7742_CLK_CMT1>;
+		clock-names = "fck";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+
+		renesas,channels-mask = <0xff>;
+
+		status = "disabled";
+	};
+
 	pfc: pfc@e6060000 {
 		compatible = "renesas,pfc-r8a7742";
 		reg = <0 0xe6060000 0 0x250>;
-- 
2.7.4

