From f9ea7f048c7c3fc9674e6bf6747c72a34220e369 Mon Sep 17 00:00:00 2001
From: Kha Tran <kha.tran.px@rvc.renesas.com>
Date: Wed, 4 Apr 2018 18:00:40 +0700
Subject: [PATCH 551/628] ARM: DTS: r8a7744: add description for tmu2 and tmu3

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7744.dtsi | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 33ad1de..38e9048 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -268,6 +268,37 @@
 		status = "disabled";
 	};
 
+	tmu2: timer@fff70000 {
+		compatible = "renesas,tmu-r8a7744", "renesas,tmu";
+		reg = <0 0xfff70000 0 0x30>;
+		interrupts = <0 303 IRQ_TYPE_LEVEL_HIGH>,
+			<0 304 IRQ_TYPE_LEVEL_HIGH>,
+			<0 305 IRQ_TYPE_LEVEL_HIGH>,
+			<0 306 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7744_CLK_TMU2>;
+		clock-names = "fck";
+		power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+
+		#renesas,channels = <3>;
+
+		status = "disabled";
+	};
+
+	tmu3: timer@fff80000 {
+		compatible = "renesas,tmu-r8a7744", "renesas,tmu";
+		reg = <0 0xfff80000 0 0x30>;
+		interrupts = <0 131 IRQ_TYPE_LEVEL_HIGH>,
+			<0 132 IRQ_TYPE_LEVEL_HIGH>,
+			<0 133 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7744_CLK_TMU3>;
+		clock-names = "fck";
+		power-domains = <&sysc R8A7744_PD_ALWAYS_ON>;
+
+		#renesas,channels = <3>;
+
+		status = "disabled";
+	};
+
 	irqc0: interrupt-controller@e61c0000 {
 		compatible = "renesas,irqc-r8a7744", "renesas,irqc";
 		#interrupt-cells = <2>;
-- 
2.7.4

