From c4ee465ac66b45aa9e886f04b32111fd599a04f2 Mon Sep 17 00:00:00 2001
From: ThaoHM2 <thaohm2@fsoft.com.vn>
Date: Fri, 1 Jun 2018 15:42:17 +0700
Subject: [PATCH 571/628] ARM DST iwg21m enable pwm

Signed-off-by: ThaoHM2 <thaohm2@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 60 ++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 951555a..eeade27 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -316,6 +316,31 @@
 		renesas,function = "tpu0";
 	};
 
+	pwm0_pins: pwm0 {
+		renesas,groups = "pwm0";
+		renesas,function = "pwm0";
+	};
+
+	pwm3_pins: pwm3 {
+		renesas,groups = "pwm3";
+		renesas,function = "pwm3";
+	};
+
+	pwm4_pins: pwm4 {
+		renesas,groups = "pwm4";
+		renesas,function = "pwm4";
+	};
+
+	pwm5_pins: pwm5 {
+		renesas,groups = "pwm5";
+		renesas,function = "pwm5";
+	};
+
+	pwm6_pins: pwm6 {
+		renesas,groups = "pwm6";
+		renesas,function = "pwm6";
+	};
+
 	mmc1_4_pins: mmc1_4 {
 		renesas,groups = "mmc1_data4", "mmc1_ctrl";
 		renesas,function = "mmc1";
@@ -796,6 +821,41 @@
 	};
 };
 
+&pwm0 {
+	pinctrl-0 = <&pwm0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&pwm3 {
+	pinctrl-0 = <&pwm3_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&pwm4 {
+	pinctrl-0 = <&pwm4_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&pwm5 {
+	pinctrl-0 = <&pwm5_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&pwm6 {
+	pinctrl-0 = <&pwm6_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
 &cmt0{
 	status = "okay";
 };
-- 
2.7.4

