From cc6967cd2a3c6ad7dde90a13ae65cadf5e96fca0 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Tue, 29 Aug 2017 18:07:51 +0700
Subject: [PATCH] vsp: vsp_drv_par: correct return error code for hgt_param

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 drv/vsp/vsp_drv_par.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drv/vsp/vsp_drv_par.c b/drv/vsp/vsp_drv_par.c
index ae5de14..46856fe 100755
--- a/drv/vsp/vsp_drv_par.c
+++ b/drv/vsp/vsp_drv_par.c
@@ -2228,13 +2228,13 @@ static long vsp_ins_check_hgt_param(struct vsp_ch_info *ch_info,
 	if ((hgt_param->x_skip != VSP_SKIP_OFF) &&
 		(hgt_param->x_skip != VSP_SKIP_1_2) &&
 		(hgt_param->x_skip != VSP_SKIP_1_4)) {
-		return E_VSP_PARA_HGO_XSKIP;
+		return E_VSP_PARA_HGT_XSKIP;
 	}
 
 	if ((hgt_param->y_skip != VSP_SKIP_OFF) &&
 		(hgt_param->y_skip != VSP_SKIP_1_2) &&
 		(hgt_param->y_skip != VSP_SKIP_1_4)) {
-		return E_VSP_PARA_HGO_YSKIP;
+		return E_VSP_PARA_HGT_YSKIP;
 	}
 
 	/* check sampling point */
-- 
1.9.1

