From cb7f739c934b1b3281a380afb6eb98650c0b69d5 Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Tue, 26 Dec 2017 16:32:46 +0700
Subject: [PATCH 003/433] ARM: dts: Switch of usbhs ip block

Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7743-skrzg1m.dts | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7743-skrzg1m.dts b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
index ec33b6d..f84e426 100644
--- a/arch/arm/boot/dts/r8a7743-skrzg1m.dts
+++ b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
@@ -421,8 +421,6 @@
 &hsusb {
 	pinctrl-0 = <&usb0_pins>;
 	pinctrl-names = "default";
-
-	status = "okay";
 };
 
 &usbphy {
-- 
2.7.4

