From 069bf7005b89f987b9e768a16c9ede5ab3fa48ee Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Fri, 22 Jul 2016 01:57:40 +0300
Subject: [PATCH 005/433] ARM: dts: Skrzg1e enable PFC DU

Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7745-skrzg1e.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7745-skrzg1e.dts b/arch/arm/boot/dts/r8a7745-skrzg1e.dts
index 871892c..69acdf0 100644
--- a/arch/arm/boot/dts/r8a7745-skrzg1e.dts
+++ b/arch/arm/boot/dts/r8a7745-skrzg1e.dts
@@ -245,6 +245,15 @@
 		renesas,groups = "audio_clkc";
 		renesas,function = "audio_clk";
 	};
+	du0_pins: du0 {
+		renesas,groups = "du0_rgb888", "du0_sync", "du0_disp", "du0_clk0_out";
+		renesas,function = "du0";
+	};
+
+	du1_pins: du1 {
+		renesas,groups = "du1_rgb666", "du1_sync", "du1_disp", "du1_clk0_out";
+		renesas,function = "du1";
+	};
 };
 
 &scif2 {
@@ -419,6 +428,8 @@
 };
 
 &du {
+	pinctrl-0 = <&du0_pins &du1_pins>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	clocks = <&mstp7_clks R8A7745_CLK_DU0>,
-- 
2.7.4

