From 8482d1ad0b94f2d1cd27624a14109c2dca57ee4e Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Fri, 5 Aug 2016 12:20:55 +0300
Subject: [PATCH 006/433] ARM: dts: Add gpio-keys to skrzg1m

Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7743-skrzg1m.dts | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743-skrzg1m.dts b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
index f84e426..0154903 100644
--- a/arch/arm/boot/dts/r8a7743-skrzg1m.dts
+++ b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
@@ -46,6 +46,32 @@
 		reg = <2 0x00000000 0 0x40000000>;
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+		autorepeat;
+		key@0 {
+					gpios = <&gpio7 0 GPIO_ACTIVE_LOW>;
+					linux,code = <2>;
+					label = "SW30";
+					gpio-key,wakeup;
+					debounce-interval = <20>;
+			};
+		key@2{
+					gpios = <&gpio7 1 GPIO_ACTIVE_LOW>;
+					linux,code = <3>;
+					label = "SW31";
+					gpio-key,wakeup;
+					debounce-interval = <20>;
+			};
+		key@3 {
+					gpios = <&gpio7 2 GPIO_ACTIVE_LOW>;
+					linux,code = <4>;
+					label = "SW32";
+					gpio-key,wakeup;
+					debounce-interval = <20>;
+			};
+	};
+
 	vcc_sdhi0: regulator@0 {
 		compatible = "regulator-fixed";
 
-- 
2.7.4

