From e8a9a51ef5d6ef26dadd7b8871c1936789a6d8da Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Tue, 13 Oct 2020 09:55:37 +0700
Subject: [PATCH 069/433] ARM: dts: r8a7742: Bind GPIO[0-5] to SYSC PM domains

Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 23b1443..915fd63 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -665,7 +665,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		gpio1: gpio@e6051000 {
@@ -679,7 +679,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO1>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		gpio2: gpio@e6052000 {
@@ -693,7 +693,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO2>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		gpio3: gpio@e6053000 {
@@ -707,7 +707,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO3>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		gpio4: gpio@e6054000 {
@@ -721,7 +721,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO4>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		gpio5: gpio@e6055000 {
@@ -735,7 +735,7 @@
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A7742_CLK_GPIO5>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		};
 
 		pfc: pin-controller@e6060000 {
-- 
2.7.4

