From 8f108a5d9f041d37d76e0e73e3ee363837222e98 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Thu, 27 Sep 2018 14:29:14 +0700
Subject: [PATCH 082/433] mmc: renesas-sdhi: Support configure clock

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 drivers/mmc/host/renesas_sdhi_core.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/mmc/host/renesas_sdhi_core.c b/drivers/mmc/host/renesas_sdhi_core.c
index 8ed05a6..aa5cad2 100644
--- a/drivers/mmc/host/renesas_sdhi_core.c
+++ b/drivers/mmc/host/renesas_sdhi_core.c
@@ -453,12 +453,14 @@ int renesas_sdhi_probe(struct platform_device *pdev,
 		       const struct tmio_mmc_dma_ops *dma_ops)
 {
 	struct tmio_mmc_data *mmd = pdev->dev.platform_data;
+	const struct device_node *np = pdev->dev.of_node;
 	const struct renesas_sdhi_of_data *of_data;
 	struct tmio_mmc_data *mmc_data;
 	struct tmio_mmc_dma *dma_priv;
 	struct tmio_mmc_host *host;
 	struct renesas_sdhi *priv;
 	struct resource *res;
+	int clk_rate = 0;
 	int irq, ret, i;
 
 	of_data = of_device_get_match_data(&pdev->dev);
@@ -490,6 +492,18 @@ int renesas_sdhi_probe(struct platform_device *pdev,
 						"state_uhs");
 	}
 
+	if (np && !of_property_read_u32(np, "renesas,clk-rate", &clk_rate)) {
+		if (clk_rate) {
+			clk_prepare_enable(priv->clk);
+			ret = clk_set_rate(priv->clk, clk_rate);
+			if (ret < 0)
+				dev_err(&pdev->dev,
+					"cannot set clock rate: %d\n", ret);
+
+			clk_disable_unprepare(priv->clk);
+		}
+	}
+
 	host = tmio_mmc_host_alloc(pdev);
 	if (!host) {
 		ret = -ENOMEM;
-- 
2.7.4

