From 0b3890e914f321cdc93ad61f7d1e0f2e9466c423 Mon Sep 17 00:00:00 2001
From: Viet Nguyen <vietn@fsoft.com.vn>
Date: Fri, 6 Jul 2018 18:05:27 +0700
Subject: [PATCH 101/433] ASoC: sgtl5000: correct power supply setting

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 sound/soc/codecs/sgtl5000.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/sgtl5000.h b/sound/soc/codecs/sgtl5000.h
index 22f3442..5930476 100644
--- a/sound/soc/codecs/sgtl5000.h
+++ b/sound/soc/codecs/sgtl5000.h
@@ -326,7 +326,7 @@
 /*
  * SGTL5000_CHIP_ANA_POWER
  */
-#define SGTL5000_ANA_POWER_DEFAULT		0x7060
+#define SGTL5000_ANA_POWER_DEFAULT		0x7260
 #define SGTL5000_DAC_STEREO			0x4000
 #define SGTL5000_LINREG_SIMPLE_POWERUP		0x2000
 #define SGTL5000_STARTUP_POWERUP		0x1000
-- 
2.7.4

