From 4d1129e3d333115e0d5cc6195523802bb12baf59 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Tue, 10 Apr 2018 10:07:07 +0700
Subject: [PATCH 121/433] spi: rspi: fix compile warning

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 drivers/spi/spi-rspi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-rspi.c b/drivers/spi/spi-rspi.c
index 8ebd2e2..2551ed5 100644
--- a/drivers/spi/spi-rspi.c
+++ b/drivers/spi/spi-rspi.c
@@ -186,7 +186,7 @@
 #define SPBFCR_RXTRG_8B		0x04	/*   8 bytes (24 bytes available) */
 #define SPBFCR_RXTRG_32B	0x07	/*   32 bytes (0 byte available) */
 
-#define SPI_BUFFER_SIZE		32u	/* transmit and receive buffer size *
+#define SPI_BUFFER_SIZE		32u	/* transmit and receive buffer size */
 /* QSPI on R-Car Gen2 */
 #define SPBFCR_TXTRG_1B		0x00	/* 31 bytes (1 byte available) */
 #define SPBFCR_TXTRG_32B	0x30	/* 0 byte (32 bytes available) */
-- 
2.7.4

