From 5b749258f730f811a7bc2c2dcd701b4e815d6ff2 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Tue, 30 Aug 2016 18:19:57 +0300
Subject: [PATCH 188/433] media: adv7180: set crop bounds

Signed-off-by: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: hao.bui.yg <hao.bui.yg@renesas.com>
---
 drivers/media/i2c/adv7180.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/media/i2c/adv7180.c b/drivers/media/i2c/adv7180.c
index 68e5bc0..aaa1ece 100644
--- a/drivers/media/i2c/adv7180.c
+++ b/drivers/media/i2c/adv7180.c
@@ -728,6 +728,13 @@ static int adv7180_g_mbus_config(struct v4l2_subdev *sd,
 static int adv7180_cropcap(struct v4l2_subdev *sd, struct v4l2_cropcap *cropcap)
 {
 	struct adv7180_state *state = to_state(sd);
+	cropcap->bounds.left = 0;
+	cropcap->bounds.top = 0;
+	/* set maximum window size */
+	cropcap->bounds.width = 720;
+	cropcap->bounds.height = state->curr_norm & V4L2_STD_525_60 ? 480 : 576;
+	cropcap->defrect = cropcap->bounds;
+	cropcap->type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 
 	if (state->curr_norm & V4L2_STD_525_60) {
 		cropcap->pixelaspect.numerator = 11;
-- 
2.7.4

