From 84ef5ffc873932a961b8d0385d20e2e9853929bc Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Tue, 26 Dec 2017 16:12:46 +0700
Subject: [PATCH 228/433] media: videodev2.h: Add format for mem2mem playback

Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 include/uapi/linux/videodev2.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index 421d274..dc7a598 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -488,6 +488,9 @@ struct v4l2_pix_format {
 #define V4L2_PIX_FMT_ARGB32  v4l2_fourcc('B', 'A', '2', '4') /* 32  ARGB-8-8-8-8  */
 #define V4L2_PIX_FMT_XRGB32  v4l2_fourcc('B', 'X', '2', '4') /* 32  XRGB-8-8-8-8  */
 
+/* RGB formats for memory output */
+#define V4L2_PIX_FMT_RGB32S  v4l2_fourcc('R', 'G', '4', 'S') /* 32 RGB-8-8-8-8 */
+
 /* Grey formats */
 #define V4L2_PIX_FMT_GREY    v4l2_fourcc('G', 'R', 'E', 'Y') /*  8  Greyscale     */
 #define V4L2_PIX_FMT_Y4      v4l2_fourcc('Y', '0', '4', ' ') /*  4  Greyscale     */
-- 
2.7.4

