From c357e960b845a9f5deea104dd266eea6a7c3224e Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 22 Jul 2019 09:20:23 +0700
Subject: [PATCH 235/433] ARM: dts: r8a77470: Enable DMA for serial ports

Signed-off-by: tungvo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index a104b8f..104414d 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -814,6 +814,8 @@
 			interrupts = <0 154 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF0>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x39>, <&dmac0 0x3a>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
@@ -824,6 +826,8 @@
 			interrupts = <0 155 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF1>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x4d>, <&dmac0 0x4e>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
@@ -834,6 +838,8 @@
 			interrupts = <0 21 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp7_clks R8A77470_CLK_HSCIF2>;
 			clock-names = "sci_ick";
+			dmas = <&dmac0 0x3b>, <&dmac0 0x3c>;
+			dma-names = "tx", "rx";
 			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			status = "disabled";
 		};
-- 
2.7.4

