From b6e7c3d0301d359c7bf816d739e6eba0cbad7a9b Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 28 Jul 2020 10:02:40 +0700
Subject: [PATCH 280/433] pwm: pwm-renesas-tpu: Add support for R8A7744 Soc

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/pwm/pwm-renesas-tpu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pwm/pwm-renesas-tpu.c b/drivers/pwm/pwm-renesas-tpu.c
index a5a542a..fafdb4b 100644
--- a/drivers/pwm/pwm-renesas-tpu.c
+++ b/drivers/pwm/pwm-renesas-tpu.c
@@ -453,6 +453,7 @@ static int tpu_remove(struct platform_device *pdev)
 static const struct of_device_id tpu_of_table[] = {
 	{ .compatible = "renesas,tpu-r8a73a4", },
 	{ .compatible = "renesas,tpu-r8a7740", },
+	{ .compatible = "renesas,tpu-r8a7744", },
 	{ .compatible = "renesas,tpu-r8a7745", },
 	{ .compatible = "renesas,tpu-r8a7790", },
 	{ .compatible = "renesas,tpu-sh7372", },
-- 
2.7.4

