From 4bd43b4bd4d072017e7cd7fcfe79ec1c996c3eca Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 4 Aug 2020 16:28:46 +0700
Subject: [PATCH 355/433] ARM: dts: r8a7745: Use sysc power domain for VSP

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 8f00c41..40c96b6 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1423,7 +1423,7 @@
 			reg = <0 0xfe928000 0 0x8000>;
 			interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1_SY>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		vspm@fe930000 {
@@ -1431,7 +1431,7 @@
 			reg = <0 0xfe930000 0 0x8000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		fdp0: fdpm@fe940000 {
@@ -1447,7 +1447,7 @@
 			reg = <0 0xfe930000 0 0x7404>;
 			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		du: display@feb00000 {
-- 
2.7.4

