diff --git a/arch/arm64/boot/dts/renesas/r9a09g011gbg-evaluation-board.dts b/arch/arm64/boot/dts/renesas/r9a09g011gbg-evaluation-board.dts
index 9022e45..6bfcc2e 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g011gbg-evaluation-board.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g011gbg-evaluation-board.dts
@@ -75,15 +75,6 @@
 
     };
 
-    udmabuf@0 {
-        compatible = "ikwzm,udmabuf-0.10.a";
-        device-name = "udmabuf0";
-        minor-number = <0>;
-        size = <0x400000>; // 4MiB
-        dma-coherent;
-	memory-region = <&image_buf0>;
-    };
-
     vcc_sdhi0: regulator-vcc-sdhi0 {
         compatible = "regulator-fixed";
 
