diff --git a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
index 7b3b17d..bfa771e 100755
--- a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg-evaluation-board.dts
@@ -84,18 +84,16 @@
     };
 
     vccq_sdhi0: regulator-vccq-sdhi0 {
-        compatible = "regulator-gpio";
+        compatible = "regulator-fixed";
 
         regulator-name = "SDHI0 VccQ";
         regulator-min-microvolt = <1800000>;
-        regulator-max-microvolt = <3300000>;
-
-        gpios-states = <1>;
-        states = <3300000 1
-              1800000 0>;
+        regulator-max-microvolt = <1800000>;
+        regulator-always-on;
+        regulator-boot-on;
     };
-
-        reg_1p8v: regulator0 {
+    
+    reg_1p8v: regulator0 {
         compatible = "regulator-fixed";
         regulator-name = "fixed-1.8V";
         regulator-min-microvolt = <1800000>;
diff --git a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg.dtsi b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg.dtsi
index 3656bc4..8f73d0f 100755
--- a/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g055ma3gbg.dtsi
@@ -459,6 +459,7 @@
             resets = <&cpg 300>;
             max-frequency = <200000000>;
             vmmc-supply = <&vcc_sdhi0>;
+            vqmmc-supply = <&vccq_sdhi0>;
             cd-gpios = <&gpio8 07 GPIO_ACTIVE_LOW>;
             bus-width = <4>;
             sd-uhs-sdr50;
