diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 11f834f..e66e0b5 100755
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -1185,6 +1185,7 @@ static void clk_core_disable_unprepare(struct clk_core *core)
 	clk_core_unprepare_lock(core);
 }
 
+#ifndef CONFIG_CLK_R9A09G055MA3GBG
 static void __init clk_unprepare_unused_subtree(struct clk_core *core)
 {
 	struct clk_core *child;
@@ -1260,6 +1261,7 @@ unprepare_out:
 	if (core->flags & CLK_OPS_PARENT_ENABLE)
 		clk_core_disable_unprepare(core->parent);
 }
+#endif
 
 static bool clk_ignore_unused __initdata;
 static int __init clk_ignore_unused_setup(char *__unused)
@@ -1269,6 +1271,7 @@ static int __init clk_ignore_unused_setup(char *__unused)
 }
 __setup("clk_ignore_unused", clk_ignore_unused_setup);
 
+#ifndef CONFIG_CLK_R9A09G055MA3GBG
 static int __init clk_disable_unused(void)
 {
 	struct clk_core *core;
@@ -1296,7 +1299,7 @@ static int __init clk_disable_unused(void)
 
 	return 0;
 }
-#ifndef CONFIG_CLK_R9A09G055MA3GBG
+
 late_initcall_sync(clk_disable_unused);
 #endif
 
