From 6d91f8795480e1367e7bc1b6a6a26f8df69c957c Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Tue, 13 Mar 2018 18:22:03 +0700
Subject: [PATCH 022/504] ARM: dts iwg21m: enable scifa2 as serial debug port

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 779a705..6fdec12 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -19,6 +19,10 @@
 	model = "iwg21m";
 	compatible = "iw,iwg21m", "renesas,r8a7742";
 
+	aliases {
+		serial2 = &scifa2;
+	};
+
 	memory@40000000 {
 		device_type = "memory";
 		reg = <0 0x40000000 0 0x40000000>;
@@ -38,3 +42,16 @@
 &extal_clk {
 	clock-frequency = <20000000>;
 };
+
+&pfc {
+	scifa2_pins: serial2 {
+		groups = "scifa2_data_c";
+		function = "scifa2";
+	};
+};
+
+&scifa2 {
+	pinctrl-0 = <&scifa2_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
-- 
2.7.4

