From 977f81dbf7fd7b383ec3e8cf8e69dde49b16b105 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Sun, 24 Dec 2017 04:35:49 +0700
Subject: [PATCH 107/504] net: ethernet: ravb: add support r8a7745 SoC.

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/net/ethernet/renesas/ravb_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index 02c253f..490229a 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -1694,6 +1694,7 @@ static const struct of_device_id ravb_match_table[] = {
 	{ .compatible = "renesas,etheravb-r8a7795", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-rcar-gen3", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-r8a7743", .data = (void *)RCAR_GEN2 },
+	{ .compatible = "renesas,etheravb-r8a7745", .data = (void *)RCAR_GEN2 },
 	{ }
 };
 MODULE_DEVICE_TABLE(of, ravb_match_table);
-- 
2.7.4

