From a8d0edcbe43b30e5c5af31d5189986bf1597ed08 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 22 Jul 2019 14:16:11 +0700
Subject: [PATCH 110/504] ARM: dts r8a77470: add Gigabit Ethernet description

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index becb30cf..28bb6bc 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -494,10 +494,11 @@
 		avb: ethernet@e6800000 {
 			compatible = "renesas,etheravb-r8a77470",
 				     "renesas,etheravb-rcar-gen2";
-			reg = <0 0xe6800000 0 0x800>, <0 0xee0e8000 0 0x4000>;
+			reg = <0 0xe6800000 0 0x7bf>;
 			interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp8_clks R8A77470_CLK_ETHERAVB>;
-			power-domains = <&cpg_clocks>;
+			phy-mode = "gmii";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			status = "disabled";
-- 
2.7.4

