From ac7b6c240a70cef29bf128037ba9052d315d486c Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 9 Apr 2018 18:02:51 +0700
Subject: [PATCH 111/504] net: ethernet: ravb: add support r8a77470 SoC.

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/net/ethernet/renesas/ravb_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index 490229a..a2cb33d 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -1695,6 +1695,7 @@ static const struct of_device_id ravb_match_table[] = {
 	{ .compatible = "renesas,etheravb-rcar-gen3", .data = (void *)RCAR_GEN3 },
 	{ .compatible = "renesas,etheravb-r8a7743", .data = (void *)RCAR_GEN2 },
 	{ .compatible = "renesas,etheravb-r8a7745", .data = (void *)RCAR_GEN2 },
+	{ .compatible = "renesas,etheravb-r8a77470", .data = (void *)RCAR_GEN2 },
 	{ }
 };
 MODULE_DEVICE_TABLE(of, ravb_match_table);
-- 
2.7.4

