From 3db637a7fd56c2a14743d35e28bb5de871af19f2 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Fri, 17 Jul 2020 09:14:22 +0700
Subject: [PATCH 131/504] ARM: dts r8a7742: add pcie to dts

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index ce54564..bbd56b5 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -22,6 +22,8 @@
 	aliases {
 		eth = &avb;
 		eth1 = &ether;
+		pciec   = &pciec;
+		pcie_clk = &pcie_bus_clk;
 		serial2 = &scifa2;
 	};
 
@@ -107,6 +109,14 @@
 	};
 };
 
+&pcie_bus_clk {
+	status = "okay";
+};
+
+&pciec {
+	status = "okay";
+};
+
 &pfc {
 	avb_pins: avb {
 		groups = "avb_mdio", "avb_gmii", "avb_phy_int";
-- 
2.7.4

