From ca0429cf3111ef354654ae779a1c6416d2de77b0 Mon Sep 17 00:00:00 2001
From: Tung Vo <tung.vo.eb@rvc.renesas.com>
Date: Mon, 20 Jul 2020 15:59:42 +0700
Subject: [PATCH 206/504] ARM: dts iwg21m: add power-supply support fan

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 3d72720..fb927c4 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -57,6 +57,7 @@
 		pwms = <&tpu 1 5000000 GPIO_ACTIVE_HIGH>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <7>;
+		power-supply = <&vcc_fan>;
 		pinctrl-0 = <&fan_pins>;
 		pinctrl-names = "default";
 	};
@@ -77,6 +78,14 @@
 		regulator-always-on;
 	};
 
+	vcc_fan: regulator@6 {
+		compatible = "regulator-fixed";
+		regulator-name = "fan vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
 	reg_3p3v: 3p3v {
 		compatible = "regulator-fixed";
 		regulator-name = "3P3V";
-- 
2.7.4

