From ee985a822c42477db376f9e2a904c058f1df6ae9 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Tue, 21 Jul 2020 08:38:25 +0700
Subject: [PATCH 224/504] ARM: dts iwg21m: enable Serial module

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 62 ++++++++++++++++++++++++++++++++++++
 1 file changed, 62 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index fb927c4..cd81a4c 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -24,7 +24,12 @@
 		eth1 = &ether;
 		pciec   = &pciec;
 		pcie_clk = &pcie_bus_clk;
+		serial0 = &scif0;
+		serial1 = &scif1;
 		serial2 = &scifa2;
+		serial3 = &scifb1;
+		serial4 = &scifb2;
+		serial5 = &hscif0;
 	};
 
 	memory@40000000 {
@@ -189,6 +194,13 @@
 	};
 };
 
+&hscif0 {
+	pinctrl-0 = <&hscif0_pins>;
+	pinctrl-names = "default";
+	ctsrts;
+	status = "okay";
+};
+
 &i2c2	{
 	status = "okay";
 	clock-frequency = <400000>;
@@ -255,6 +267,11 @@
 		function = "tpu0";
 	};
 
+	hscif0_pins: serial5 {
+		groups = "hscif0_data", "hscif0_ctrl";
+		function = "hscif0";
+	};
+
 	mmc1_8_pins: mmc1_8 {
 		groups = "mmc1_data8", "mmc1_ctrl";
 		function = "mmc1";
@@ -285,11 +302,31 @@
 		function = "pwm6";
 	};
 
+	scif0_pins: serial0 {
+		groups = "scif0_data";
+		function = "scif0";
+	};
+
+	scif1_pins: serial1 {
+		groups = "scif1_data";
+		function = "scif1";
+	};
+
 	scifa2_pins: serial2 {
 		groups = "scifa2_data_c";
 		function = "scifa2";
 	};
 
+	scifb1_pins: serial3 {
+		groups = "scifb1_data";
+		function = "scifb1";
+	};
+
+	scifb2_pins: serial4 {
+		groups = "scifb2_data", "scifb2_ctrl";
+		function = "scifb2";
+	};
+
 	sdhi2_pins: sd2 {
 		groups = "sdhi2_data4", "sdhi2_ctrl", "sdhi2_cd";
 		function = "sdhi2";
@@ -370,12 +407,37 @@
 	};
 };
 
+&scif0 {
+	pinctrl-0 = <&scif0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&scif1 {
+	pinctrl-0 = <&scif1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
 &scifa2 {
 	pinctrl-0 = <&scifa2_pins>;
 	pinctrl-names = "default";
 	status = "okay";
 };
 
+&scifb1 {
+	pinctrl-0 = <&scifb1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&scifb2 {
+	pinctrl-0 = <&scifb2_pins>;
+	pinctrl-names = "default";
+	ctsrts;
+	status = "okay";
+};
+
 &sdhi2 {
 	pinctrl-0 = <&sdhi2_pins>;
 	pinctrl-1 = <&sdhi2_pins_uhs>;
-- 
2.7.4

