From 9cb0e5bee71ac583ac33d589af5fb00862c07340 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Wed, 22 Jul 2020 10:08:41 +0700
Subject: [PATCH 242/504] ARM: dts: r8a7742: add SYSC power-domains for
 I2C[0-3]

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index abe39ff..c3850dc 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1216,6 +1216,7 @@
 		reg = <0 0xe6508000 0 0x40>;
 		interrupts = <0 287 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp9_clks R8A7742_CLK_I2C0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		status = "disabled";
 	};
 
@@ -1227,6 +1228,7 @@
 		reg = <0 0xe6518000 0 0x40>;
 		interrupts = <0 288 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp9_clks R8A7742_CLK_I2C1>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		status = "disabled";
 	};
 
@@ -1238,6 +1240,7 @@
 		reg = <0 0xe6530000 0 0x40>;
 		interrupts = <0 286 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp9_clks R8A7742_CLK_I2C2>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		status = "disabled";
 	};
 
@@ -1249,6 +1252,7 @@
 		reg = <0 0xe6540000 0 0x40>;
 		interrupts = <0 290 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp9_clks R8A7742_CLK_I2C3>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 		status = "disabled";
 	};
 
-- 
2.7.4

