From 33f1edb741ec3b1a15ecfb8c3cbb31d2a4c5b497 Mon Sep 17 00:00:00 2001
From: Kha Tran <kha.tran.px@rvc.renesas.com>
Date: Thu, 25 Jan 2018 10:37:00 +0700
Subject: [PATCH 317/504] ARM: dts r8a7744: add aliases for VIN[0,1,2]

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7744.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 22d5ed1..9a97612 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -29,6 +29,9 @@
 		spi1 = &msiof0;
 		spi2 = &msiof1;
 		spi3 = &msiof2;
+		vin0 = &vin0;
+		vin1 = &vin1;
+		vin2 = &vin2;
 	};
 
 	/*
-- 
2.7.4

