From ab4a0bedd688c66fe1c71d34cb271a77b45f2940 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 27 Jul 2020 16:54:35 +0700
Subject: [PATCH 326/504] ARM: dts: iwg21m: Enable YCbCr 8bit mode for vin2

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 355eb1c..fa442ba 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -674,6 +674,7 @@
 		vin2ep: endpoint {
 			remote-endpoint = <&ov7725_2>;
 			bus-width = <8>;
+			ycbcr_8b_g_enable;
 		};
 	};
 };
-- 
2.7.4

