From 911cea6578f28fff6397f6f470ae105728dbac9d Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 28 Jul 2020 11:09:35 +0700
Subject: [PATCH 447/504] ARM: dts: iwg21m: enable tmu1, tmu2 and tmu3

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 2dd3e25..0c725b4 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -721,6 +721,18 @@
 	status = "okay";
 };
 
+&tmu1 {
+	status = "okay";
+};
+
+&tmu2 {
+	status = "okay";
+};
+
+&tmu3 {
+	status = "okay";
+};
+
 &sdhi2 {
 	pinctrl-0 = <&sdhi2_pins>;
 	pinctrl-1 = <&sdhi2_pins_uhs>;
-- 
2.7.4

