From 0aecf3bbfd91b22bee52a31bbaee83bf048381e8 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 28 Jul 2020 13:27:24 +0700
Subject: [PATCH 451/504] ARM: dts: iwg21m: enable cmt0 and cmt1

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 0c725b4..d1e0bd6 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -721,6 +721,14 @@
 	status = "okay";
 };
 
+&cmt0{
+	status = "okay";
+};
+
+&cmt1{
+	status = "okay";
+};
+
 &tmu1 {
 	status = "okay";
 };
-- 
2.7.4

