From 49490f47890d89c597da7d9df408a2e397c4aad4 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Tue, 25 Sep 2018 18:10:36 +0700
Subject: [PATCH 475/504] usb: gadget: composite: Check and return if cdev is
 NULL

Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 drivers/usb/gadget/composite.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 10f8fde..8fb4112 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -1890,6 +1890,10 @@ void composite_disconnect(struct usb_gadget *gadget)
 	struct usb_composite_dev	*cdev = get_gadget_data(gadget);
 	unsigned long			flags;
 
+	/*workaround: check if cdev is NULL. Exit function*/
+	if (!cdev)
+		return;
+
 	/* REVISIT:  should we have config and device level
 	 * disconnect callbacks?
 	 */
-- 
2.7.4

