From 601aa5e5727965431e26b0c79130e07076bbfe78 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Mon, 3 Aug 2020 16:20:59 +0700
Subject: [PATCH 488/504] ARM: dts: iwg21m: enable USB PHY

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index b82f6fc..1bc97cf 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -313,6 +313,10 @@
 	status = "okay";
 };
 
+&usbphy {
+	status = "okay";
+};
+
 &i2c0 {
 	pinctrl-0 = <&i2c0_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

