From 980f80604dda43f7952d82700944dac6434420d8 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Mon, 3 Aug 2020 16:23:37 +0700
Subject: [PATCH 490/504] ARM: dts: iwg21m: Add VBUS and PWEN gpio for USB OTG

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 2c39ff9..af7d996 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -315,6 +315,10 @@
 
 &usbphy {
 	status = "okay";
+	renesas,id-gpio = <&gpio5 20 GPIO_ACTIVE_LOW>;
+	renesas,vbus-gpio = <&gpio5 19 GPIO_ACTIVE_LOW>;
+	renesas,vbus-pwr-gpio = <&gpio5 18 GPIO_ACTIVE_LOW>;
+	renesas,wakeup;
 };
 
 &i2c0 {
-- 
2.7.4

