diff --git a/include/configs/rzg3e-dev.h b/include/configs/rzg3e-dev.h
index 0ef3e47ba8..3dbf7e65ba 100644
--- a/include/configs/rzg3e-dev.h
+++ b/include/configs/rzg3e-dev.h
@@ -47,13 +47,16 @@
 #define CONFIG_BOARD_SIZE_LIMIT		1048576
 
 /* ENV setting */
-#define CONFIG_EXTRA_ENV_SETTINGS	\
-	"bootm_size=0x10000000\0"
-
-#define CONFIG_BOOTCOMMAND	\
-	"tftp 0x48080000 Image; " \
-	"tftp 0x48000000 Image-"CONFIG_DEFAULT_FDT_FILE"; " \
-	"booti 0x48080000 - 0x48000000"
+#define CFG_EXTRA_ENV_SETTINGS	\
+	"bootm_size=0x10000000 \0"	\
+	"prodsdbootargs=setenv bootargs console=ttySC0,115200 rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
+	"prodemmcbootargs=setenv bootargs console=ttySC0,115200 rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
+	"bootimage=booti 0x48080000 - 0x48000000 \0" \
+	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g047e57-smarc.dtb;run prodemmcbootargs \0" \
+	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a09g047e57-smarc.dtb;run prodsdbootargs \0" \
+	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
+
+#define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
 
 /* For board */
 /* Ethernet RAVB */
-- 
2.34.1

