From 878dbeef722f0f69744a6aff8702120e420d7de5 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Thu, 29 Oct 2020 10:29:07 +0700
Subject: [PATCH 290/442] ARM: dts: r8a7742-iwg21d-q7: Enable PCI[01]

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index dd3b5b7..5ee4e94 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -389,6 +389,14 @@
 	};
 };
 
+&pci0 {
+	status = "okay";
+};
+
+&pci1 {
+	status = "okay";
+};
+
 &rcar_sound {
 	pinctrl-0 = <&sound_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

