From 48b5636b3a5a3fe4ebe51ec00bec7530165de63c Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Mon, 18 Oct 2021 14:05:05 +0700
Subject: [PATCH] waylandsink: Add mising code for scale feature

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 ext/wayland/wlwindow.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index b8377e4..f9e8926 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -356,11 +356,9 @@ gst_wl_window_new_toplevel (GstWlDisplay * display, const GstVideoInfo * info,
     goto error;
   }
 
-  /* set the initial size to be the same as the reported video size */
+  /* set the initial size to be the same as the reported configuration*/
   if (!(display->xdg_wm_base && fullscreen)) {
-    gint width =
-        gst_util_uint64_scale_int_round (info->width, info->par_n, info->par_d);
-    gst_wl_window_set_render_rectangle (window, 0, 0, width, info->height);
+    gst_wl_window_set_render_rectangle (window, 0, 0, surf_w, surf_h);
   }
 
   return window;
@@ -420,7 +418,7 @@ gst_wl_window_resize_video_surface (GstWlWindow * window, gboolean commit)
 {
   GstVideoRectangle src = { 0, };
   GstVideoRectangle dst = { 0, };
-  GstVideoRectangle res;
+  GstVideoRectangle res = { 0, };
 
   /* center the video_subsurface inside area_subsurface */
   src.w = window->video_width;
@@ -429,7 +427,10 @@ gst_wl_window_resize_video_surface (GstWlWindow * window, gboolean commit)
   dst.h = window->render_rectangle.h;
 
   if (window->video_viewport) {
-    gst_video_sink_center_rect (src, dst, &res, TRUE);
+    //  gst_video_sink_center_rect (src, dst, &res, TRUE);
+    /*  We do not want to center the video since we can scale it*/
+    res.w = dst.w;
+    res.h = dst.h;
     wp_viewport_set_destination (window->video_viewport, res.w, res.h);
   } else {
     gst_video_sink_center_rect (src, dst, &res, FALSE);
-- 
2.7.4
