From aa8591fc5cd3ec9b504968c6646ee499398d9699 Mon Sep 17 00:00:00 2001
From: Lad Prabhakar <prabhakar.mahadev-lad.rj@bp.renesas.com>
Date: Thu, 18 Feb 2021 18:57:20 +0000
Subject: [PATCH 080/135] arm64: dts: r9a07g044l: Add ADC node to SoC dtsi

Signed-off-by: Lad Prabhakar <prabhakar.mahadev-lad.rj@bp.renesas.com>
---
 arch/arm64/boot/dts/renesas/r9a07g044l.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
index 000fe34..5941253 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
@@ -224,6 +224,16 @@
 			status = "disabled";
 		};
 
+		adc: adc@10059000 {
+			compatible = "renesas,adc-r9a07g044l";
+			reg = <0 0x10059000 0 0x400>;
+			interrupts = <GIC_SPI 347  IRQ_TYPE_EDGE_RISING>;
+			clocks = <&cpg CPG_MOD R9A07G044L_CLK_ADC>;
+			resets = <&cpg R9A07G044L_CLK_ADC>;
+			power-domains = <&cpg>;
+			status = "disabled";
+		};
+
 		dma0: dma-controller@11800000 {
 			compatible = "renesas,rza-dma";
 			reg =   <0 0x11820000 0 0x10000>,
-- 
2.7.4

