diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
index 15598e2..d660359 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
@@ -609,6 +609,24 @@
 			status = "disabled";
 		};
 
+                fcpcs: vcp4@10820000 {
+                        compatible = "renesas,vcp4-fcpcs";
+                        reg = <0 0x10820000 0 0x200>;
+                        renesas,#ch = <0>;
+                };
+
+                vcpl4: vcp4@10800000 {
+                        compatible = "renesas,vcp4-vcpl4";
+                        reg = <0 0x10800000 0 0x200>, <0 0x10810200 0 0x200>;
+                        interrupts = <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+                                     <GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>;
+                        clocks = <&cpg CPG_MOD R9A07G044L_CLK_H264>;
+                        power-domains = <&cpg>;
+                        resets = <&cpg R9A07G044L_CLK_H264>;
+                        renesas,#ch = <0>;
+                        renesas,#fcp_ch = <0>;
+                };
+
 		vspd0: vsp@10870000 {
 			compatible = "renesas,vsp2";
 			reg = <0 0x10870000 0 0x10000>;

 
