diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
index 149a02b..7318df7 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-smarc.dts
@@ -28,8 +28,24 @@
 			compatible = "shared-dma-pool";
 			linux,cma-default;
 			reusable;
-			reg = <0x0 0x58000000 0x0 0x10000000>;
+			reg = <0x0 0x58000000 0x0 0x14000000>;
 		};
+
+                mmp_reserved: linux,multimedia {
+                        compatible = "shared-dma-pool";
+                        reusable;
+                        reg = <0x00000000 0x6C000000 0x0 0xa000000>;
+                };
 	};
+
+        mmngr {
+                compatible = "renesas,mmngr";
+                memory-region = <&mmp_reserved>;
+        };
+
+        mmngrbuf {
+                compatible = "renesas,mmngrbuf";
+        };
+
 };
 
