diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
index d6603595a2ef..070d327ffe30 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
@@ -627,6 +627,23 @@
                         renesas,#fcp_ch = <0>;
                 };
 
+                isu: isum@10840000 {
+                        compatible = "renesas,isum";
+                        reg = <0 0x10840000 0 0x01F4>;
+                        interrupts =    <GIC_SPI 157 IRQ_TYPE_LEVEL_HIGH>,
+                                        <GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
+                                        <GIC_SPI 159 IRQ_TYPE_LEVEL_HIGH>,
+                                        <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>;
+                        clocks = <&cpg CPG_MOD R9A07G044L_CLK_ISU>;
+                        clock-names = "isu";
+                        resets = <&cpg R9A07G044L_CLK_ISU>;
+                        renesas,#ch = <0>;
+                        renesas,#rpf = <1>;
+                        renesas,#wpf = <1>;
+                        renesas,has-rs;
+                        power-domains = <&cpg>;
+                };
+
 		vspd0: vsp@10870000 {
 			compatible = "renesas,vsp2";
 			reg = <0 0x10870000 0 0x10000>;

 
