From 42ba756e2f1e5567e0101540e8638c627ddd6451 Mon Sep 17 00:00:00 2001
From: Tai Huynh <tai.huynh.cp@renesas.com>
Date: Wed, 18 May 2022 09:55:16 +0700
Subject: [PATCH] chg-l2cache-clear

Signed-off-by: Tai Huynh <tai.huynh.cp@renesas.com>
---
 arch/arm/cpu/armv8/cache.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/cpu/armv8/cache.S b/arch/arm/cpu/armv8/cache.S
index e04907dd8c..e7acc7c9be 100644
--- a/arch/arm/cpu/armv8/cache.S
+++ b/arch/arm/cpu/armv8/cache.S
@@ -87,7 +87,7 @@ loop_level:
 	add	x12, x12, x0		/* x0 <- tripled cache level */
 	lsr	x12, x10, x12
 	and	x12, x12, #7		/* x12 <- cache type */
-	cmp	x12, #2
+	cmp     x12, #1
 	b.lt	skip			/* skip if no cache or icache */
 	bl	__asm_dcache_level	/* x1 = 0 flush, 1 invalidate */
 skip:
-- 
2.25.1

