diff --git a/drivers/tty/serial/8250/8250.h b/drivers/tty/serial/8250/8250.h
index 34aa271..afeb5df 100644
--- a/drivers/tty/serial/8250/8250.h
+++ b/drivers/tty/serial/8250/8250.h
@@ -133,11 +133,18 @@ static inline void serial_dl_write(struct uart_8250_port *up, int value)
 
 static inline bool serial8250_set_THRI(struct uart_8250_port *up)
 {
+	struct uart_port *port = &up->port;
+	struct circ_buf *xmit = &port->state->xmit;
+	
 	if (up->ier & UART_IER_THRI)
 		return false;
-	up->ier |= UART_IER_THRI;
-	serial_out(up, UART_IER, up->ier);
-	return true;
+	
+	if (!uart_circ_empty(xmit)) {
+		up->ier |= UART_IER_THRI;
+		serial_out(up, UART_IER, up->ier);
+		return true;
+	} else
+		return false;
 }
 
 static inline bool serial8250_clear_THRI(struct uart_8250_port *up)
