From e94c524d31f06fefee2caac71809410d50514d14 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Linh [FGA.BU13]" <LinhNV25@fsoft.com.vn>
Date: Tue, 10 Apr 2018 08:46:53 +0700
Subject: [PATCH 289/642] ARM: dts: r8a77470: enable USB2.0

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470-iwg23s.dts | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470-iwg23s.dts b/arch/arm/boot/dts/r8a77470-iwg23s.dts
index b4e4348..987c212 100644
--- a/arch/arm/boot/dts/r8a77470-iwg23s.dts
+++ b/arch/arm/boot/dts/r8a77470-iwg23s.dts
@@ -275,6 +275,26 @@
 	status = "okay";
 };
 
+&usbphy0 {
+	status = "okay";
+};
+
+&usbphy1 {
+	status = "okay";
+};
+
+&hsusb {
+	status = "okay";
+};
+
+&ohci_1 {
+	status = "okay";
+};
+
+&ehci_1 {
+	status = "okay";
+};
+
 &can0 {
 	pinctrl-0 = <&can0_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

