diff --git a/drivers/net/ravb.c b/drivers/net/ravb.c
index 6a628c0..2589943 100755
--- a/drivers/net/ravb.c
+++ b/drivers/net/ravb.c
@@ -538,7 +538,9 @@ static int ravb_probe(struct udevice *dev)
 {
 	struct eth_pdata *pdata = dev_get_plat(dev);
 	struct ravb_priv *eth = dev_get_priv(dev);
+#ifndef CONFIG_R9A09G055MA3GBG
 	struct ofnode_phandle_args phandle_args;
+#endif
 	struct mii_dev *mdiodev;
 	void __iomem *iobase;
 	int ret;
