diff --git a/board/renesas/rzv2ma-dev/rzv2ma-dev.c b/board/renesas/rzv2ma-dev/rzv2ma-dev.c
index 1dd5ced..98c98d4 100755
--- a/board/renesas/rzv2ma-dev/rzv2ma-dev.c
+++ b/board/renesas/rzv2ma-dev/rzv2ma-dev.c
@@ -417,53 +417,6 @@ static void dmaa_init(void)
 	CPG_WaitResetMon(0, CPG_RST_MON_DMAA,0);
 }
 
-static void icb_init(void)
-{
-    /* VCD_M0R */
-    CMN_REG_Write32(0x98004B08,  0x80000000);
-    CMN_REG_Write32(0x98004B0C,  0x00000001);
-    CMN_REG_Write32(0x98004B10,  0x00000500);
-    CMN_REG_Write32(0x98004B14,  0x00000040);
-
-    /* VCD_M0W */
-    CMN_REG_Write32(0x98005208,  0x80000000);
-    CMN_REG_Write32(0x9800520C,  0x00000001);
-    CMN_REG_Write32(0x98005210,  0x00000500);
-    CMN_REG_Write32(0x98005214,  0x00000040);
-
-    /* DRPA_M0 */
-    CMN_REG_Write32(0x98005308,  0x80000000);
-    CMN_REG_Write32(0x9800530C,  0x00000001);
-    CMN_REG_Write32(0x98005310,  0x00000700);
-    CMN_REG_Write32(0x98005314,  0x00000010);
-
-    /* DRPA_M1 */
-    CMN_REG_Write32(0x98005408,  0x80000000);
-    CMN_REG_Write32(0x9800540C,  0x00000001);
-    CMN_REG_Write32(0x98005410,  0x00000700);
-    CMN_REG_Write32(0x98005414,  0x00000010);
-
-    /* DRPA_M2 */
-    CMN_REG_Write32(0x98005508,  0x80000000);
-    CMN_REG_Write32(0x9800550C,  0x00000001);
-    CMN_REG_Write32(0x98005510,  0x00000700);
-    CMN_REG_Write32(0x98005514,  0x00000010);
-
-    /* DRPA_M3 */
-    CMN_REG_Write32(0x98004E08,  0x80000000);
-    CMN_REG_Write32(0x98004E0C,  0x00000001);
-    CMN_REG_Write32(0x98004E10,  0x00000280);
-    CMN_REG_Write32(0x98004E14,  0x00000010);
-
-    /* DRPB_M0 */
-    CMN_REG_Write32(0x98005808,  0x80000000);
-    CMN_REG_Write32(0x9800580C,  0x00000001);
-    CMN_REG_Write32(0x98005810,  0x00000280);
-    CMN_REG_Write32(0x98005814,  0x00000010);
-    CMN_DelayInUS(1);
-}
-
-
 static void evk_cpg_init(void)
 {
 	do
@@ -525,10 +478,6 @@ static void evk_cpg_init(void)
 		vcd_init();
 		pcie_init();
 
-		/**********************************/
-		/*    BUS(ICB)                    */
-		/**********************************/
-		icb_init();
 	}
 	while (0);
 	return;
