From efea0b539286488e37af6a60c0872788bbd5e2ce Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 13 Aug 2015 14:36:17 +0300
Subject: [PATCH] qt event notifier

---
 src/evdev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/evdev.c b/src/evdev.c
index 888dfbd..d090bae 100644
--- a/src/evdev.c
+++ b/src/evdev.c
@@ -377,6 +377,8 @@ fallback_process(struct evdev_dispatch *dispatch,
 		break;
 	case EV_SYN:
 		evdev_flush_pending_event(device, time);
+		if (device->seat_caps & EVDEV_SEAT_TOUCH)
+			notify_touch_frame(device->seat);
 		break;
 	}
 }
-- 
1.9.1

