From a8ef7b684b8d2aa6efcb48d14f964a92fee77aaa Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Tue, 28 Jul 2015 18:46:19 +0300
Subject: [PATCH] workaround hotplug

---
 src/udev-seat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev-seat.c b/src/udev-seat.c
index 8e7405d..3983de9 100644
--- a/src/udev-seat.c
+++ b/src/udev-seat.c
@@ -185,7 +185,7 @@ udev_input_add_devices(struct udev_input *input, struct udev *udev)
 			"\t- seats misconfigured "
 			"(Weston backend option 'seat', "
 			"udev device property ID_SEAT)\n");
-		return -1;
+		return 0; /*workaround hotplug   -1;*/
 	}
 
 	return 0;
-- 
1.9.1

