From d66801f576011331bb1e23b3e7d3510fe17db594 Mon Sep 17 00:00:00 2001
From: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
Date: Wed, 18 Feb 2015 14:09:09 +0300
Subject: [PATCH 2/3] Add non-blocking display_poll routine to dispatch events


Signed-off-by: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
---
 clients/window.c |   41 +++++++++++++++++++++++++++++++++++++++++
 clients/window.h |    3 +++
 2 files changed, 44 insertions(+)

diff --git a/clients/window.c b/clients/window.c
index 4592ef9..182e101 100644
--- a/clients/window.c
+++ b/clients/window.c
@@ -5531,6 +5531,47 @@ display_unwatch_fd(struct display *display, int fd)
 	epoll_ctl(display->epoll_fd, EPOLL_CTL_DEL, fd, NULL);
 }
 
+int
+display_poll(struct display *display, int timeout)
+{
+	struct task *task;
+	struct epoll_event ep[16];
+	int i, count, ret;
+
+	display->running = 1;
+	while (!wl_list_empty(&display->deferred_list)) {
+		task = container_of(display->deferred_list.prev,
+				    struct task, link);
+		wl_list_remove(&task->link);
+		task->run(task, 0);
+	}
+
+	wl_display_dispatch_pending(display->display);
+
+	if (!display->running)
+		return -1;
+
+	ret = wl_display_flush(display->display);
+	if (ret < 0 && errno == EAGAIN) {
+		ep[0].events =
+			EPOLLIN | EPOLLOUT | EPOLLERR | EPOLLHUP;
+		ep[0].data.ptr = &display->display_task;
+
+		epoll_ctl(display->epoll_fd, EPOLL_CTL_MOD,
+			  display->display_fd, &ep[0]);
+	} else if (ret < 0) {
+		return -1;
+	}
+
+	count = epoll_wait(display->epoll_fd,
+			   ep, ARRAY_LENGTH(ep), timeout);
+	for (i = 0; i < count; i++) {
+		task = ep[i].data.ptr;
+		task->run(task, ep[i].events);
+	}
+	return 0;
+}
+
 void
 display_run(struct display *display)
 {
diff --git a/clients/window.h b/clients/window.h
index 4e4ccc1..40dc3f6 100644
--- a/clients/window.h
+++ b/clients/window.h
@@ -173,6 +173,9 @@ display_watch_fd(struct display *display,
 void
 display_unwatch_fd(struct display *display, int fd);
 
+int
+display_poll(struct display *display, int timeout);
+
 void
 display_run(struct display *d);
 
-- 
1.7.10.4

