From 89d767953851620368787ad389ee90d336779b15 Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Tue, 5 May 2015 21:39:20 +0300
Subject: [PATCH] notify client immediately on buffer release.

---
 src/compositor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor.c b/src/compositor.c
index 574db2d..3ea8aeb 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -1491,7 +1491,7 @@ weston_buffer_reference(struct weston_buffer_reference *ref,
 		ref->buffer->busy_count--;
 		if (ref->buffer->busy_count == 0) {
 			assert(wl_resource_get_client(ref->buffer->resource));
-			wl_resource_queue_event(ref->buffer->resource,
+			wl_resource_post_event(ref->buffer->resource,
 						WL_BUFFER_RELEASE);
 		}
 		wl_list_remove(&ref->destroy_listener.link);
-- 
1.9.1

