From 8a272436f8f8effcf928d8f9c33ce2de1309a75d Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Sat, 25 Apr 2015 15:42:19 +0300
Subject: [PATCH] compositor-drm: do not fail if there are no input devices

---
 src/compositor-drm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 38f1618..c3ab077 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -2931,11 +2931,15 @@ drm_compositor_create(struct wl_display *display,
 	wl_list_init(&ec->sprite_list);
 	create_sprites(ec);
 
+#if 0
 	if (udev_input_init(&ec->input,
 			    &ec->base, ec->udev, param->seat_id) < 0) {
 		weston_log("failed to create input devices\n");
 		goto err_sprite;
 	}
+#else
+	udev_input_init(&ec->input, &ec->base, ec->udev, param->seat_id);
+#endif
 
 	if (create_outputs(ec, param->connector, drm_device) < 0) {
 		weston_log("failed to create output for %s\n", path);
-- 
1.9.1

