From 2ef31647aa7c34c91a65d628dc9ea53814080894 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Fri, 31 Jul 2020 17:56:37 +0700
Subject: [PATCH 362/440] ARM: dts: r8a7742: Add vspd{0,1} and vspm nodes

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 7d76b13..b467a9e 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1286,6 +1286,38 @@
 			status = "disabled";
 		};
 
+		vspm@fe920000 {
+			compatible = "renesas,vspm-vspr", "renesas-vspm";
+			reg = <0 0xfe920000 0 0x8000>;
+			interrupts = <0 266 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_S>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspm@fe928000 {
+			compatible = "renesas,vspm-vsps", "renesas-vspm";
+			reg = <0 0xfe928000 0 0x7404>;
+			interrupts = <0 267 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_S>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspd0: vspd0@fe930000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe930000 0 0x7404>;
+			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_DU0>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
+		vspd1: vspd1@fe938000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe938000 0 0x7404>;
+			interrupts = <0 247 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7742_CLK_VSP1_DU1>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		};
+
 		scifb0: serial@e6c20000 {
 			compatible = "renesas,scifb-r8a7742",
 				     "renesas,scifb";
@@ -2074,6 +2106,8 @@
 			clock-names = "du.0", "du.1", "du.2", "lvds.0", "lvds.1";
 			status = "disabled";
 
+			vsps = <&vspd0 &vspd1>;
+
 			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
-- 
2.7.4

