From 855860aead0674db2e7f88219efa4c43988e067f Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 13 Jul 2020 16:30:44 +0700
Subject: [PATCH 134/441] net: can: rcar_can: Add r8a7742 support

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/net/can/rcar_can.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/can/rcar_can.c b/drivers/net/can/rcar_can.c
index 8807724..3b3940e 100644
--- a/drivers/net/can/rcar_can.c
+++ b/drivers/net/can/rcar_can.c
@@ -903,6 +903,7 @@ static int __maybe_unused rcar_can_resume(struct device *dev)
 static SIMPLE_DEV_PM_OPS(rcar_can_pm_ops, rcar_can_suspend, rcar_can_resume);
 
 static const struct of_device_id rcar_can_of_table[] __maybe_unused = {
+	{ .compatible = "renesas,can-r8a7742" },
 	{ .compatible = "renesas,can-r8a7744" },
 	{ .compatible = "renesas,can-r8a7745" },
 	{ .compatible = "renesas,can-r8a77470" },
-- 
2.7.4

