From 0670263789690a2b519fdf17c8ca52aadec49a3e Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 14 Jul 2020 10:32:03 +0700
Subject: [PATCH 065/433] ARM: dts: r8a77470: correct gpio-ranges settings

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index f29461e..1d1c1f9 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -102,7 +102,7 @@
 			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
 			#gpio-cells = <2>;
 			gpio-controller;
-			gpio-ranges = <&pfc 0 0 23>;
+			gpio-ranges = <&pfc 0 0 32>;
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO0>;
@@ -116,7 +116,7 @@
 			interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>;
 			#gpio-cells = <2>;
 			gpio-controller;
-			gpio-ranges = <&pfc 0 32 23>;
+			gpio-ranges = <&pfc 0 32 32>;
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO1>;
@@ -144,7 +144,7 @@
 			interrupts = <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>;
 			#gpio-cells = <2>;
 			gpio-controller;
-			gpio-ranges = <&pfc 0 96 30>;
+			gpio-ranges = <&pfc 0 96 32>;
 			gpio-reserved-ranges = <17 10>;
 			#interrupt-cells = <2>;
 			interrupt-controller;
@@ -159,7 +159,7 @@
 			interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
 			#gpio-cells = <2>;
 			gpio-controller;
-			gpio-ranges = <&pfc 0 128 26>;
+			gpio-ranges = <&pfc 0 128 32>;
 			#interrupt-cells = <2>;
 			interrupt-controller;
 			clocks = <&mstp9_clks R8A77470_CLK_GPIO4>;
-- 
2.7.4

