From 238dcfff8801411272961f189890f2d4e89d881f Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Fri, 17 Jul 2020 15:45:27 +0700
Subject: [PATCH 070/433] gpio: rcar: Add subsys_initcall for early timing
 startup

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 drivers/gpio/gpio-rcar.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-rcar.c b/drivers/gpio/gpio-rcar.c
index e1b5914..f171665 100644
--- a/drivers/gpio/gpio-rcar.c
+++ b/drivers/gpio/gpio-rcar.c
@@ -618,7 +618,17 @@ static struct platform_driver gpio_rcar_device_driver = {
 	}
 };
 
-module_platform_driver(gpio_rcar_device_driver);
+static int __init gpio_rcar_init(void)
+{
+	return platform_driver_register(&gpio_rcar_device_driver);
+}
+subsys_initcall(gpio_rcar_init);
+
+static void __exit gpio_rcar_exit(void)
+{
+	platform_driver_unregister(&gpio_rcar_device_driver);
+}
+module_exit(gpio_rcar_exit);
 
 MODULE_AUTHOR("Magnus Damm");
 MODULE_DESCRIPTION("Renesas R-Car GPIO Driver");
-- 
2.7.4

