From 3e359d01a7713ac2cb3e9cdeb39544d9ff0f288b Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 14 Jul 2020 13:23:42 +0700
Subject: [PATCH 073/433] i2c: rcar: Fix random address read issue

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 drivers/i2c/busses/i2c-rcar.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rcar.c b/drivers/i2c/busses/i2c-rcar.c
index aef95a9..fac2d6c 100644
--- a/drivers/i2c/busses/i2c-rcar.c
+++ b/drivers/i2c/busses/i2c-rcar.c
@@ -450,6 +450,7 @@ static irqreturn_t rcar_i2c_irq(int irq, void *ptr)
 		/* HW automatically sends STOP after received NACK */
 		rcar_i2c_write(priv, ICMIER, RCAR_IRQ_STOP);
 		rcar_i2c_flags_set(priv, ID_NACK);
+		rcar_i2c_write(priv, ICMSR, 0);
 		goto out;
 	}
 
@@ -457,6 +458,7 @@ static irqreturn_t rcar_i2c_irq(int irq, void *ptr)
 	if (msr & MST) {
 		priv->msgs_left--; /* The last message also made it */
 		rcar_i2c_flags_set(priv, ID_DONE);
+		rcar_i2c_write(priv, ICMSR, 0);
 		goto out;
 	}
 
@@ -468,7 +470,6 @@ static irqreturn_t rcar_i2c_irq(int irq, void *ptr)
 out:
 	if (rcar_i2c_flags_has(priv, ID_DONE)) {
 		rcar_i2c_write(priv, ICMIER, 0);
-		rcar_i2c_write(priv, ICMSR, 0);
 		wake_up(&priv->wait);
 	}
 
-- 
2.7.4

