From 4f45391bfcd2159a2f01c64855dd4385cb1aa987 Mon Sep 17 00:00:00 2001
From: Anh Tran <anh.tran.jc@renesas.com>
Date: Wed, 15 Nov 2017 17:24:11 +0700
Subject: [PATCH 100/433] ASoC: simple-card: Add DAPM Jack component

Signed-off-by: Anh Tran <anh.tran.jc@renesas.com>
---
 sound/soc/generic/simple-card.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sound/soc/generic/simple-card.c b/sound/soc/generic/simple-card.c
index 0b1b6fc..7c19320 100644
--- a/sound/soc/generic/simple-card.c
+++ b/sound/soc/generic/simple-card.c
@@ -213,6 +213,14 @@ static int asoc_simple_card_dai_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
+static const struct snd_soc_dapm_widget sgtl5000_dapm_widgets[] = {
+	SND_SOC_DAPM_MIC("Mic Jack", NULL),
+	SND_SOC_DAPM_LINE("Line In Jack", NULL),
+	SND_SOC_DAPM_HP("Headphone Jack", NULL),
+	SND_SOC_DAPM_SPK("Line Out Jack", NULL),
+	SND_SOC_DAPM_SPK("Ext Spk", NULL),
+};
+
 static int
 asoc_simple_card_sub_parse_of(struct device_node *np,
 			      struct asoc_simple_dai *dai,
@@ -563,6 +571,9 @@ static int asoc_simple_card_probe(struct platform_device *pdev)
 	priv->snd_card.dai_link = dai_link;
 	priv->snd_card.num_links = num_links;
 
+	priv->snd_card.dapm_widgets = sgtl5000_dapm_widgets;
+	priv->snd_card.num_dapm_widgets = ARRAY_SIZE(sgtl5000_dapm_widgets);
+
 	priv->gpio_hp_det = -ENOENT;
 	priv->gpio_mic_det = -ENOENT;
 
-- 
2.7.4

