From 6d728fa2f43d1eb01d8afbcd6d3915a6aee24629 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Mon, 22 Jul 2019 11:43:44 +0700
Subject: [PATCH 113/433] ARM: dts: r8a77470: Add msiof{0,1,2} support

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.comL>
---
 arch/arm/boot/dts/r8a77470.dtsi | 42 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 860f882..afd72eb 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -23,6 +23,9 @@
 		i2c4 = &i2c4;
 		spi0 = &qspi0;
 		spi1 = &qspi1;
+		spi2 = &msiof0;
+		spi3 = &msiof1;
+		spi4 = &msiof2;
 	};
 
 	cpus {
@@ -525,6 +528,45 @@
 			status = "disabled";
 		};
 
+		msiof0: spi@e6e20000 {
+			compatible = "renesas,msiof-r8a77470";
+			reg = <0 0xe6e20000 0 0x0064>;
+			interrupts = <0 156 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp0_clks R8A77470_CLK_MSIOF0>;
+			dmas = <&dmac0 0x51>, <&dmac0 0x52>;
+			dma-names = "tx", "rx";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		msiof1: spi@e6e10000 {
+			compatible = "renesas,msiof-r8a77470";
+			reg = <0 0xe6e10000 0 0x0064>;
+			interrupts = <0 157 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp2_clks R8A77470_CLK_MSIOF1>;
+			dmas = <&dmac0 0x55>, <&dmac0 0x56>;
+			dma-names = "tx", "rx";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		msiof2: spi@e6e00000 {
+			compatible = "renesas,msiof-r8a77470";
+			reg = <0 0xe6e00000 0 0x0064>;
+			interrupts = <0 158 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp2_clks R8A77470_CLK_MSIOF2>;
+			dmas = <&dmac0 0x41>, <&dmac0 0x42>;
+			dma-names = "tx", "rx";
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
 		scif0: serial@e6e60000 {
 			compatible = "renesas,scif-r8a77470", "renesas,scif";
 			reg = <0 0xe6e60000 0 0x40>;
-- 
2.7.4

