From 3cea2fdd2bd481766400fcf50934c6db8dd7bda0 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 3 Aug 2020 14:33:18 +0700
Subject: [PATCH 150/433] phy: rcar-gen2: Add support for r8a7744 USB OTG

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/phy/phy-rcar-gen2.c | 33 +++++++++++++++++++++++----------
 1 file changed, 23 insertions(+), 10 deletions(-)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index 2a27ee3..d8c1610 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -235,7 +235,9 @@ static int rcar_gen2_phy_power_on(struct phy *p)
 	writew(value, usbhs_base + USBHS_LPSTS);
 
 	/* USBHS_UGCTRL_CONNECT bit only exists on RZG1H,M,N,E SoC USBHS_UGSTS reg */
-	if (of_machine_is_compatible("renesas,r8a7743") || of_machine_is_compatible("renesas,r8a7745")) {
+	if (of_machine_is_compatible("renesas,r8a7743")
+		|| of_machine_is_compatible("renesas,r8a7744")
+		|| of_machine_is_compatible("renesas,r8a7745")) {
 		for (i = 0; i < 20; i++) {
 			value = readl(usbhs_base + USBHS_UGSTS);
 			if ((value & USBHS_UGSTS_LOCK) == USBHS_UGSTS_LOCK) {
@@ -273,7 +275,9 @@ static int rcar_gen2_phy_power_off(struct phy *p)
 	spin_lock_irqsave(&drv->lock, flags);
 
 	/* Power off USBHS PHY */
-	if (of_machine_is_compatible("renesas,r8a7743") || of_machine_is_compatible("renesas,r8a7745")) {
+	if (of_machine_is_compatible("renesas,r8a7743")
+		|| of_machine_is_compatible("renesas,r8a7744")
+		|| of_machine_is_compatible("renesas,r8a7745")) {
 		value = readl(usbhs_base + USBHS_UGCTRL);
 		value &= ~USBHS_UGCTRL_CONNECT;
 		writel(value, usbhs_base + USBHS_UGCTRL);
@@ -394,7 +398,8 @@ static void gpio_id_work(struct work_struct *work)
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743"))
+		else if (of_machine_is_compatible("renesas,r8a7743") ||
+			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -403,7 +408,8 @@ static void gpio_id_work(struct work_struct *work)
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743"))
+		else if (of_machine_is_compatible("renesas,r8a7743") ||
+			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -436,7 +442,8 @@ static void gpio_vbus_work(struct work_struct *work)
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743"))
+		else if (of_machine_is_compatible("renesas,r8a7743") ||
+			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -445,7 +452,8 @@ static void gpio_vbus_work(struct work_struct *work)
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
-		else if (of_machine_is_compatible("renesas,r8a7743"))
+		else if (of_machine_is_compatible("renesas,r8a7743") ||
+			of_machine_is_compatible("renesas,r8a7744"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -558,8 +566,9 @@ static int rcar_gen2_usb_set_peripheral(struct usb_otg *otg,
 		otg->gadget = gadget;
 
 		/* initialize connection state */
-		if (of_machine_is_compatible("renesas,r8a7743") ||
-			of_machine_is_compatible("renesas,r8a7745")) {
+		if (of_machine_is_compatible("renesas,r8a7743")
+			|| of_machine_is_compatible("renesas,r8a7744")
+			|| of_machine_is_compatible("renesas,r8a7745")) {
 			gpio_vbus_irq(channel->irq_vbus, channel);
 		} else if (of_machine_is_compatible("renesas,r8a77470"))
 			gpio_vbus_irq(channel->irq_id, channel);
@@ -852,7 +861,9 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 			of_node_put(np);
 			return error;
 		}
-		if (of_machine_is_compatible("renesas,r8a7743") || of_machine_is_compatible("renesas,r8a7745")) {
+		if (of_machine_is_compatible("renesas,r8a7743")
+			|| of_machine_is_compatible("renesas,r8a7744")
+			|| of_machine_is_compatible("renesas,r8a7745")) {
 			channel->select_mask = select_mask[channel_num];
 			phys_per_channel = PHYS_PER_CHANNEL;
 		} else if (of_machine_is_compatible("renesas,r8a77470")) {
@@ -865,7 +876,9 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 
 			phy->channel = channel;
 			phy->number = n;
-			if (of_machine_is_compatible("renesas,r8a7743") || of_machine_is_compatible("renesas,r8a7745"))
+			if (of_machine_is_compatible("renesas,r8a7743")
+				|| of_machine_is_compatible("renesas,r8a7744")
+				|| of_machine_is_compatible("renesas,r8a7745"))
 				phy->select_value = select_value[channel_num][n];
 			else if (of_machine_is_compatible("renesas,r8a77470")) {
 				if (drv->channels->use_otg)
-- 
2.7.4

