From a864feb4c3ae5a8b2e30c8b52c3388ac6de72c56 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 3 Aug 2020 15:09:17 +0700
Subject: [PATCH 151/433] phy: rcar-gen2: Add support for r8a7742 USB OTG

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/phy/phy-rcar-gen2.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index d8c1610..c9fd96d 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -237,6 +237,7 @@ static int rcar_gen2_phy_power_on(struct phy *p)
 	/* USBHS_UGCTRL_CONNECT bit only exists on RZG1H,M,N,E SoC USBHS_UGSTS reg */
 	if (of_machine_is_compatible("renesas,r8a7743")
 		|| of_machine_is_compatible("renesas,r8a7744")
+		|| of_machine_is_compatible("renesas,r8a7742")
 		|| of_machine_is_compatible("renesas,r8a7745")) {
 		for (i = 0; i < 20; i++) {
 			value = readl(usbhs_base + USBHS_UGSTS);
@@ -277,6 +278,7 @@ static int rcar_gen2_phy_power_off(struct phy *p)
 	/* Power off USBHS PHY */
 	if (of_machine_is_compatible("renesas,r8a7743")
 		|| of_machine_is_compatible("renesas,r8a7744")
+		|| of_machine_is_compatible("renesas,r8a7742")
 		|| of_machine_is_compatible("renesas,r8a7745")) {
 		value = readl(usbhs_base + USBHS_UGCTRL);
 		value &= ~USBHS_UGCTRL_CONNECT;
@@ -399,7 +401,8 @@ static void gpio_id_work(struct work_struct *work)
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
 		else if (of_machine_is_compatible("renesas,r8a7743") ||
-			of_machine_is_compatible("renesas,r8a7744"))
+			of_machine_is_compatible("renesas,r8a7744") ||
+			of_machine_is_compatible("renesas,r8a7742"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -409,7 +412,8 @@ static void gpio_id_work(struct work_struct *work)
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
 		else if (of_machine_is_compatible("renesas,r8a7743") ||
-			of_machine_is_compatible("renesas,r8a7744"))
+			of_machine_is_compatible("renesas,r8a7744") ||
+			of_machine_is_compatible("renesas,r8a7742"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -443,7 +447,8 @@ static void gpio_vbus_work(struct work_struct *work)
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
 		else if (of_machine_is_compatible("renesas,r8a7743") ||
-			of_machine_is_compatible("renesas,r8a7744"))
+			of_machine_is_compatible("renesas,r8a7744") ||
+			of_machine_is_compatible("renesas,r8a7742"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -453,7 +458,8 @@ static void gpio_vbus_work(struct work_struct *work)
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_PCI);
 		else if (of_machine_is_compatible("renesas,r8a7743") ||
-			of_machine_is_compatible("renesas,r8a7744"))
+			of_machine_is_compatible("renesas,r8a7744") ||
+			of_machine_is_compatible("renesas,r8a7742"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
 		else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -568,6 +574,7 @@ static int rcar_gen2_usb_set_peripheral(struct usb_otg *otg,
 		/* initialize connection state */
 		if (of_machine_is_compatible("renesas,r8a7743")
 			|| of_machine_is_compatible("renesas,r8a7744")
+			|| of_machine_is_compatible("renesas,r8a7742")
 			|| of_machine_is_compatible("renesas,r8a7745")) {
 			gpio_vbus_irq(channel->irq_vbus, channel);
 		} else if (of_machine_is_compatible("renesas,r8a77470"))
@@ -863,6 +870,7 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 		}
 		if (of_machine_is_compatible("renesas,r8a7743")
 			|| of_machine_is_compatible("renesas,r8a7744")
+			|| of_machine_is_compatible("renesas,r8a7742")
 			|| of_machine_is_compatible("renesas,r8a7745")) {
 			channel->select_mask = select_mask[channel_num];
 			phys_per_channel = PHYS_PER_CHANNEL;
@@ -878,6 +886,7 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 			phy->number = n;
 			if (of_machine_is_compatible("renesas,r8a7743")
 				|| of_machine_is_compatible("renesas,r8a7744")
+				|| of_machine_is_compatible("renesas,r8a7742")
 				|| of_machine_is_compatible("renesas,r8a7745"))
 				phy->select_value = select_value[channel_num][n];
 			else if (of_machine_is_compatible("renesas,r8a77470")) {
-- 
2.7.4

