From 780111660a2fad38bb14017378a8299db0c4278a Mon Sep 17 00:00:00 2001
From: "Nguyen Van Linh [FGA.BU13]" <LinhNV25@fsoft.com.vn>
Date: Tue, 10 Apr 2018 09:26:31 +0700
Subject: [PATCH 195/433] media: v4l2-ctrl: Add CID base for r8a7747x

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 include/uapi/linux/v4l2-controls.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/uapi/linux/v4l2-controls.h b/include/uapi/linux/v4l2-controls.h
index 1bdce50..0fb2d32 100644
--- a/include/uapi/linux/v4l2-controls.h
+++ b/include/uapi/linux/v4l2-controls.h
@@ -178,6 +178,10 @@ enum v4l2_colorfx {
  * We reserve 16 controls for this driver. */
 #define V4L2_CID_USER_TC358743_BASE		(V4L2_CID_USER_BASE + 0x1080)
 
+/* The base for the r8a7747x driver controls.
+ * We reserve 16 controls for this driver. */
+#define V4L2_CID_USER_R8A7747X_BASE		(V4L2_CID_USER_BASE + 0x1090)
+
 /* MPEG-class control IDs */
 /* The MPEG controls are applicable to all codec controls
  * and the 'MPEG' part of the define is historical */
-- 
2.7.4

