From 76a47d61b1761236677ed82c2600babdf917d80e Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Fri, 23 Oct 2020 11:08:10 +0700
Subject: [PATCH 248/433] ARM: dts: r8a7742-iwg21d-q7: Enable tmu1

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21d-q7.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
index fe3cf6c..bbfbe27 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7.dts
@@ -379,6 +379,10 @@
 	status = "okay";
 };
 
+&tmu1 {
+	status = "okay";
+};
+
 &sdhi2 {
 	pinctrl-0 = <&sdhi2_pins>;
 	pinctrl-1 = <&sdhi2_pins_uhs>;
-- 
2.7.4

