From 3515e8b1e223734e42f3a938fd96167027321c00 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Tue, 14 Jul 2020 09:02:57 +0700
Subject: [PATCH 275/433] ARM: dts: r8a77470: Add PWM device nodes

Signed-off-by: tungvo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index b62d1c4..ffb096f 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -201,6 +201,33 @@
 			reg = <0 0xe6060000 0 0x118>;
 		};
 
+		pwm0: pwm@e6e30000 {
+			compatible = "renesas,pwm-r8a77470";
+			reg = <0 0xe6e30000 0 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&mstp5_clks R8A77470_CLK_PWM>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
+		pwm2: pwm@e6e32000 {
+			compatible = "renesas,pwm-r8a77470";
+			reg = <0 0xe6e32000 0 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&mstp5_clks R8A77470_CLK_PWM>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
+		pwm6: pwm@e6e36000 {
+			compatible = "renesas,pwm-r8a77470";
+			reg = <0 0xe6e36000 0 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&mstp5_clks R8A77470_CLK_PWM>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			status = "disabled";
+		};
+
 		apmu@e6151000 {
 			compatible = "renesas,r8a77470-apmu", "renesas,apmu";
 			reg = <0 0xe6151000 0 0x188>;
-- 
2.7.4

