From 51fd2ed9c551eef9ebf8e85af774110aea330575 Mon Sep 17 00:00:00 2001
From: Nhat Thieu <nhat.thieu.xr@renesas.com>
Date: Mon, 5 Jul 2021 21:03:16 +0700
Subject: [PATCH] Modify-vspm_public.h-for-ISUM

---
 .../files/vspm_if/include/vspm_public.h       | 93 +++++++++++++++++++
 1 file changed, 93 insertions(+)

diff --git a/vspm_if-module/files/vspm_if/include/vspm_public.h b/vspm_if-module/files/vspm_if/include/vspm_public.h
index 0125054..d441894 100644
--- a/vspm_if-module/files/vspm_if/include/vspm_public.h
+++ b/vspm_if-module/files/vspm_if/include/vspm_public.h
@@ -9,6 +9,7 @@
 
 #include "vsp_drv.h"
 #include "fdp_drv.h"
+#include "isu_drv.h"
 #include "vspm_cmn.h"
 
 /* callback function */
@@ -483,6 +484,86 @@ typedef struct {
 	unsigned char out_req;
 } T_FDP_STATUS;
 
+/* R-Car H2/M2 compatible ISU params */
+typedef struct {
+	unsigned char asel;
+	unsigned char anum0;
+	unsigned char anum1;
+	unsigned char anum2;
+	unsigned char athres0;
+	unsigned char athres1;
+	unsigned char acsc;
+} T_ISU_ALPHA;
+
+typedef struct {
+	unsigned char	grada_mode;
+	unsigned char	grada_step;
+	unsigned int	init_val;
+} T_ISU_TD;
+
+
+typedef struct {
+	unsigned int	hard_addr;
+	void		*virt_addr;
+	unsigned short	tbl_num;
+	void		*mem_par;
+} T_ISU_DL;
+
+typedef struct {
+	unsigned char	csc;
+	unsigned char	clrcng;
+	unsigned int	k_matrix[3][3];
+	unsigned int	offset[3][2];
+	unsigned int	clip[3][2];
+} T_ISU_CSC;
+
+typedef struct {
+	void *addr;					/* Y or RGB buffer address */
+	void *addr_c;				/* CbCr or CB buffer address */
+	unsigned int 	stride;
+	unsigned int 	stride_c;
+	unsigned int 	width;
+	unsigned int 	height;
+	unsigned char 	format;
+	unsigned char 	swap;
+	T_ISU_TD 	*td;
+	T_ISU_ALPHA 	*alpha;
+	unsigned char 	uv_conv;
+	unsigned char	ip_conv;
+} T_ISU_IN;
+
+typedef struct {
+	void 		*addr;					/* Y or RGB buffer address */
+	void 		*addr_c;				/* CbCr or CB buffer address */
+	unsigned int 	stride;
+	unsigned int 	stride_c;
+	unsigned char 	format;
+	unsigned char 	swap;
+	T_ISU_CSC	*csc;
+	T_ISU_ALPHA     *alpha;
+} T_ISU_OUT;
+
+typedef struct {
+	unsigned int	start_x;
+	unsigned int	start_y;
+	unsigned int	tune_x;
+	unsigned int	tune_y;
+	unsigned int	crop_w;
+	unsigned int	crop_h;
+	unsigned char	pad_mode;
+	unsigned int	pad_val;
+	unsigned int	x_ratio;
+	unsigned int	y_ratio;
+} T_ISU_RS;
+
+typedef struct {
+	T_ISU_IN 	*src_par;				/* source2 parameter */
+	T_ISU_OUT 	*dst_par;				/* destination parameter */
+	T_ISU_RS 	*rs_par;				/* source4 parameter */
+	T_ISU_DL 	dl_par;			/* module parameter */
+} T_ISU_START;
+
+
 /* R-Car H2/M2 compatible defines */
 #define R_FDPM_OK		(0)
 #define R_FDPM_NG		(-1)
@@ -518,16 +599,28 @@ typedef struct {
 
 typedef void			(*FP)(void);
 typedef T_VSP_START		VSPM_VSP_PAR;
+typedef T_ISU_START		VSPM_ISU_PAR;
 typedef void			VSPM_2DDMAC_PAR;
 
 typedef struct {
 	unsigned short uhType;
 	union {
 		VSPM_VSP_PAR *ptVsp;
+		VSPM_ISU_PAR *ptisu;
 		VSPM_2DDMAC_PAR *pt2dDmac;
 	} unionIpParam;
 } VSPM_IP_PAR;
 
+/* ISU IP convert parameters */
+//typedef struct {
+//	unsigned short uhType;
+//	union {
+//		VSPM_VSP_PAR *ptVsp;
+//		VSPM_2DDMAC_PAR *pt2dDmac;
+//	} unionIpParam;
+//} VSPM_ISU_IPC_PAR;
+
+
 /* R-Car H2/M2 compatible APIs */
 long VSPM_lib_DriverInitialize(
 	unsigned long *handle);
-- 
2.17.1

