From 2f65ffd4cfe36a1eae8e567e488400fb8344ed74 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Mon, 19 Oct 2020 13:17:28 +0700
Subject: [PATCH 160/445] ARM: dts: r8a7742: Use SYSC PM domain for cmt{0,1}

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index d46a4b9..b80bfab 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1956,7 +1956,7 @@
 				     <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7742_CLK_CMT0>;
 			clock-names = "fck";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 
 			renesas,channels-mask = <0x60>;
 
@@ -1977,7 +1977,7 @@
 				     <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp3_clks R8A7742_CLK_CMT1>;
 			clock-names = "fck";
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 
 			renesas,channels-mask = <0xff>;
 
-- 
2.7.4

