From dcbb6be2616c248fbd2f2a6e7b3daae0a5eb37af Mon Sep 17 00:00:00 2001
From: Hung Dong <hung.dong.xd@rvc.renesas.com>
Date: Tue, 8 Oct 2019 10:34:02 +0700
Subject: [PATCH 355/439] ARM: dts: r8a77470: Add DU support to SoC dtsi

Signed-off-by: Hung Dong <hung.dong.xd@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a77470.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index ffb096f..3d9095f 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -511,6 +511,32 @@
 			dma-channels = <2>;
 		};
 
+		vspdu0: vspdu0@fe930000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe930000 0 0x7404>;
+			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A77470_CLK_VSP1DU0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+		};
+
+		du: display@feb00000 {
+			compatible = "renesas,du-r8a77470";
+			reg = <0 0xfeb00000 0 0x40000>;
+			reg-names = "du";
+			interrupts = <0 256 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp7_clks R8A77470_CLK_DU0>;
+			power-domains = <&sysc R8A77470_PD_ALWAYS_ON>;
+			clock-names = "du.0";
+			status = "disabled";
+
+			vsps = <&vspdu0>;
+
+			port@0 {
+				du_out_rgb0: endpoint {
+				};
+			};
+		};
+
 		dmac0: dma-controller@e6700000 {
 			compatible = "renesas,dmac-r8a77470",
 				     "renesas,rcar-dmac";
-- 
2.7.4

