From c7939e701c2bd332d1c0020c98ca57ade7d4ba8d Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Mon, 19 Mar 2018 20:31:31 +0700
Subject: [PATCH 024/504] ARM: dts: r8a7742: Add sysc support to SoC dtsi

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 209afcb..1f7058b 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -145,6 +145,12 @@
 		status = "disabled";
 	};
 
+	sysc: system-controller@e6180000 {
+		compatible = "renesas,r8a7742-sysc";
+		reg = <0 0xe6180000 0 0x0200>;
+		#power-domain-cells = <1>;
+	};
+
 	clocks {
 		#address-cells = <2>;
 		#size-cells = <2>;
-- 
2.7.4

