From e3f3f2626645309cdf14cd1bb852ee2c6e14f3c8 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Tue, 17 Apr 2018 09:53:48 +0700
Subject: [PATCH 079/504] ARM: dts r8a7742: Enable UHS SDR50 for sdhi2

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 9f9bf57..2116904 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -246,6 +246,9 @@
 		dmas = <&dmac1 0xc1>, <&dmac1 0xc2>;
 		dma-names = "tx", "rx";
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,clk-rate = <97500000>;
+		renesas,mmc-scc-tapnum = <8>;
+		sd-uhs-sdr50;
 		status = "disabled";
 	};
 
-- 
2.7.4

