From 09e216e85bc8b64c68a77dd3b47cf0799d6a48a2 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Wed, 15 Jul 2020 10:19:14 +0700
Subject: [PATCH 101/504] ARM: dts r8a7742: Add device node support CAN

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 2116904..596160c 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -295,6 +295,26 @@
 		#power-domain-cells = <1>;
 	};
 
+	can0: can@e6e80000 {
+		compatible = "renesas,can-r8a7742";
+		reg = <0 0xe6e80000 0 0x1000>;
+		interrupts = <0 186 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp9_clks R8A7742_CLK_RCAN0>;
+		clock-names = "clkp1";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+	};
+
+	can1: can@e6e88000 {
+		compatible = "renesas,can-r8a7742";
+		reg = <0 0xe6e88000 0 0x1000>;
+		interrupts = <0 187 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp9_clks R8A7742_CLK_RCAN1>;
+		clock-names = "clkp1";
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		status = "disabled";
+	};
+
 	clocks {
 		#address-cells = <2>;
 		#size-cells = <2>;
-- 
2.7.4

