From 1e87b7e6a8e482cc719013196b300380ebb7195d Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Tue, 21 Jul 2020 08:31:37 +0700
Subject: [PATCH 205/504] ARM: dts iwg21m: add power-supply support buzzer

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 44d0bbe..3d72720 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -47,6 +47,7 @@
 		pwms = <&tpu 0 5000000 GPIO_ACTIVE_HIGH>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <7>;
+		power-supply = <&vcc_buzzer>;
 		pinctrl-0 = <&buzzer_pins>;
 		pinctrl-names = "default";
 	};
@@ -113,6 +114,14 @@
 		clock-output-names = "audio_clock";
 	};
 
+	vcc_buzzer: regulator@5 {
+		compatible = "regulator-fixed";
+		regulator-name = "buzzer vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
 	vcc_sdhi2: fixedregulator@1 {
 		compatible = "regulator-fixed";
 		regulator-name = "SDHI2 Vcc";
-- 
2.7.4

