From 90c508be2d0a21e389d5bff8c38dd83e8645b753 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Wed, 22 Jul 2020 10:18:59 +0700
Subject: [PATCH 243/504] ARM: dts: iwg21m: Add RTC node into i2c0

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 00090a4..a06fdf8 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -207,6 +207,11 @@
 
 	clock-frequency = <400000>;
 	status = "okay";
+
+	rtc@68 {
+		compatible = "bq32000";
+		reg = <0x68>;
+	};
 };
 
 &i2c1 {
-- 
2.7.4

