From 72d0014842989cd370340179e5ed7a2503487ec6 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Wed, 22 Jul 2020 09:04:24 +0700
Subject: [PATCH 280/504] ARM: dts iwg21m: Add qspi and msiof0 support

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 60 ++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index a06fdf8..730a0cf 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -185,6 +185,17 @@
 	clock-frequency = <20000000>;
 };
 
+/* When “GP0_18” pin is set to low, QSPI is selected
+ * When “GP0_18” pin is set to high, VIN2 is selected */
+&gpio0 {
+	enable-qspi{
+		gpio-hog;
+		gpios = <18 GPIO_ACTIVE_HIGH>;
+		output-low;
+		line-name = "enable-qspi";
+	};
+};
+
 &gpio1 {
 	can1-trx-en-gpio{
 		gpio-hog;
@@ -326,6 +337,11 @@
 		function = "mmc1";
 	};
 
+	msiof0_pins: spi1 {
+		groups = "msiof0_clk", "msiof0_tx", "msiof0_rx";
+		function = "msiof0";
+	};
+
 	pwm0_pins: pwm0 {
 		groups = "pwm0";
 		function = "pwm0";
@@ -351,6 +367,11 @@
 		function = "pwm6";
 	};
 
+	qspi_pins: spi0 {
+		groups = "qspi_ctrl", "qspi_data2";
+		function = "qspi";
+	};
+
 	scif0_pins: serial0 {
 		groups = "scif0_data";
 		function = "scif0";
@@ -404,6 +425,25 @@
 	status = "okay";
 };
 
+&msiof0 {
+	pinctrl-0 = <&msiof0_pins>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
+
+	status = "okay";
+
+	flash1: flash1@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "sst,sst25vf016b";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+		m25p,fast-read;
+	};
+};
+
 &pwm0 {
 	pinctrl-0 = <&pwm0_pins>;
 	pinctrl-names = "default";
@@ -439,6 +479,26 @@
 	status = "okay";
 };
 
+&qspi {
+	pinctrl-0 = <&qspi_pins>;
+	pinctrl-names = "default";
+
+	status = "okay"; /* multiplexed with VI2 */
+
+	flash: flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "sst,sst25vf016b";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+		m25p,fast-read;
+		spi-cpol;
+		spi-cpha;
+	};
+};
+
 &rcar_sound {
 	pinctrl-0 = <&sound_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

