From 745943e3f3681f2d652de426850b18df65a31502 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Mon, 20 Jul 2020 15:22:19 +0700
Subject: [PATCH 406/504] ARM: dts: iwg21m: Add LCD panel support for r8a7742

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 43 ++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 656ae2c..7fd7154 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -47,6 +47,15 @@
 		#size-cells = <1>;
 	};
 
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&tpu 2 5000000 GPIO_ACTIVE_HIGH>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <7>;
+		pinctrl-0 = <&backlight_pins>;
+		pinctrl-names = "default";
+	};
+
 	buzzer {
 		compatible = "pwm-backlight";
 		pwms = <&tpu 0 5000000 GPIO_ACTIVE_HIGH>;
@@ -155,6 +164,35 @@
 		states = <3300000 1
 			  1800000 0>;
 	};
+
+	vcc_lvds: regulator@4 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "panel vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	panel_edt: panel@0 {
+		compatible = "edt,etm0700g0dh6", "simple-panel";
+		power-supply = <&vcc_lvds>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0>;
+				panel_input: endpoint {
+					reg = <0>;
+					remote-endpoint = <&du_out_lvds0>;
+				};
+			};
+		};
+	};
 };
 
 &avb {
@@ -358,6 +396,11 @@
 		function = "eth";
 	};
 
+	backlight_pins: backlight {
+		groups = "tpu0_to2";
+		function = "tpu0";
+	};
+
 	fan_pins: fan {
 		groups = "tpu0_to1";
 		function = "tpu0";
-- 
2.7.4

