From 69c278a6f6f8b5718b87190d1e6a0c9d52209227 Mon Sep 17 00:00:00 2001
From: Tung Vo <tung.vo.eb@rvc.renesas.com>
Date: Fri, 16 Mar 2018 15:29:14 +0700
Subject: [PATCH 407/504] ARM: dts: iwg21m: Add power-supply support backlight

Signed-off-by: Tung Vo <tung.vo.eb@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 7fd7154..4044fca 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -52,6 +52,7 @@
 		pwms = <&tpu 2 5000000 GPIO_ACTIVE_HIGH>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <7>;
+		power-supply = <&vcc_lvds>;
 		pinctrl-0 = <&backlight_pins>;
 		pinctrl-names = "default";
 	};
-- 
2.7.4

