From 5ac894b5d704cc62d6d6188fa8c2d489b20e2a30 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 28 Jul 2020 15:51:51 +0700
Subject: [PATCH 431/504] ARM: dts r8a7742: Add description for watchdog

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7742.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index dc2f761..e7cbabf 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -1392,6 +1392,14 @@
 		};
 	};
 
+	rwdt: watchdog@e6020000 {
+		compatible = "renesas,r8a7742-wdt", "renesas,rcar-gen2-wdt";
+		reg = <0 0xe6020000 0 0x0c>;
+		clocks = <&mstp4_clks R8A7742_CLK_RWDT>;
+		timeout-sec = <60>;
+		status = "disabled";
+	};
+
 	i2c0: i2c@e6508000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
2.7.4

