From d2ab36ffc8481a59ec025905dd04e50e210d0245 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 22 Jul 2019 13:28:28 +0700
Subject: [PATCH 476/504] usb: renesas_usbhs: Add compatibility for r8a77470

Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 drivers/usb/renesas_usbhs/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/renesas_usbhs/common.c b/drivers/usb/renesas_usbhs/common.c
index 2fa4479..9884376 100644
--- a/drivers/usb/renesas_usbhs/common.c
+++ b/drivers/usb/renesas_usbhs/common.c
@@ -477,6 +477,10 @@ static const struct of_device_id usbhs_of_match[] = {
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
 	},
 	{
+		.compatible = "renesas,usbhs-r8a77470",
+		.data = (void *)USBHS_TYPE_RCAR_GEN2,
+	},
+	{
 		/* Gen3 is compatible with Gen2 */
 		.compatible = "renesas,usbhs-r8a7795",
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
-- 
2.7.4

