From 0a2668e8738be465cbdd69e979a3e4bd92d7e205 Mon Sep 17 00:00:00 2001
From: vietn <vietn@fsoft.com.vn>
Date: Fri, 6 Apr 2018 11:19:45 +0700
Subject: [PATCH 484/504] usb: host: xhci-plat: Add support for r8a7742

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 drivers/usb/host/xhci-plat.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 4dc008b..57630a9 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -55,7 +55,8 @@ static void xhci_plat_quirks(struct device *dev, struct xhci_hcd *xhci)
 	 * xhci_gen_setup().
 	 */
 	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7743") ||
-	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744"))
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744") ||
+		of_device_is_compatible(of_node, "renesas,xhci-r8a7742"))
 		xhci->quirks |= XHCI_NO_64BIT_SUPPORT;
 }
 
@@ -67,6 +68,7 @@ static int xhci_plat_setup(struct usb_hcd *hcd)
 
 	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7790") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7791") ||
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7742") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7743") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744")) {
 		ret = xhci_rcar_init_quirk(hcd);
@@ -83,6 +85,7 @@ static int xhci_plat_start(struct usb_hcd *hcd)
 
 	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7790") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7791") ||
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7742") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7743") ||
 	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744"))
 		xhci_rcar_start(hcd);
@@ -286,6 +289,7 @@ static const struct of_device_id usb_xhci_of_match[] = {
 	{ .compatible = "marvell,armada-380-xhci"},
 	{ .compatible = "renesas,xhci-r8a7743"},
 	{ .compatible = "renesas,xhci-r8a7744"},
+	{ .compatible = "renesas,xhci-r8a7742"},
 	{ .compatible = "renesas,xhci-r8a7790"},
 	{ .compatible = "renesas,xhci-r8a7791"},
 	{ },
-- 
2.7.4

