From dd653636c1ec32740432ffbcc6a58fb1955dd16e Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Date: Mon, 3 Aug 2020 16:24:26 +0700
Subject: [PATCH 491/504] ARM: dts: iwg21m: enable hsusb

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index af7d996..ba91fd2 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -435,6 +435,10 @@
 	};
 };
 
+&hsusb {
+	status = "okay";
+};
+
 &pcie_bus_clk {
 	status = "okay";
 };
-- 
2.7.4

