From c88b2b194cfe3e955bb80f2b4b9e8fbd1ec900c7 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 4 Aug 2020 12:55:56 +0700
Subject: [PATCH 494/504] usb: renesas_usbhs: Add compatibility for r8a7745

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/usb/renesas_usbhs/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/renesas_usbhs/common.c b/drivers/usb/renesas_usbhs/common.c
index a21d377..aba2dc0 100644
--- a/drivers/usb/renesas_usbhs/common.c
+++ b/drivers/usb/renesas_usbhs/common.c
@@ -481,6 +481,10 @@ static const struct of_device_id usbhs_of_match[] = {
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
 	},
 	{
+		.compatible = "renesas,usbhs-r8a7745",
+		.data = (void *)USBHS_TYPE_RCAR_GEN2,
+	},
+	{
 		.compatible = "renesas,usbhs-r8a7794",
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
 	},
-- 
2.7.4

