From b71cd6b3dc614102cd0b6d879e6d9eea97d66156 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 4 Aug 2020 16:28:46 +0700
Subject: [PATCH 496/504] ARM: dts: r8a7745: Use sysc power domain for VSP

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 09e6677..28f0c5f 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1429,7 +1429,7 @@
 			reg = <0 0xfe928000 0 0x8000>;
 			interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1_SY>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 
 			renesas,has-lut;
 			renesas,has-sru;
@@ -1443,7 +1443,7 @@
 			reg = <0 0xfe930000 0 0x8000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 
 			renesas,has-lif;
 			renesas,has-lut;
@@ -1457,7 +1457,7 @@
 			reg = <0 0xfe930000 0 0x7404>;
 			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
-			power-domains = <&cpg_clocks>;
+			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
 		fdp0: fdpm@fe940000 {
-- 
2.7.4

