From 80f718d7667b86dc556ad0d338463a3af76b8aad Mon Sep 17 00:00:00 2001
From: "Pham Thanh Tam [FGA.AIS]" <tampt8@fsoft.com.vn>
Date: Mon, 22 Jul 2019 13:28:28 +0700
Subject: [PATCH 281/629] usb: renesas_usbhs: add compatible for r8a77470

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Pham Thanh Tam [FGA.AIS] <tampt8@fsoft.com.vn>
---
 drivers/usb/renesas_usbhs/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/renesas_usbhs/common.c b/drivers/usb/renesas_usbhs/common.c
index 87e08eb63aaa..d46c9e466d8a 100644
--- a/drivers/usb/renesas_usbhs/common.c
+++ b/drivers/usb/renesas_usbhs/common.c
@@ -480,6 +480,10 @@ static const struct of_device_id usbhs_of_match[] = {
 		.compatible = "renesas,usbhs-r8a7745",
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
 	},
+	{
+		.compatible = "renesas,usbhs-r8a77470",
+		.data = (void *)USBHS_TYPE_RCAR_GEN2,
+	},
 	{
 		/* Gen3 is compatible with Gen2 */
 		.compatible = "renesas,usbhs-r8a7795",
-- 
2.17.1

