From 0176714ee73cc8b025befd756ff758d31c0990ee Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Thu, 18 Oct 2018 11:19:53 +0700
Subject: [PATCH 410/629] clk: shmobile: rcar-gen2: enable CPG support for
 r8a7742

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 drivers/clk/shmobile/clk-rcar-gen2.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/clk/shmobile/clk-rcar-gen2.c b/drivers/clk/shmobile/clk-rcar-gen2.c
index 3b2db3f585c9..d2889a4416a2 100644
--- a/drivers/clk/shmobile/clk-rcar-gen2.c
+++ b/drivers/clk/shmobile/clk-rcar-gen2.c
@@ -347,6 +347,11 @@ static const struct of_device_id cpg_of_match[] = {
 		.data = &var_pll0_ratio,
 	},
 
+	{
+		.compatible = "renesas,r8a7742-cpg-clocks",
+		.data = &var_pll0_ratio,
+	},
+
 	{
 		.compatible = "renesas,r8a7791-cpg-clocks",
 		.data = &var_pll0_ratio,
-- 
2.17.1

