From 1a66e63b0d6ce8111092e6e00bc59fad894dd512 Mon Sep 17 00:00:00 2001
From: thongsyho <thong.ho.px@rvc.renesas.com>
Date: Mon, 25 Jun 2018 14:06:34 +0700
Subject: [PATCH 535/629] usb: renesas_usbhs: add support for r8a7742

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
---
 drivers/usb/renesas_usbhs/common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/renesas_usbhs/common.c b/drivers/usb/renesas_usbhs/common.c
index 7d0759aed886..ca895def7f28 100644
--- a/drivers/usb/renesas_usbhs/common.c
+++ b/drivers/usb/renesas_usbhs/common.c
@@ -472,6 +472,10 @@ static const struct of_device_id usbhs_of_match[] = {
 		.compatible = "renesas,usbhs-r8a7791",
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
 	},
+	{
+		.compatible = "renesas,usbhs-r8a7742",
+		.data = (void *)USBHS_TYPE_RCAR_GEN2,
+	},
 	{
 		.compatible = "renesas,usbhs-r8a7744",
 		.data = (void *)USBHS_TYPE_RCAR_GEN2,
-- 
2.17.1

