From 4c5df92adf4ecfaaf56609aaa801184f7ce708d2 Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@rvc.com.vn>
Date: Fri, 21 Oct 2016 22:52:45 +0700
Subject: [PATCH 15/15] fix r8a7743.dtsi clock issue

---
 arch/arm/boot/dts/r8a7743.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index 3146664..9366ea1 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -905,7 +905,7 @@
 			clock-output-names =
 				"vcp0", "vpc0", "tmu1", "pvrsrvkm",
 				"tddmac", "fdp1", "fdp0", "tmu3", "tmu2",
-				"cmt0", "vsp1-du1", "vsp1-du0", "vsps";
+				"cmt0", "tmu0", "vsp1-du1", "vsp1-du0", "vsps";
 		};
 		mstp2_clks: mstp2_clks@e6150138 {
 			compatible = "renesas,r8a7743-mstp-clocks", "renesas,cpg-mstp-clocks";
-- 
1.9.1

