From 9213328d80b1007b390302696d9f82f04e813581 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Sat, 1 Jul 2017 00:12:32 +0700
Subject: [PATCH 01/12] ARM: DTS: iwg20m: enable display

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7743-iwg20m.dts | 115 +++++++++++++++++++++++++++++++++++
 1 file changed, 115 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743-iwg20m.dts b/arch/arm/boot/dts/r8a7743-iwg20m.dts
index 55f048c..dbcff46 100644
--- a/arch/arm/boot/dts/r8a7743-iwg20m.dts
+++ b/arch/arm/boot/dts/r8a7743-iwg20m.dts
@@ -68,6 +68,52 @@
 		states = <3300000 1
 				  1800000 0>;
 	};
+
+	vcc_lvds: regulator@4 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "panel vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	x3_clk: x3-clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <148500000>;
+	};
+
+	hdmi-out {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi_con: endpoint {
+				remote-endpoint = <&adv7511_out>;
+			};
+		};
+	};
+
+	panel_hannstar: panel@0 {
+		compatible = "hannstar,hsd070pww1", "simple-panel";
+		power-supply = <&vcc_lvds>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0>;
+				panel_input: endpoint {
+					reg = <0>;
+					remote-endpoint = <&du_out_lvds0>;
+				};
+			};
+		};
+	};
 };
 
 &extal_clk {
@@ -151,6 +197,11 @@
 		renesas,groups = "i2c2";
 		renesas,function = "i2c2";
 	};
+
+	du_pins: du {
+		renesas,groups = "du_rgb888", "du_sync", "du_disp", "du_clk_out_0";
+		renesas,function = "du";
+	};
 };
 
 &scif0 {
@@ -305,3 +356,67 @@
 		reg = <0x68>;
 	};
 };
+
+&i2c5 {
+	pinctrl-names = "default";
+	status = "okay";
+	clock-frequency = <400000>;
+
+	hdmi@39 {
+		compatible = "adi,adv7511w";
+		reg = <0x39>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <5 IRQ_TYPE_LEVEL_LOW>;
+
+		adi,input-depth = <8>;
+		adi,input-colorspace = "rgb";
+		adi,input-clock = "1x";
+		adi,input-style = <1>;
+		adi,input-justification = "evenly";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				adv7511_in: endpoint {
+					remote-endpoint = <&du_out_rgb>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				adv7511_out: endpoint {
+					remote-endpoint = <&hdmi_con>;
+				};
+			};
+		};
+	};
+};
+
+&du {
+	pinctrl-0 = <&du_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	clocks = <&mstp7_clks R8A7743_CLK_DU0>,
+		 <&mstp7_clks R8A7743_CLK_DU1>,
+		 <&mstp7_clks R8A7743_CLK_LVDS0>,
+		 <&x3_clk>;
+	clock-names = "du.0", "du.1", "lvds.0",
+		      "dclkin.0";
+
+	ports {
+		port@0 {
+			endpoint {
+				remote-endpoint = <&adv7511_in>;
+			};
+		};
+		port@1 {
+			endpoint {
+				remote-endpoint = <&panel_input>;
+			};
+		};
+	};
+};
-- 
1.9.1

