From c7896003327cac8df9d54cfe438cc81382c8c80c Mon Sep 17 00:00:00 2001
From: Nam Nguyen <nam.nguyen.xb@rvc.renesas.com>
Date: Fri, 30 Jun 2017 13:06:02 +0700
Subject: [PATCH 05/20] ARM: DTS: r8a7743: add PWM module

Signed-off-by: Nam Nguyen <nam.nguyen.xb@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7743.dtsi | 57 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 57 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index 70faa0c..18d2a18 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -1584,6 +1584,63 @@
 		status = "disabled";
 	};
 
+	
+	pwm0: pwm@e6e30000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e30000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm1: pwm@e6e31000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e31000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm2: pwm@e6e32000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e32000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm3: pwm@e6e33000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e33000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm4: pwm@e6e34000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e34000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm5: pwm@e6e35000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e35000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
+	pwm6: pwm@e6e36000 {
+		compatible = "renesas,pwm-rcar";
+		reg = <0 0xe6e36000 0 0x8>;
+		clocks = <&mstp5_clks R8A7743_CLK_PWM>;
+		status = "disabled";
+		#pwm-cells = <2>;
+	};
+
 	pci0: pci@ee090000 {
 		compatible = "renesas,pci-r8a7743";
 		device_type = "pci";
-- 
2.1.4

