From 43576ddebb72c2b2003794a2ce7527e781799626 Mon Sep 17 00:00:00 2001
From: Nam Nguyen <nam.nguyen.xb@rvc.renesas.com>
Date: Fri, 30 Jun 2017 13:27:06 +0700
Subject: [PATCH 07/20] pwm: tpu: add support r8a7743 Soc

Signed-off-by: Nam Nguyen <nam.nguyen.xb@rvc.renesas.com>
---
 drivers/pwm/pwm-renesas-tpu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pwm/pwm-renesas-tpu.c b/drivers/pwm/pwm-renesas-tpu.c
index 075c1a7..a381ba2 100644
--- a/drivers/pwm/pwm-renesas-tpu.c
+++ b/drivers/pwm/pwm-renesas-tpu.c
@@ -454,6 +454,7 @@ static int tpu_remove(struct platform_device *pdev)
 static const struct of_device_id tpu_of_table[] = {
 	{ .compatible = "renesas,tpu-r8a73a4", },
 	{ .compatible = "renesas,tpu-r8a7740", },
+	{ .compatible = "renesas,tpu-r8a7743", },
 	{ .compatible = "renesas,tpu-r8a7790", },
 	{ .compatible = "renesas,tpu-sh7372", },
 	{ .compatible = "renesas,tpu", },
-- 
2.1.4

