commit be36c3e11baa4cdb5da2839c71599f3df7b370cc
Author: linhnv25 <linhnv25@fsoft.com.vn>
Date:   Sat Jan 14 12:02:25 2017 +0700

    Add match of compatible r8a7742-iwg21m_q7 for i2c driver

diff --git a/drivers/i2c/busses/i2c-rcar.c b/drivers/i2c/busses/i2c-rcar.c
index 1d33e9b..eb4f908 100644
--- a/drivers/i2c/busses/i2c-rcar.c
+++ b/drivers/i2c/busses/i2c-rcar.c
@@ -577,6 +577,8 @@ static int rcar_i2c_master_xfer(struct i2c_adapter *adap,
 	unsigned long flags;
 	int i, ret, timeout;
 
+	pm_runtime_get_sync(dev);
+
 	/*-------------- spin lock -----------------*/
 	spin_lock_irqsave(&priv->lock, flags);
 
@@ -646,6 +648,9 @@ static int rcar_i2c_master_xfer(struct i2c_adapter *adap,
 
 		ret = i + 1; /* The number of transfer */
 	}
+
+	pm_runtime_put(dev);
+	
 out:
 	if (ret < 0 && ret != -ENXIO)
 		dev_err(dev, "error %d : %x\n", ret, priv->flags);
@@ -673,6 +678,7 @@ static const struct of_device_id rcar_i2c_dt_ids[] = {
 	{ .compatible = "renesas,i2c-r8a7792", .data = (void *)I2C_RCAR_GEN2 },
 	{ .compatible = "renesas,i2c-r8a7793", .data = (void *)I2C_RCAR_GEN2 },
 	{ .compatible = "renesas,i2c-r8a7794", .data = (void *)I2C_RCAR_GEN2 },
+	{ .compatible = "renesas,i2c-r8a7742", .data = (void *)I2C_RCAR_GEN2 },
 	{ .compatible = "renesas,i2c-r8a7743", .data = (void *)I2C_RCAR_GEN2 },
 	{ .compatible = "renesas,i2c-r8a7745", .data = (void *)I2C_RCAR_GEN2 },
 	{},
@@ -749,7 +755,6 @@ static int rcar_i2c_probe(struct platform_device *pdev)
 	}
 
 	pm_runtime_enable(dev);
-	pm_runtime_get_sync(dev);
 	platform_set_drvdata(pdev, priv);
 
 	dev_info(dev, "probed\n");
@@ -763,7 +768,6 @@ static int rcar_i2c_remove(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 
 	i2c_del_adapter(&priv->adap);
-	pm_runtime_put(dev);
 	pm_runtime_disable(dev);
 
 	return 0;
