From 8ced67729167e0b42c3b82d17df839ce14a00fd1 Mon Sep 17 00:00:00 2001
From: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
Date: Fri, 24 Nov 2017 14:51:56 +0700
Subject: [PATCH 2/2] board: renesas: iwg22m_sm: remove BSP info

Signed-off-by: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
---
 board/renesas/iwg22m_sm/iwg22m_sm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/board/renesas/iwg22m_sm/iwg22m_sm.c b/board/renesas/iwg22m_sm/iwg22m_sm.c
index 6c68763..8d6733c 100644
--- a/board/renesas/iwg22m_sm/iwg22m_sm.c
+++ b/board/renesas/iwg22m_sm/iwg22m_sm.c
@@ -396,7 +396,6 @@ void print_board_info(void)
 {
         printf("\n");
         printf("Board Info:\n");
-        printf("\tBSP Version     : %s\n", BSP_VERSION);
         printf("\tSOM Version     : %s\n", SOM_VERSION);
         printf("\n");
 }
-- 
1.9.1

