From: Thong Ho <thong.ho.px@rvc.renesas.com>
Date: Fri, 6 Apr 2018 14:57:12 +0700
Subject: [PATCH] fdpm: support board G1H r8a7742

Signed-off-by: Thong Ho <thong.ho.px@rvc.renesas.com>

diff -Naur /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/fdpm_main.c fdpm/drv/fdpm_main.c
--- /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/fdpm_main.c	2018-01-31 16:26:31.000000000 +0700
+++ fdpm/drv/fdpm_main.c	2018-04-09 11:23:15.996797214 +0700
@@ -401,6 +401,8 @@
 		pdrv->pdev = pdev;
 	if (channel == 1)
 		pdrv->pdev_c2 = pdev;
+	if (channel == 2)
+		pdrv->pdev_c3 = pdev;
 
 	sema_init(&pdrv->resource_sem, 1); /* unlock */
 	sema_init(&pdrv->init_sem, 1);/* unlock */
diff -Naur /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/fdpm_sub.c fdpm/drv/fdpm_sub.c
--- /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/fdpm_sub.c	2018-01-31 20:15:00.431847248 +0700
+++ fdpm/drv/fdpm_sub.c	2018-03-28 10:58:22.000000000 +0700
@@ -145,10 +145,14 @@
 	struct fdpm_drvdata    *pdrv = priv->pdrv;
 	struct platform_device *pdev = pdrv->pdev;
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	struct platform_device *pdev_c2 = pdrv->pdev_c2;
 #endif
 
+#if defined(H2CONFIG)
+	struct platform_device *pdev_c3 = pdrv->pdev_c3;
+#endif
+
 	int                    ercd;
 	long                   sub_ercd;
 
@@ -159,12 +163,18 @@
 	pm_runtime_enable(&pdev->dev);
 	pm_runtime_get_sync(&pdev->dev);
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	pm_suspend_ignore_children(&pdev_c2->dev, true);
 	pm_runtime_enable(&pdev_c2->dev);
 	pm_runtime_get_sync(&pdev_c2->dev);
 #endif
 
+#if defined(H2CONFIG)
+	pm_suspend_ignore_children(&pdev_c3->dev, true);
+	pm_runtime_enable(&pdev_c3->dev);
+	pm_runtime_get_sync(&pdev_c3->dev);
+#endif
+
 	/* enable clock */
 	pdrv->fdpm_clk[0] = clk_get(NULL, "fdp1-0");
 	if (IS_ERR(pdrv->fdpm_clk[0])) {
@@ -217,7 +227,7 @@
 		goto exit;
 	}
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	ret_val = fdpm_lib_driver_initialize(pdev_c2, priv, &sub_ercd);
 	if (ret_val != 0) {
 		APRINT("failed to fdpm_lib_driver_initialize %d (%ld)\n",
@@ -227,6 +237,16 @@
 	}
 #endif
 
+#if defined(H2CONFIG)
+	ret_val = fdpm_lib_driver_initialize(pdev_c3, priv, &sub_ercd);
+	if (ret_val != 0) {
+		APRINT("failed to fdpm_lib_driver_initialize %d (%ld)\n",
+		       ret_val, sub_ercd);
+		ercd = -EFAULT;
+		goto exit;
+	}
+#endif
+
 	DPRINT("done\n");
 	return 0;
 
@@ -265,6 +285,9 @@
 #if defined(M2CONFIG)
 	for (i = 0; i < FDPM_FDP_NUM - 1; i++) {
 #endif
+#if defined(H2CONFIG)
+	for (i = 0; i < FDPM_FDP_NUM - 2; i++) {
+#endif
 #if defined(E2CONFIG)
 	for (i = 0; i < FDPM_FDP_NUM; i++) {
 #endif
@@ -307,6 +330,9 @@
 #if defined(M2CONFIG)
 	for (i = 0; i < FDPM_FDP_NUM - 1; i++) {
 #endif
+#if defined(H2CONFIG)
+	for (i = 0; i < FDPM_FDP_NUM - 2; i++) {
+#endif
 #if defined(E2CONFIG)
 	for (i = 0; i < FDPM_FDP_NUM; i++) {
 #endif
@@ -322,10 +348,14 @@
 		}
 	}
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	if (channel == 1)
 		fdpm_thread(priv);
 #endif
+#if defined(H2CONFIG)
+	if (channel == 2)
+		fdpm_thread(priv);
+#endif
 #if defined(E2CONFIG)
 	fdpm_thread(priv);
 #endif
@@ -364,10 +394,14 @@
 	struct fdpm_drvdata    *pdrv = priv->pdrv;
 	struct platform_device *pdev = pdrv->pdev;
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	struct platform_device *pdev_c2 = pdrv->pdev_c2;
 #endif
 
+#if defined(H2CONFIG)
+	struct platform_device *pdev_c3 = pdrv->pdev_c3;
+#endif
+
 	long                   wret;
 
 	DPRINT("called\n");
@@ -396,11 +430,16 @@
 	pm_runtime_put_sync(&pdev->dev);
 	pm_runtime_disable(&pdev->dev);
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	pm_runtime_put_sync(&pdev_c2->dev);
 	pm_runtime_disable(&pdev_c2->dev);
 #endif
 
+#if defined(H2CONFIG)
+	pm_runtime_put_sync(&pdev_c3->dev);
+	pm_runtime_disable(&pdev_c3->dev);
+#endif
+
 	DPRINT("done\n");
 	return 0;
 exit:
@@ -423,10 +462,14 @@
 
 	pdrv = priv->pdrv;
 	pdev = pdrv->pdev;
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 	struct platform_device *pdev_c2 = pdrv->pdev_c2;
 #endif
 
+#if defined(H2CONFIG)
+	struct platform_device *pdev_c3 = pdrv->pdev_c3;
+#endif
+
 	fdpm_resource_del_all_list(&pdrv->fdpm_independ.fdpm_itable);
 	for (i = 0; i < FDPM_FDP_NUM; i++) {
 		complete(&priv->set_thread[i].start);
@@ -440,11 +483,16 @@
 		if (i == 0)
 			fdpm_free_inth(pdev, 0, priv->FDP_obj[i]);
 
-#if defined(M2CONFIG)
+#if defined(M2CONFIG) || defined(H2CONFIG)
 		if (i == 1)
 			fdpm_free_inth(pdev_c2, 0, priv->FDP_obj[i]);
 #endif
 
+#if defined(H2CONFIG)
+		if (i == 2)
+			fdpm_free_inth(pdev_c3, 0, priv->FDP_obj[i]);
+#endif
+
 		ercd = drv_FDP_Quit(priv->FDP_obj[i]);
 		if (ercd) {
 			DPRINT("drv_FDP_Quit fail %d\n", ercd);
diff -Naur /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/include/fdpm_main.h fdpm/drv/include/fdpm_main.h
--- /data1/thongsyho/SOURCE/Yocto_board_iwg23s_cip_mmp_full/build/tmp/work/iwg23s-poky-linux-gnueabi/kernel-module-fdpm/1.0-r0/fdpm/drv/include/fdpm_main.h	2018-01-31 16:26:31.000000000 +0700
+++ fdpm/drv/include/fdpm_main.h	2018-04-09 11:23:47.016796406 +0700
@@ -122,6 +122,7 @@
 struct fdpm_drvdata {
 	struct platform_device *pdev;
 	struct platform_device *pdev_c2;
+	struct platform_device *pdev_c3;
 	struct class *pcls;
 	struct cdev cdev;
 	struct task_struct *task;
