From 5a66a400d438f57ec18102b0317c2e2abab17f8a Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 15 Sep 2017 14:21:28 +0700
Subject: [PATCH 2/2 for kernel module fdpm, UT bug-fix] drv: manager: fdpm_seq: correct operator and operator
 precedence

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 drv/manager/fdpm_seq.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drv/manager/fdpm_seq.c b/drv/manager/fdpm_seq.c
index fd74aca..8d85fbd 100755
--- a/drv/manager/fdpm_seq.c
+++ b/drv/manager/fdpm_seq.c
@@ -339,9 +339,9 @@ int fdpm_dec_decodeinfo2(T_FDPD_PICPAR *pic_par)
 			case 0x2:
 				decode_out = 0x12; break;
 			case 0x3:
-				if (chk_value && 0x3 == 0) {
+				if ((chk_value & 0x3) == 0) {
 					decode_out = 0x13; break;
-				} else if (chk_value && 0x3 == 1) {
+				} else if ((chk_value & 0x3) == 1) {
 					decode_out = 0x14; break;
 				} else {
 					decode_out = 0x10; break;
-- 
2.7.4

