From 60420e1b5a3538e7143ed7087e352eb03dde3f73 Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@renesas.com>
Date: Fri, 10 Nov 2017 13:46:59 +0700
Subject: [PATCH] udev:disable udev cache

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
---
 udev-cache.default | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/udev-cache.default b/udev-cache.default
index a3b7326..866fc1d 100644
--- a/udev-cache.default
+++ b/udev-cache.default
@@ -1,5 +1,5 @@
 # Default for /etc/init.d/udev
 
 # Comment this out to disable device cache
-DEVCACHE="/etc/udev-cache.tar.gz"
+#DEVCACHE="/etc/udev-cache.tar.gz"
 PROBE_PLATFORM_BUS="yes"
-- 
1.9.1

