--- a/Makefile.am	2016-03-29 18:41:32.580321638 +0300
+++ b/Makefile.am	2016-03-29 18:38:36.140327908 +0300
@@ -591,7 +591,7 @@
 	src/keymap/force-release-maps/samsung-other \
 	src/keymap/force-release-maps/common-volume-keys
 
-src/keymap/keys.txt: $(INCLUDE_PREFIX)/linux/input.h
+src/keymap/keys.txt: $(INCLUDE_PREFIX)/linux/input-event-codes.h
 	$(AM_V_at)mkdir -p src/keymap
 	$(AM_V_GEN)$(AWK) '/^#define.*KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' < $< | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
 
