diff --git a/drivers/tty/serial/8250/8250_dma.c b/drivers/tty/serial/8250/8250_dma.c
index 004dada..7c2111b 100755
--- a/drivers/tty/serial/8250/8250_dma.c
+++ b/drivers/tty/serial/8250/8250_dma.c
@@ -52,7 +52,7 @@ static void __dma_rx_complete(void *param)
 
 #if defined(CONFIG_ARCH_R9A09G011GBG) || defined(CONFIG_ARCH_R9A09G055MA3GBG)
 	if (!dma->rx_running)
-		return 
+		return;
 #endif /* defined(CONFIG_ARCH_R9A09G011GBG)  || defined(CONFIG_ARCH_R9A09G055MA3GBG) */
 	dma->rx_running = 0;
 	dmaengine_tx_status(dma->rxchan, dma->rx_cookie, &state);
