From d20681e739d35d91ed830f255635f494b9a187a1 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Tue, 14 Jul 2020 16:48:08 +0700
Subject: [PATCH 044/433] ARM: dts: r8a7745: Add operating-point for CPU0

Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 58bf2e5..8e31ae0 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -74,6 +74,7 @@
 			reg = <0>;
 			clock-frequency = <1000000000>;
 			clocks = <&z2_clk>;
+			operating-points = <1000000 1000000>;
 			next-level-cache = <&L2_CA7>;
 		};
 
-- 
2.7.4

