From df14d0288cd60bd729b8c00a766737d57312ae0b Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 14 Jul 2020 13:30:26 +0700
Subject: [PATCH 071/433] i2c: rcar: Add subsys_initcall for early timing
 startup

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 drivers/i2c/busses/i2c-rcar.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-rcar.c b/drivers/i2c/busses/i2c-rcar.c
index 75aa031..aef95a9 100644
--- a/drivers/i2c/busses/i2c-rcar.c
+++ b/drivers/i2c/busses/i2c-rcar.c
@@ -692,7 +692,18 @@ static struct platform_driver rcar_i2c_driver = {
 	.remove		= rcar_i2c_remove,
 };
 
-module_platform_driver(rcar_i2c_driver);
+static int __init rcar_i2c_adap_init(void)
+{
+	return platform_driver_register(&rcar_i2c_driver);
+}
+
+static void __exit rcar_i2c_adap_exit(void)
+{
+	platform_driver_unregister(&rcar_i2c_driver);
+}
+
+subsys_initcall(rcar_i2c_adap_init);
+module_exit(rcar_i2c_adap_exit);
 
 MODULE_LICENSE("GPL v2");
 MODULE_DESCRIPTION("Renesas R-Car I2C bus driver");
-- 
2.7.4

