From 3cbcf6bb377ac4b8c3e61094ee9e827d68d1ea32 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Tue, 10 Apr 2018 08:23:06 +0700
Subject: [PATCH 115/433] spi: spidev: Add compatible for r8a77470

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 8ff1633..964d99f 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -699,6 +699,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
 	{ .compatible = "renesas,spidev-r8a7745" },
+	{ .compatible = "renesas,spidev-r8a77470" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

