From 1700b083e6d64285b71766b9232d7b34eee755aa Mon Sep 17 00:00:00 2001
From: Tin Le <tin.le.wh@rvc.renesas.com>
Date: Wed, 25 Dec 2019 16:38:49 +0700
Subject: [PATCH 149/433] phy: rcar-gen2: Add support for r8a77470 USB OTG

Signed-off-by: Tin Le <tin.le.wh@rvc.renesas.com>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/phy/phy-rcar-gen2.c | 93 +++++++++++++++++++++++++++++++++------------
 1 file changed, 69 insertions(+), 24 deletions(-)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index a4002eb..2a27ee3 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -48,6 +48,8 @@
 #define USBHS_UGCTRL2_USB0SEL_USB20	0x00000010
 #define USBHS_UGCTRL2_USB0SEL_HS_USB20	0x00000020
 
+#define USBHS_UGCTRL2_USB0SEL_HSUSB_R8A77470	0x00000020
+
 /* USB General status register (UGSTS) */
 #define USBHS_UGSTS_LOCK		0x00000100 /* From technical update */
 
@@ -158,7 +160,13 @@ static int rcar_gen2_phy_init(struct phy *p)
 		 * which role to use.
 		 */
 		clk_prepare_enable(drv->clk);
-		schedule_delayed_work(&channel->work_vbus, 100);
+		if (gpio_is_valid(drv->channels->gpio_vbus))
+			schedule_delayed_work(&channel->work_vbus, 100);
+		else {
+			if (cmpxchg(&channel->selected_phy, -1, phy->number) != -1)
+				return -EBUSY;
+			rcar_gen2_phy_switch(channel, phy->select_value);
+		}
 	}
 #else
 	if (cmpxchg(&channel->selected_phy, -1, phy->number) != -1)
@@ -166,6 +174,8 @@ static int rcar_gen2_phy_init(struct phy *p)
 
 	clk_prepare_enable(drv->clk);
 
+	rcar_gen2_phy_switch(channel, phy->select_value);
+#endif
 	if (of_machine_is_compatible("renesas,r8a77470")) {
 		/* Initialize USB2 part */
 		writel(USB20_INT_ENABLE_INIT, drv->usb20_base + USB20_INT_ENABLE_REG);
@@ -173,8 +183,6 @@ static int rcar_gen2_phy_init(struct phy *p)
 		writel(USB20_OC_TIMSET_INIT, drv->usb20_base + USB20_OC_TIMSET_REG);
 	}
 
-	rcar_gen2_phy_switch(channel, phy->select_value);
-#endif
 	return 0;
 }
 
@@ -389,8 +397,23 @@ static void gpio_id_work(struct work_struct *work)
 		else if (of_machine_is_compatible("renesas,r8a7743"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
+		else if (of_machine_is_compatible("renesas,r8a77470"))
+			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_USB20);
+	} else {
+		if (of_machine_is_compatible("renesas,r8a7745"))
+			rcar_gen2_phy_switch(channel,
+				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_PCI);
+		else if (of_machine_is_compatible("renesas,r8a7743"))
+			rcar_gen2_phy_switch(channel,
+				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
+		else if (of_machine_is_compatible("renesas,r8a77470"))
+			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HSUSB_R8A77470);
 	}
 
+	/* FIXME: hard set r8a77470 USB OTG function mode */
+	if (of_machine_is_compatible("renesas,r8a77470"))
+		rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HSUSB_R8A77470);
+
 	/* If VBUS is powered and we are not the initial Host, turn off VBUS */
 	if (gpio_is_valid(channel->gpio_vbus_pwr))
 		gpio_direction_output(channel->gpio_vbus_pwr, !(id));
@@ -416,6 +439,8 @@ static void gpio_vbus_work(struct work_struct *work)
 		else if (of_machine_is_compatible("renesas,r8a7743"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_HS_USB | USBHS_UGCTRL2_USB2SEL_USB30);
+		else if (of_machine_is_compatible("renesas,r8a77470"))
+			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_HSUSB_R8A77470);
 	} else {
 		if (of_machine_is_compatible("renesas,r8a7745"))
 			rcar_gen2_phy_switch(channel,
@@ -423,6 +448,8 @@ static void gpio_vbus_work(struct work_struct *work)
 		else if (of_machine_is_compatible("renesas,r8a7743"))
 			rcar_gen2_phy_switch(channel,
 				USBHS_UGCTRL2_USB0SEL_PCI | USBHS_UGCTRL2_USB2SEL_USB30);
+		else if (of_machine_is_compatible("renesas,r8a77470"))
+			rcar_gen2_phy_switch(channel, USBHS_UGCTRL2_USB0SEL_USB20);
 	}
 
 	if (!channel->otg->gadget)
@@ -460,11 +487,11 @@ static irqreturn_t gpio_vbus_irq(int irq, void *data)
 		if (otg->gadget) {
 			queue_delayed_work(channel->work_queue,
 			&channel->work_vbus, msecs_to_jiffies(100));
-			}
-		} else {
+		}
+	} else {
 		queue_delayed_work(channel->work_queue,
 		&channel->work_id, msecs_to_jiffies(100));
-		}
+	}
 	return IRQ_HANDLED;
 }
 
@@ -480,16 +507,18 @@ static int probe_otg(struct platform_device *pdev,
 	ch->gpio_vbus = of_get_named_gpio_flags(dev->of_node,
 				"renesas,vbus-gpio", 0, NULL);
 	/* Need valid ID and VBUS gpios for Host/Fn switching */
-	if (gpio_is_valid(ch->gpio_id) && gpio_is_valid(ch->gpio_vbus)) {
+	if (gpio_is_valid(ch->gpio_id)) {
 		ch->use_otg = 1;
 		/* GPIO for ID input */
 		ret = devm_gpio_request_one(dev, ch->gpio_id, GPIOF_IN, "id");
 		if (ret)
 			return ret;
 		/* GPIO for VBUS input */
-		ret = devm_gpio_request_one(dev, ch->gpio_vbus, GPIOF_IN, "vbus");
-		if (ret)
-			return ret;
+		if (gpio_is_valid(ch->gpio_vbus)) {
+			ret = devm_gpio_request_one(dev, ch->gpio_vbus, GPIOF_IN, "vbus");
+			if (ret)
+				return ret;
+		}
 
 		/* Optional GPIO for VBUS power */
 		ch->gpio_vbus_pwr = of_get_named_gpio_flags(dev->of_node,
@@ -520,15 +549,21 @@ static int rcar_gen2_usb_set_peripheral(struct usb_otg *otg,
 	struct rcar_gen2_channel *channel;
 
 	channel = container_of(otg->usb_phy, struct rcar_gen2_channel, usbphy);
-	if (!gadget) {
-		usb_gadget_vbus_disconnect(otg->gadget);
-		otg->state = OTG_STATE_UNDEFINED;
-		return -EPROBE_DEFER;
+	if (channel->use_otg) {
+		if (!gadget) {
+			usb_gadget_vbus_disconnect(otg->gadget);
+			otg->state = OTG_STATE_UNDEFINED;
+			return -EPROBE_DEFER;
+		}
+		otg->gadget = gadget;
+
+		/* initialize connection state */
+		if (of_machine_is_compatible("renesas,r8a7743") ||
+			of_machine_is_compatible("renesas,r8a7745")) {
+			gpio_vbus_irq(channel->irq_vbus, channel);
+		} else if (of_machine_is_compatible("renesas,r8a77470"))
+			gpio_vbus_irq(channel->irq_id, channel);
 	}
-	otg->gadget = gadget;
-
-	/* initialize connection state */
-	gpio_vbus_irq(channel->irq_vbus, channel);
 
 	return 0;
 }
@@ -706,9 +741,14 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 	drv->channels->otg      = otg;
 
 	/* USB0 Host/Function switching info */
-	err = probe_otg(pdev, drv);
-	if (err)
-		return err;
+	if (of_find_property((&pdev->dev)->of_node, "renesas,id-gpio", NULL)) {
+		err = probe_otg(pdev, drv);
+		if (err) {
+			return err;
+		}
+	} else {
+		drv->channels->usbphy.type = USB_PHY_TYPE_USB2;
+	}
 
 	if (drv->channels->use_otg) {
 		if (gpio_is_valid(drv->channels->gpio_id)) {
@@ -827,8 +867,12 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 			phy->number = n;
 			if (of_machine_is_compatible("renesas,r8a7743") || of_machine_is_compatible("renesas,r8a7745"))
 				phy->select_value = select_value[channel_num][n];
-			else if (of_machine_is_compatible("renesas,r8a77470"))
-				phy->select_value = USBHS_UGCTRL2_USB0SEL_USB20;
+			else if (of_machine_is_compatible("renesas,r8a77470")) {
+				if (drv->channels->use_otg)
+					phy->select_value = USBHS_UGCTRL2_USB0SEL_HSUSB_R8A77470;
+				else
+					phy->select_value = USBHS_UGCTRL2_USB0SEL_USB20;
+			}
 
 			phy->phy = devm_phy_create(dev, NULL,
 						   data->gen2_phy_ops);
@@ -858,7 +902,8 @@ static int rcar_gen2_phy_probe(struct platform_device *pdev)
 	* and ID signals.
 	*/
 	if (drv->channels->use_otg) {
-		schedule_delayed_work(&drv->channels->work_vbus, msecs_to_jiffies(100));
+		if (gpio_is_valid(drv->channels->gpio_vbus))
+			schedule_delayed_work(&drv->channels->work_vbus, msecs_to_jiffies(100));
 		schedule_delayed_work(&drv->channels->work_id, msecs_to_jiffies(100));
 	}
 #endif
-- 
2.7.4

