From 437068c3472b935e72725feaf2630827dcfd28cd Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@renesas.com>
Date: Thu, 20 Jul 2017 10:14:09 +0700
Subject: [PATCH 207/433] v4l: tvp5150: Add querystd function

Signed-off-by: Nhan Nguyen <nhan.nguyen.yb@renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 drivers/media/i2c/tvp5150.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/media/i2c/tvp5150.c b/drivers/media/i2c/tvp5150.c
index 00d5ecc..2d6e033 100644
--- a/drivers/media/i2c/tvp5150.c
+++ b/drivers/media/i2c/tvp5150.c
@@ -1081,6 +1081,24 @@ static int tvp5150_g_tuner(struct v4l2_subdev *sd, struct v4l2_tuner *vt)
 	return 0;
 }
 
+static int tvp5150_querystd(struct v4l2_subdev *sd, v4l2_std_id *std_id)
+{
+	struct tvp5150 *decoder = to_tvp5150(sd);
+	u8 std, current_std ;
+
+	if (std_id == NULL)
+		return -EINVAL;
+
+	std = tvp5150_read(sd, TVP5150_VIDEO_STD);
+	if ((std & VIDEO_STD_MASK) == VIDEO_STD_AUTO_SWITCH_BIT)
+		*std_id = tvp5150_read_std(sd);
+	else
+		/* use the standard register itself */
+		*std_id = V4L2_STD_UNKNOWN;
+
+	return 0;
+}
+
 /* ----------------------------------------------------------------------- */
 
 static const struct v4l2_ctrl_ops tvp5150_ctrl_ops = {
@@ -1109,6 +1127,7 @@ static const struct v4l2_subdev_video_ops tvp5150_video_ops = {
 	.g_std = tvp5150_g_std,
 	.g_tvnorms = tvp5150_g_tvnorms,
 	.g_mbus_config = tvp5150_g_mbus_config,
+	.querystd = tvp5150_querystd,
 };
 
 static const struct v4l2_subdev_vbi_ops tvp5150_vbi_ops = {
-- 
2.7.4

