From 82f4740a2f7d415306c1851d4d73bc862c59a549 Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Thu, 9 Apr 2020 13:25:53 +0700
Subject: [PATCH 057/433] ARM: debug-ll: Add support for r8a7744

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: khatran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 arch/arm/Kconfig.debug | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/Kconfig.debug b/arch/arm/Kconfig.debug
index a8d4753..b03b658 100644
--- a/arch/arm/Kconfig.debug
+++ b/arch/arm/Kconfig.debug
@@ -819,11 +819,11 @@ choice
 	config DEBUG_RCAR_GEN2_SCIF0
 		bool "Kernel low-level debugging messages via SCIF0 on R-Car Gen2 and RZ/G1"
 		depends on ARCH_R8A7743 || ARCH_R8A7744 || ARCH_R8A7790 || \
-			ARCH_R8A7791 || ARCH_R8A7793
+			ARCH_R8A7791 || ARCH_R8A7793 || ARCH_R8A7744
 		help
 		  Say Y here if you want kernel low-level debugging support
 		  via SCIF0 on Renesas RZ/G1M (R8A7743), RZ/G1N (R8A7744),
-		  R-Car H2 (R8A7790), M2-W (R8A7791), or M2-N (R8A7793).
+		  R-Car H2 (R8A7790), M2-W (R8A7791), M2-N (R8A7793), Renesas RZ/G1N (R8A7744).
 
 	config DEBUG_RCAR_GEN2_SCIF1
 		bool "Kernel low-level debugging messages via SCIF1 on R8A77470"
-- 
2.7.4

