From 36e4a7eaeda1a4bd34109f8285520e0a6206afd9 Mon Sep 17 00:00:00 2001
From: thanhbui <thanh.bui.fz@renesas.com>
Date: Tue, 14 Jul 2020 10:36:14 +0700
Subject: [PATCH 066/433] ARM: dts: iwg22m: enable leds GPIO

Signed-off-by: thanhbui <thanh.bui.fz@renesas.com>
---
 arch/arm/boot/dts/r8a7745-iwg22m.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7745-iwg22m.dtsi b/arch/arm/boot/dts/r8a7745-iwg22m.dtsi
index 29b6e10..355e92a 100644
--- a/arch/arm/boot/dts/r8a7745-iwg22m.dtsi
+++ b/arch/arm/boot/dts/r8a7745-iwg22m.dtsi
@@ -19,6 +19,23 @@
 		reg = <0 0x40000000 0 0x20000000>;
 	};
 
+	leds {
+		compatible = "gpio-leds";
+
+		led1 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+		};
+		led2 {
+			gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
+		};
+		led3 {
+			gpios = <&gpio5 7 GPIO_ACTIVE_HIGH>;
+		};
+		led4 {
+			gpios = <&gpio5 8 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
 	reg_3p3v: 3p3v {
 		compatible = "regulator-fixed";
 		regulator-name = "3P3V";
-- 
2.7.4

