From f55f5583d1414306f3de920b8c9dfdbb265c9e81 Mon Sep 17 00:00:00 2001
From: khanhluu <khanh.luu.xw@rvc.renesas.com>
Date: Tue, 31 Oct 2017 22:05:33 +0700
Subject: [PATCH 107/433] spi: spidev: Add compatible for r8a7745

Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 7969f54..8ff1633 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -698,6 +698,7 @@ static struct class *spidev_class;
 static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
+	{ .compatible = "renesas,spidev-r8a7745" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

