From eb5368f55ee6c1a2354dae74e704bcef97b9a53c Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 4 Aug 2020 12:54:03 +0700
Subject: [PATCH 153/433] phy: rcar-gen2: Add compatibility for r8a7745

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/phy/phy-rcar-gen2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index 6201f83..3eaca7b 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -619,6 +619,10 @@ static const struct of_device_id rcar_gen2_phy_match_table[] = {
 		.data = &rcar_gen2_usb_phy_data,
 	},
 	{
+		.compatible = "renesas,usb-phy-r8a7745",
+		.data = &rcar_gen2_usb_phy_data,
+	},
+	{
 		.compatible = "renesas,rcar-gen2-usb-phy",
 		.data = &rcar_gen2_usb_phy_data,
 	},
-- 
2.7.4

