From 5d262d362e417b9ca27bd598d8f81e0bd4c75b78 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Thu, 27 Jul 2017 10:58:30 +0700
Subject: [PATCH 208/433] v4l: tvp5150: remove unused variables

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 drivers/media/i2c/tvp5150.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/media/i2c/tvp5150.c b/drivers/media/i2c/tvp5150.c
index 2d6e033..442c4d4 100644
--- a/drivers/media/i2c/tvp5150.c
+++ b/drivers/media/i2c/tvp5150.c
@@ -1083,8 +1083,7 @@ static int tvp5150_g_tuner(struct v4l2_subdev *sd, struct v4l2_tuner *vt)
 
 static int tvp5150_querystd(struct v4l2_subdev *sd, v4l2_std_id *std_id)
 {
-	struct tvp5150 *decoder = to_tvp5150(sd);
-	u8 std, current_std ;
+	u8 std;
 
 	if (std_id == NULL)
 		return -EINVAL;
-- 
2.7.4

