From c23004a075da44bd95fe2aa70e66a630195dd69f Mon Sep 17 00:00:00 2001
From: Kha Tran <kha.tran.px@rvc.renesas.com>
Date: Thu, 25 Jan 2018 09:33:35 +0700
Subject: [PATCH 351/433] ARM: dts: r8a7744: Support VSP control by DU

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7744.dtsi | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 1fbb821..658f4a9 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -1585,21 +1585,15 @@
 			power-domains = <&cpg_clocks>;
 		};
 
-		vsp@fe928000 {
-			compatible = "renesas,vsp1";
+		vspm@fe928000 {
+			compatible = "renesas,vspm-vsps","renesas-vspm";
 			reg = <0 0xfe928000 0 0x8000>;
 			interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7744_CLK_VSP1_S>;
 			power-domains = <&cpg_clocks>;
-
-			renesas,has-lut;
-			renesas,has-sru;
-			renesas,#rpf = <5>;
-			renesas,#uds = <1>;
-			renesas,#wpf = <4>;
 		};
 
-		vsp@fe930000 {
+		vspd0: vspd0@fe930000 {
 			compatible = "renesas,vsp1";
 			reg = <0 0xfe930000 0 0x8000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
@@ -1613,7 +1607,7 @@
 			renesas,#wpf = <1>;
 		};
 
-		vsp@fe938000 {
+		vspd1: vspd1@fe938000 {
 			compatible = "renesas,vsp1";
 			reg = <0 0xfe938000 0 0x8000>;
 			interrupts = <GIC_SPI 247 IRQ_TYPE_LEVEL_HIGH>;
@@ -1683,6 +1677,8 @@
 			clock-names = "du.0", "du.1", "lvds.0";
 			status = "disabled";
 
+			vsps = <&vspd0 &vspd1>;
+
 			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
-- 
2.7.4

