--- a/ld/emulparams/aarch64linux.sh
+++ b/ld/emulparams/aarch64linux.sh
@@ -38,15 +38,12 @@
 
 # Linux modifies the default library search path to first include
 # a 64-bit specific directory.
-
-# not for multiarch systems ...
-
-#case "$target" in
-#  aarch64*-linux*)
-#    case "$EMULATION_NAME" in
-#      aarch64linux*) LIBPATH_SUFFIX=64 ;;
-#    esac
-#    ;;
-#esac
+case "$target" in
+  aarch64*-linux*)
+    case "$EMULATION_NAME" in
+      aarch64linux*) LIBPATH_SUFFIX=64 ;;
+    esac
+    ;;
+esac

 ELF_INTERPRETER_NAME=\"/lib/ld-linux-aarch64.so.1\"
