From 8d2284cba29bcc3ab7cc29324e57807565078984 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 11 Aug 2016 04:33:47 +0700
Subject: [PATCH 1/2] Weston-add-support-set-position-from-client

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 desktop-shell/shell.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index 3c6a3da..6383263 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -2816,6 +2816,9 @@ set_surface_type(struct shell_surface *shsurf)
						 pev->geometry.x + shsurf->transient.x,
						 pev->geometry.y + shsurf->transient.y);
		}
+		weston_view_set_position(shsurf->view,
+		shsurf->transient.x,
+		shsurf->transient.y);
		break;

	case SHELL_SURFACE_XWAYLAND:
--
1.9.1
