From e641caba3772092b78ed2d5e00826eb4947f6a8b Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 9 Jun 2016 11:02:47 +0300
Subject: [PATCH] notify client immediately on buffer release

---
 src/compositor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor.c b/src/compositor.c
index cbef4dc..1d5c9d2 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -1920,7 +1920,7 @@ weston_buffer_reference(struct weston_buffer_reference *ref,
 		ref->buffer->busy_count--;
 		if (ref->buffer->busy_count == 0) {
 			assert(wl_resource_get_client(ref->buffer->resource));
-			wl_resource_queue_event(ref->buffer->resource,
+			wl_resource_post_event(ref->buffer->resource,
 						WL_BUFFER_RELEASE);
 		}
 		wl_list_remove(&ref->destroy_listener.link);
-- 
1.9.1

