From 85dd32d28d98d15cd48fd67b7ef877fbf9a21507 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 11 Aug 2016 04:38:20 +0700
Subject: [PATCH 2/2] Weston-add-support-set-scaling-from-client

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 desktop-shell/shell.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index 6383263..1030cc6 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -2802,11 +2802,20 @@ set_surface_type(struct shell_surface *shsurf)
	struct weston_surface *pes = shsurf->parent;
	struct weston_view *pev = get_default_view(pes);

+	struct weston_matrix *matrix;
+	float scale_w, scale_h;
+	int width, height, tmp;
+	struct weston_surface *surface = shsurf->surface;
+	int32_t surf_x, surf_y, surf_width, surf_height;
+
	reset_surface_type(shsurf);

	shsurf->state = shsurf->next_state;
	shsurf->state_changed = false;

+	surface_subsurfaces_boundingbox(shsurf->surface, &surf_x, &surf_y,
+	                                &surf_width, &surf_height);
+
	switch (shsurf->type) {
	case SHELL_SURFACE_TOPLEVEL:
		if (shsurf->state.maximized || shsurf->state.fullscreen) {
@@ -2816,6 +2825,23 @@ set_surface_type(struct shell_surface *shsurf)
						 pev->geometry.x + shsurf->transient.x,
						 pev->geometry.y + shsurf->transient.y);
		}
+
+		/*call scale function*/
+		matrix = &shsurf->fullscreen.transform.matrix;
+		weston_matrix_init(matrix);
+        /*calculate scaling ratio*/
+		if ((shsurf->transient.flags) != 0){
+			tmp = shsurf->transient.flags;
+			width = (shsurf->transient.flags) >> 16;
+			height = tmp & 0xffff;
+			scale_w = (float) width / (float) surf_width;
+			scale_h = (float) height / (float) surf_height;
+			weston_matrix_scale(matrix, scale_w, scale_h, 1);
+			wl_list_remove(&shsurf->fullscreen.transform.link);
+			wl_list_insert(&shsurf->view->geometry.transformation_list,
+			       &shsurf->fullscreen.transform.link);
+		}
+
		weston_view_set_position(shsurf->view,
		shsurf->transient.x,
		shsurf->transient.y);
--
1.9.1
