From 309e200b82f1232fba6b553a2f630e7a7ae4c015 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 22 Sep 2016 18:38:17 +0300
Subject: [PATCH 2/4] Fix memory leaks

---
 src/v4l2-renderer.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/v4l2-renderer.c b/src/v4l2-renderer.c
index 69aab92..734a2b3 100644
--- a/src/v4l2-renderer.c
+++ b/src/v4l2-renderer.c
@@ -1421,8 +1421,8 @@ static int
 v4l2_renderer_init(struct weston_compositor *ec, int drm_fd, char *drm_fn)
 {
 	struct v4l2_renderer *renderer;
-	char *device;
-	const char *device_name;
+	char *device = NULL;
+	const char *device_name = NULL;
 	const struct media_device_info *info;
 	struct weston_config_section *section;
 
@@ -1531,9 +1531,17 @@ v4l2_renderer_init(struct weston_compositor *ec, int drm_fd, char *drm_fn)
 
 	wl_signal_init(&renderer->destroy_signal);
 
+	free((void *)device_name);
+	free((void *)device);
+
 	return 0;
 
 error:
+	if (device_name)
+		free((void *)device_name);
+	if (device)
+		free((void *)device);
+
 	free(renderer);
 	weston_log("V4L2 renderer initialization failed.\n");
 	return -1;
-- 
1.9.1

