From eadcf4d2cd152acf5c63a4cd80875bd02352bc7e Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Fri, 23 Sep 2016 15:53:38 +0300
Subject: [PATCH 4/4] Fix gl-fallback

---
 src/v4l2-renderer.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/v4l2-renderer.c b/src/v4l2-renderer.c
index 9ef252a..308caf7 100644
--- a/src/v4l2-renderer.c
+++ b/src/v4l2-renderer.c
@@ -174,8 +174,8 @@ v4l2_create_gl_renderer(struct weston_compositor *ec, struct v4l2_renderer *rend
 {
 	EGLint format = GBM_FORMAT_XRGB8888;
 
-	if (gl_renderer->create(ec, renderer->gbm, EGL_PLATFORM_WAYLAND_KHR,
-				gl_renderer->opaque_attribs, &format, 0) < 0) {
+	if (gl_renderer->create(ec, EGL_PLATFORM_WAYLAND_KHR, renderer->gbm,
+				gl_renderer->opaque_attribs, &format, 1) < 0) {
 		return -1;
 	}
 	renderer->gl_renderer = ec->renderer;
@@ -214,7 +214,7 @@ v4l2_init_gl_output(struct weston_output *output, struct v4l2_renderer *renderer
 	output->renderer_state = NULL;
 	read_format = output->compositor->read_format;
 	if (gl_renderer->output_create(output, state->gbm_surface, state->gbm_surface,
-				       gl_renderer->opaque_attribs, &format, 0) < 0) {
+				       gl_renderer->opaque_attribs, &format, 1) < 0) {
 		weston_log("%s: failed to create gl renderer output state\n", __func__);
 		gbm_surface_destroy(state->gbm_surface);
 		return -1;
-- 
1.9.1

