From bc55cc496b3da653be5d5a2fdfe7f7798295efa6 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 6 Oct 2016 12:47:35 +0300
Subject: [PATCH] workaround-rcar-crtc

---
 src/compositor-drm.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index e47f680..32050cf 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -1869,7 +1869,9 @@ find_crtc_for_connector(struct drm_backend *b,
 		for (i = 0; i < resources->count_crtcs; i++) {
 			if (possible_crtcs & (1 << i) &&
 			    !(b->crtc_allocator & (1 << resources->crtcs[i])))
-				return i;
+				/* workaround RCAR crtc */
+				if (!(possible_crtcs > 1 && i == 0))
+					return i;
 		}
 	}
 
-- 
1.9.1

