diff --git a/Makefile.am b/Makefile.am
index a7cb39e..aea992d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -178,14 +178,14 @@ v4l2_renderer_la_SOURCES =			\
 	src/v4l2-renderer.h			\
 	src/v4l2-device.h
 
-module_LTLIBRARIES += v4l2-vsp1-device.la
-v4l2_vsp1_device_la_LDFLAGS = -module -avoid-version
-v4l2_vsp1_device_la_LIBADD = $(COMPOSITOR_LIBS) $(LIBMEDIACTL_V4L2_LIBS)
-v4l2_vsp1_device_la_CFLAGS =				\
+module_LTLIBRARIES += v4l2-vsp2-device.la
+v4l2_vsp2_device_la_LDFLAGS = -module -avoid-version
+v4l2_vsp2_device_la_LIBADD = $(COMPOSITOR_LIBS) $(LIBMEDIACTL_V4L2_LIBS)
+v4l2_vsp2_device_la_CFLAGS =				\
 	$(COMPOSITOR_CFLAGS)			\
 	$(LIBMEDIACTL_V4L2_CFLAGS)		\
 	$(GCC_CFLAGS)
-v4l2_vsp1_device_la_SOURCES =			\
+v4l2_vsp2_device_la_SOURCES =			\
 	src/vsp-renderer.c			\
 	src/v4l2-device.h
 endif
diff --git a/src/vsp-renderer.c b/src/vsp-renderer.c
index ac197de..d6641a0 100644
--- a/src/vsp-renderer.c
+++ b/src/vsp-renderer.c
@@ -284,8 +284,8 @@ vsp_init(struct media_device *media, struct weston_config *config)
 	else
 		device_name = info->bus_info;
 
-	if (strncmp(device_name, "vsp1.", 5)) {
-		weston_log("The device is not VSP1.");
+	if (strncmp(device_name, "vsp2.", 5)) {
+		weston_log("The device is not VSP2.");
 		goto error;
 	}
 
